{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project\\\n",
    "--match 'author: (cluster)-[]->(qnode), nih_author:(qnihauthor)-[]->(qnode), inv:(project)-[]->(qnihauthor), \\\n",
    "        item:(qnode)-[:P108]->(org_node), label:(org_node)-[]->(org_name),label:(qnode)-[]->(author_name), \\\n",
    "        coor:(org_node)-[:P625]->(organization_coor)'\\\n",
    "--opt 'project: (project)-[:PNIHcost]->(cost)' \\\n",
    "--return 'cluster, author_name, org_name,project, organization_coor,cost' \\\n",
    "--limit 50# Generating Useful Wikidata Files\n",
    "\n",
    "This notebook generates files that contain derived data that is useful in many applications. The input to the notebook is the full Wikidata or a subset of Wikidata. It also works for arbutrary KGs as long as they follow the representation requirements of Wikidata:\n",
    "\n",
    "- the *instance of* relation is represented using the `P31` property\n",
    "- the *subclass of* relation is represented using the `P279` property\n",
    "- all properties declare a datatype, and the data types must be one of the datatypes in Wikidata.\n",
    "\n",
    "Inputs:\n",
    "\n",
    "- `claims_file`: contains all statements, which consist of edges `node1/label/node2` where `label` is a property in Wikidata (e.g., sitelinks, labels, aliases and description are not in the claims file.\n",
    "- `item_file`: the subset of the `claims_file` consistin of edges for property of data type `wikibase-item`\n",
    "- `label_file`, `alias_file` and `description_file` containing labels, aliases and descriptions. It is assume that these files contain the labels, aliases and descriptions of all nodes appearing in the claims file. Users may provide these files for specific languages only.\n",
    "\n",
    "Outputs:\n",
    "\n",
    "- **Instance of (P31):** `derived.P31.tsv.gz` contains all the `instance of (P31)` edges present in the claims file.\n",
    "- **Subclass of (P279):** `derived.P279.tsv.gz` contains all the `subclass of (P279)` edges present in the claims file.\n",
    "- **Is A (isa):** `derived.isa.tsv.gz` contains edges `node`isa/node2` where either `node1/P31/node2` or `node1/P279/node2`\n",
    "- **Closure of subclass of (P279star):** `derived.P279star.tsv.gz` contains edges `node1/P279star/node2` where `node2` is reachable from `node1` via zero or more hops using the `P279` property. Note that for example, `Q44/P279star/Q44`. An example when this file is useful is when you want to find all the instance of a class, including instances of subclasses of the given class.\n",
    "- **In/out degrees:** `metadata.out_degree.tsv.gz` contains the out degree of every node, and `metadata.in_degree.tsv.gz` contains the in degree of every node.\n",
    "- **Pagerank:** outputs page rank on the directed graph in `metadata.pagerank.directed.tsv.gz` and page rank of the directed graph in `metadata.pagerank.undirected.tsv.gz`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Batch Invocation\n",
    "Example batch command. The second argument is a notebook where the output will be stored. You can load it to see progress.\n",
    "\n",
    "```\n",
    "papermill Wikidata\\ Useful\\ Files.ipynb useful-files.out.ipynb \\\n",
    "-p claims_file /Volumes/GoogleDrive/Shared\\ drives/KGTK-public-graphs/wikidata-20200803-v4/all.tsv.gz \\\n",
    "-p label_file /Volumes/GoogleDrive/Shared\\ drives/KGTK-public-graphs/wikidata-20200803-v4/part.label.en.tsv.gz \\\n",
    "-p item_file /Volumes/GoogleDrive/Shared\\ drives/KGTK-public-graphs/wikidata-20200803-v4/part.wikibase-item.tsv.gz \\\n",
    "-p property_item_file = /Volumes/GoogleDrive/Shared\\ drives/KGTK-public-graphs/wikidata-20200803-v4/part.property.wikibase-item.tsv.gz \\\n",
    "-p output_path <local folder> \\\n",
    "-p output_folder useful_files_v4 \\\n",
    "-p temp_folder temp.useful_files_v4 \\\n",
    "-p delete_database no \n",
    "-p languages es,ru,zh-cn\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Setup environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: kgtk [options] command [ / command]*\n"
     ]
    }
   ],
   "source": [
    "!kgtk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "\n",
    "# Folder on local machine where to create the output and temporary folders\n",
    "output_path = \"/Users/grantxie/Downloads/NIH\"\n",
    "\n",
    "# The names of the output and temporary folders\n",
    "output_folder = \"useful_wikidata_files\"\n",
    "temp_folder = \"temp.useful_wikidata_files\"\n",
    "\n",
    "# The location of input files\n",
    "wiki_root_folder = \"/Users/grantxie/Downloads/NIH/\"\n",
    "claims_file = \"claims.tsv.gz\"\n",
    "label_file = \"labels.en.tsv.gz\"\n",
    "alias_file = \"aliases.en.tsv.gz\"\n",
    "description_file = \"descriptions.en.tsv.gz\"\n",
    "item_file = \"claims.wikibase-item.tsv.gz\"\n",
    "\n",
    "label_all = \"labels.tsv.gz\"\n",
    "alias_all = \"aliases.tsv.gz\"\n",
    "description_all = \"descriptions.tsv.gz\"\n",
    "\n",
    "# Location of the cache database for kypher\n",
    "cache_path = \"/Users/grantxie/Downloads/NIH/temp.useful_wikidata_files\"\n",
    "\n",
    "# Whether to delete the cache database\n",
    "delete_database = False\n",
    "\n",
    "# Whether to compute pagerank as it may not run on the laptop\n",
    "compute_pagerank = True\n",
    "languages = 'ru,es,zh-cn,de,it,nl,pl,fr,pt,sv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "languages = languages.split(',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import os\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import altair as alt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up environment and folders to store the files\n",
    "\n",
    "- `OUT` folder where the output files go\n",
    "- `TEMP` folder to keep temporary files , including the database\n",
    "- `kgtk` shortcut to invoke the kgtk software\n",
    "- `kypher` shortcut to invoke `kgtk query with the cache database\n",
    "- `CLAIMS` the `all.tsv` file of wikidata that contains all edges except label/alias/description\n",
    "- `LABELS` the file with the English labels\n",
    "- `ITEMS` the wikibase-item file (currently does not include node1 that are properties so for now we need the net file\n",
    "- `STORE` location of the cache file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "if cache_path:\n",
    "    os.environ['STORE'] = \"{}/wikidata.sqlite3.db\".format(cache_path)\n",
    "else:\n",
    "    os.environ['STORE'] = \"{}/{}/wikidata.sqlite3.db\".format(output_path, temp_folder)\n",
    "os.environ['OUT'] = \"{}/{}\".format(output_path, output_folder)\n",
    "os.environ['TEMP'] = \"{}/{}\".format(output_path, temp_folder)\n",
    "os.environ['kgtk'] = \"kgtk\"\n",
    "os.environ['kgtk'] = \"kgtk --debug\"\n",
    "os.environ['kypher'] = \"kgtk --debug query --graph-cache \" + os.environ['STORE']\n",
    "os.environ['CLAIMS'] = wiki_root_folder + claims_file\n",
    "os.environ['LABELS'] = wiki_root_folder + label_file\n",
    "os.environ['ALIASES'] = wiki_root_folder + alias_file\n",
    "os.environ['DESCRIPTIONS'] = wiki_root_folder + description_file\n",
    "os.environ['ITEMS'] = wiki_root_folder + item_file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Echo the variables to see if they are all set correctly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/Downloads/NIH/useful_wikidata_files\n",
      "/Users/grantxie/Downloads/NIH/temp.useful_wikidata_files\n",
      "kgtk --debug\n",
      "kgtk --debug query --graph-cache /Users/grantxie/Downloads/NIH/temp.useful_wikidata_files/wikidata.sqlite3.db\n",
      "/Users/grantxie/Downloads/NIH/claims.tsv.gz\n",
      "/Users/grantxie/Downloads/NIH/labels.en.tsv.gz\n",
      "/Users/grantxie/Downloads/NIH/aliases.en.tsv.gz\n",
      "/Users/grantxie/Downloads/NIH/labels.en.tsv.gz\n",
      "\n",
      "/Users/grantxie/Downloads/NIH/temp.useful_wikidata_files/wikidata.sqlite3.db\n"
     ]
    }
   ],
   "source": [
    "!echo $OUT\n",
    "!echo $TEMP\n",
    "!echo $kgtk\n",
    "!echo $kypher\n",
    "!echo $CLAIMS\n",
    "!echo $LABELS\n",
    "!echo $ALIASES\n",
    "!echo $LABELS\n",
    "!echo $DESCRIPTIONSa\n",
    "!echo $STORE\n",
    "!alias col=\"column -t -s $'\\t' \""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Go to the output directory and create the subfolders for the output files and the temporary files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/Downloads/NIH\n"
     ]
    }
   ],
   "source": [
    "cd $output_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p $OUT\n",
    "!mkdir -p $TEMP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clean up the output and temp folders before we start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !rm $OUT/*.tsv $OUT/*.tsv.gz\n",
    "# !rm $TEMP/*.tsv $TEMP/*.tsv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "if delete_database:\n",
    "    print(\"Deleteddatabase\") \n",
    "    !rm $STORE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 240088\n",
      "-rw-r--r--  1 grantxie  staff      9025 Jul 10 14:52 Untitled.ipynb\n",
      "-rw-r--r--@ 1 grantxie  staff   4125423 Jul  9 11:27 coinvestigators 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   4125423 Jul  9 11:56 coinvestigators 3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   4125423 Jul  9 11:58 coinvestigators 4.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   2511815 Jul  9 12:26 coinvestigators.compact 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   2511815 Jul  9 11:23 coinvestigators.compact.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff    780738 Jul  9 12:25 coinvestigators.compact.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff   4125423 Jul  9 11:24 coinvestigators.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff        63 Sep 23 08:21 coinvestigators.tsv.gz\n",
      "-rw-r--r--  1 grantxie  staff      4378 Jul  8 20:46 coinvestigators11.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  8 21:51 nih_investigators_for_tl 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  8 21:51 nih_investigators_for_tl 3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 20:45 nih_investigators_for_tl.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     10861 Jul  8 21:51 nih_investigators_for_tl.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff     38766 Jul  9 11:22 nih_investigators_for_tl3 10.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 20:48 nih_investigators_for_tl3 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  8 20:58 nih_investigators_for_tl3 3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  8 21:09 nih_investigators_for_tl3 4.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 21:20 nih_investigators_for_tl3 5.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 21:25 nih_investigators_for_tl3 6.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 21:26 nih_investigators_for_tl3 7.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff  32600976 Jul  9 11:21 nih_investigators_for_tl3 8.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff  32600976 Jul  9 11:21 nih_investigators_for_tl3 9.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 20:45 nih_investigators_for_tl3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff      9270 Jul  9 11:22 nih_investigators_for_tl3.tsv.gz\n",
      "-rw-r--r--  1 grantxie  staff      7660 Jul  8 21:07 nih_investigators_for_tl3_sort.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff    109758 Jul  8 21:28 nih_investigators_for_tl4 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff    115718 Jul  8 21:37 nih_investigators_for_tl4 3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     29143 Jul  8 21:42 nih_investigators_for_tl4 4.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   5875902 Jul  8 21:47 nih_investigators_for_tl4 5.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  8 21:27 nih_investigators_for_tl4.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff        55 Jul  8 21:48 nih_investigators_for_tl4.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  9 08:48 nih_investigators_for_tl_2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     12121 Jul  8 20:48 nih_investigators_for_tl_2.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 21:55 nih_investigators_for_tl_new 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  9 08:48 nih_investigators_for_tl_new 3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  9 12:02 nih_investigators_for_tl_new 4.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     27653 Jul  8 21:52 nih_investigators_for_tl_new.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff      6931 Jul  8 21:51 nih_investigators_for_tl_new.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff     38766 Jul  9 12:02 nih_investigators_for_tl_no_error.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff      9278 Jul  9 12:02 nih_investigators_for_tl_no_error.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff     38766 Jul  9 12:28 nih_investigators_for_tl_no_error_1225 2.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     38766 Jul  9 12:29 nih_investigators_for_tl_no_error_1225 3.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   6459815 Jul  9 12:32 nih_investigators_for_tl_no_error_1225 4.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   6459815 Jul  9 12:36 nih_investigators_for_tl_no_error_1225 5.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   6459815 Jul  9 13:03 nih_investigators_for_tl_no_error_1225 6.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     38766 Jul  9 12:26 nih_investigators_for_tl_no_error_1225.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff   1587292 Jul  9 12:31 nih_investigators_for_tl_no_error_1225.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff   6459815 Jul  9 16:40 nih_investigators_for_tl_no_error_1225_7.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     92690 Jul  8 21:53 nih_investigators_for_tl_original.tsv\n",
      "-rw-r--r--@ 1 grantxie  staff     12128 Jul  8 21:52 nih_investigators_for_tl_original.tsv.gz\n",
      "-rw-r--r--  1 grantxie  staff        33 Jul 13 12:00 org.tsv.gz\n",
      "-rw-r--r--  1 grantxie  staff        57 Sep 23 08:21 test_nih.tsv.gz\n",
      "wikidata.sqlite3.db\n",
      "-rw-r--r--@ 1 grantxie  staff  3827281 Jul 26 09:40 /Users/grantxie/Downloads/NIH/claims.tsv.gz\n",
      "-rw-r--r--@ 1 grantxie  staff  2247582427 Jul 26 16:13 /Users/grantxie/Downloads/NIH/labels.en.tsv.gz\n",
      "ls: /Users/grantxie/Downloads/NIH/aliases.en.tsv.gz: No such file or directory\n",
      "-rw-r--r--@ 1 grantxie  staff  2247582427 Jul 26 16:13 /Users/grantxie/Downloads/NIH/labels.en.tsv.gz\n",
      "ls: /Users/grantxie/Downloads/NIH/descriptions.en.tsv.gz: No such file or directory\n",
      "/Users/grantxie/Downloads/NIH/temp.useful_wikidata_files/wikidata.sqlite3.db\n"
     ]
    }
   ],
   "source": [
    "!ls -l $OUT\n",
    "!ls $TEMP\n",
    "!ls -l \"$CLAIMS\"\n",
    "!ls -l \"$LABELS\"\n",
    "!ls -l \"$ALIASES\"\n",
    "!ls -l \"$LABELS\"\n",
    "!ls -l \"$DESCRIPTIONS\"\n",
    "!ls $STORE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate and load KGTK NIH Reporter files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-07-08 20:43:41 sqlstore]: DROP graph data table graph_7 from person\n",
      "[2021-07-08 20:43:41 sqlstore]: IMPORT graph directly into table graph_14 from /Users/grantxie/Downloads/NIH/nih_person_2021_002.id.tsv ...\n",
      "[2021-07-08 20:43:41 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_14 AS graph_14_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPER1860172-label-60f77b\tQNIHPER1860172\tlabel\t\"BOUCHER, RICHARD CHARLES\"\n",
      "QNIHPER1860172-PNIHID-894faa\tQNIHPER1860172\tPNIHID\t\"1860172\"\n",
      "QNIHPER1860172-P108-QNIHORG578206\tQNIHPER1860172\tP108\tQNIHORG578206\n",
      "QNIHORG578206-PNIHID-a061ee\tQNIHORG578206\tPNIHID\t578206\n",
      "QNIHPER9569267-label-f089d6\tQNIHPER9569267\tlabel\t\"CULLY, JEFFREY ALEXANDER\"\n",
      "QNIHPER9569267-PNIHID-718775\tQNIHPER9569267\tPNIHID\t\"9569267\"\n",
      "QNIHPER9569267-P108-QNIHORG481087\tQNIHPER9569267\tP108\tQNIHORG481087\n",
      "QNIHORG481087-PNIHID-809d11\tQNIHORG481087\tPNIHID\t481087\n",
      "QNIHPER10426588-label-dc7443\tQNIHPER10426588\tlabel\t\"FLETCHER, TERRI LYNN\"\n",
      "QNIHPER10426588-PNIHID-fef028\tQNIHPER10426588\tPNIHID\t\"10426588\"\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"nih_person_2021_002.id.tsv\" --as person \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-07-08 20:43:44 sqlstore]: DROP graph data table graph_8 from investigator\n",
      "[2021-07-08 20:43:44 sqlstore]: IMPORT graph directly into table graph_15 from /Users/grantxie/Downloads/NIH/nih_investigator_2021_002.id.tsv ...\n",
      "[2021-07-08 20:43:44 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_15 AS graph_15_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPRO9983961-P8329-QNIHPER1860172\tQNIHPRO9983961\tP8329\tQNIHPER1860172\n",
      "QNIHPRO9950743-P8329-QNIHPER9569267\tQNIHPRO9950743\tP8329\tQNIHPER9569267\n",
      "QNIHPRO9950743-P8329-QNIHPER10426588\tQNIHPRO9950743\tP8329\tQNIHPER10426588\n",
      "QNIHPRO10121452-P8329-QNIHPER6081502\tQNIHPRO10121452\tP8329\tQNIHPER6081502\n",
      "QNIHPRO10121452-P8329-QNIHPER16339995\tQNIHPRO10121452\tP8329\tQNIHPER16339995\n",
      "QNIHPRO9993026-P8329-QNIHPER15644446\tQNIHPRO9993026\tP8329\tQNIHPER15644446\n",
      "QNIHPRO10069157-P8329-QNIHPER6690859\tQNIHPRO10069157\tP8329\tQNIHPER6690859\n",
      "QNIHPRO10140852-P8329-QNIHPER15644249\tQNIHPRO10140852\tP8329\tQNIHPER15644249\n",
      "QNIHPRO9993926-P8329-QNIHPER9097221\tQNIHPRO9993926\tP8329\tQNIHPER9097221\n",
      "QNIHPRO10123330-P8329-QNIHPER16386398\tQNIHPRO10123330\tP8329\tQNIHPER16386398\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"nih_investigator_2021_002.id.tsv\" --as investigator \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-07-08 20:43:47 sqlstore]: DROP graph data table graph_10 from organization\n",
      "[2021-07-08 20:43:47 sqlstore]: IMPORT graph directly into table graph_10 from /Users/grantxie/Downloads/NIH/nih_organization_2021_002.id.tsv ...\n",
      "[2021-07-08 20:43:47 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_10 AS graph_10_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHORG578206-label-49dca4\tQNIHORG578206\tlabel\t\"UNIV OF NORTH CAROLINA CHAPEL HILL\"\n",
      "QNIHORG578206-P159string-3b31cc\tQNIHORG578206\tP159string\t\"CHAPEL HILL\"\n",
      "QNIHORG578206-PNIHstate-ad9d45\tQNIHORG578206\tPNIHstate\t\"NC\"\n",
      "QNIHORG578206-P17string-aeae66\tQNIHORG578206\tP17string\t\"UNITED STATES\"\n",
      "QNIHORG481087-label-681560\tQNIHORG481087\tlabel\t\"MICHAEL E DEBAKEY VA MEDICAL CENTER\"\n",
      "QNIHORG481087-P159string-db2679\tQNIHORG481087\tP159string\t\"HOUSTON\"\n",
      "QNIHORG481087-PNIHstate-58dd50\tQNIHORG481087\tPNIHstate\t\"TX\"\n",
      "QNIHORG481087-P17string-aeae66\tQNIHORG481087\tP17string\t\"UNITED STATES\"\n",
      "QNIHORG8246001-label-69ae51\tQNIHORG8246001\tlabel\t\"TENNESSEE STATE DEPARTMENT OF HEALTH\"\n",
      "QNIHORG8246001-P159string-d87ead\tQNIHORG8246001\tP159string\t\"NASHVILLE\"\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"nih_organization_2021_002.id.tsv\" --as organization \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup coinvestigators and output tl files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:24:06 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT DISTINCT graph_26_c1.\"node2\" \"_aLias.node1\", ? \"_aLias.label\", graph_58_c3.\"node2\" \"_aLias.node2\"\n",
      "     FROM graph_26 AS graph_26_c1, graph_26 AS graph_26_c2, graph_58 AS graph_58_c3\n",
      "     WHERE graph_58_c3.\"label\"=?\n",
      "     AND graph_26_c1.\"node1\"=graph_26_c2.\"node1\"\n",
      "     AND graph_26_c2.\"node2\"=graph_58_c3.\"node1\"\n",
      "     AND (graph_26_c1.\"node2\" != graph_26_c2.\"node2\")\n",
      "     ORDER BY graph_26_c1.\"node2\" ASC\n",
      "  PARAS: ['Pcoinvestigator_names', 'label']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i investigator -i person \\\n",
    "--match '\\\n",
    "    investigator: (project)-[]->(person1), \\\n",
    "    investigator: (project)-[]->(person2), \\\n",
    "    person: (person2)-[:label]->(person2_name)' \\\n",
    "--where 'person1 != person2' \\\n",
    "--return 'distinct person1 as node1, \"Pcoinvestigator_names\" as label, person2_name as `node2`' \\\n",
    "--order-by 'person1' \\\n",
    "-o \"$OUT\"/coinvestigators.tsv.gz\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kgtk sort --columns node1 -i \"$OUT\"/coinvestigators.tsv.gz \\\n",
    "/ compact --mode=NONE --columns node1 label --presorted True \\\n",
    "-o \"$OUT\"/coinvestigators.compact.tsv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:24:22 sqlstore]: DROP graph data table graph_11 from /Users/grantxie/Downloads/NIH/useful_wikidata_files/coinvestigators.compact.tsv.gz\n",
      "[2021-11-14 00:24:22 sqlstore]: IMPORT graph directly into table graph_60 from /Users/grantxie/Downloads/NIH/useful_wikidata_files/coinvestigators.compact.tsv.gz ...\n",
      "[2021-11-14 00:24:22 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT graph_60_c1.\"node1\" \"_aLias.person\", graph_58_c3.\"node2\" \"_aLias.person_name\", graph_25_c4.\"node1\" \"_aLias.organization\", graph_25_c4.\"node2\" \"_aLias.organization_name\", graph_60_c1.\"node2\" \"_aLias.coinvestigator_names\"\n",
      "     FROM graph_25 AS graph_25_c4, graph_58 AS graph_58_c2, graph_58 AS graph_58_c3, graph_60 AS graph_60_c1\n",
      "     WHERE graph_58_c2.\"label\"=?\n",
      "     AND graph_58_c3.\"label\"=?\n",
      "     AND graph_25_c4.\"node1\"=graph_58_c2.\"node2\"\n",
      "     AND graph_58_c2.\"node1\"=graph_58_c3.\"node1\"\n",
      "     AND graph_58_c2.\"node1\"=graph_60_c1.\"node1\"\n",
      "  PARAS: ['P108', 'label']\n",
      "---------------------------------------------\n",
      "[2021-11-14 00:24:22 sqlstore]: CREATE INDEX on table graph_60 column node1 ...\n",
      "[2021-11-14 00:24:22 sqlstore]: ANALYZE INDEX on table graph_60 column node1 ...\n",
      "[2021-11-14 00:24:22 sqlstore]: CREATE INDEX on table graph_58 column node2 ...\n",
      "[2021-11-14 00:24:22 sqlstore]: ANALYZE INDEX on table graph_58 column node2 ...\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i \"$OUT\"/coinvestigators.compact.tsv.gz -i organization -i person \\\n",
    "--match ' \\\n",
    "    coinvestigators: (person)-[]->(coinvestigator_names), \\\n",
    "    person: (person)-[:P108]->(organization), \\\n",
    "    person: (person)-[:label]->(person_name), \\\n",
    "    organization: (organization)-[]->(organization_name)' \\\n",
    "--return ' \\\n",
    "    person as person, \\\n",
    "    person_name as person_name, \\\n",
    "    organization as organization, \\\n",
    "    organization_name as organization_name, \\\n",
    "coinvestigator_names as coinvestigator_names' \\\n",
    "-o \"$OUT\"/nih_investigators_for_tl_original.tsv.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:25:14 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT graph_13_c2.\"node1\" \"_aLias.person\", graph_58_c4.\"node2\" \"_aLias.person_name\", graph_58_c3.\"node2\" \"_aLias.organization\", graph_25_c1.\"node2\" \"_aLias.organization_name\", graph_13_c2.\"node2\" \"_aLias.coinvestigator_names\"\n",
      "     FROM graph_13 AS graph_13_c2, graph_25 AS graph_25_c1, graph_58 AS graph_58_c3, graph_58 AS graph_58_c4\n",
      "     WHERE graph_25_c1.\"label\"=?\n",
      "     AND graph_58_c3.\"label\"=?\n",
      "     AND graph_58_c4.\"label\"=?\n",
      "     AND graph_13_c2.\"node1\"=graph_58_c3.\"node1\"\n",
      "     AND graph_25_c1.\"node1\"=graph_58_c3.\"node2\"\n",
      "     AND graph_58_c3.\"node1\"=graph_58_c4.\"node1\"\n",
      "  PARAS: ['label', 'P108', 'label']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i \"coinvestigators.compact.tsv\" -i organization -i person \\\n",
    "--match ' \\\n",
    "    organization: (organization)-[:label]->(organization_name),\\\n",
    "    coinvestigators: (person)-[]->(coinvestigator_names), \\\n",
    "    person: (person)-[:P108]->(organization), \\\n",
    "    person: (person)-[:label]->(person_name)'\\\n",
    "--return ' \\\n",
    "    person as person, \\\n",
    "    person_name as person_name, \\\n",
    "    organization as organization, \\\n",
    "    organization_name as organization_name, \\\n",
    "    coinvestigator_names as coinvestigator_names' \\\n",
    "-o \"$OUT\"/nih_investigators_for_tl3.tsv.gz\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-16 20:14:08 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_17 AS graph_17_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\tnode2;wikidatatype\n",
      "P10-P1628-32b85d-7927ece6-0\tP10\tP1628\t\"http://www.w3.org/2006/vcard/ns#Video\"\t\n",
      "P10-P1628-acf60d-b8950832-0\tP10\tP1628\t\"https://schema.org/video\"\t\n",
      "P10-P1629-Q34508-bcc39400-0\tP10\tP1629\tQ34508\t\n",
      "P10-P1659-P1651-c4068028-0\tP10\tP1659\tP1651\t\n",
      "P10-P1659-P18-5e4b9c4f-0\tP10\tP1659\tP18\t\n",
      "P10-P1659-P4238-d21d1ac0-0\tP10\tP1659\tP4238\t\n",
      "P10-P1659-P51-86aca4c5-0\tP10\tP1659\tP51\t\n",
      "P10-P1855-Q15075950-7eff6d65-0\tP10\tP1855\tQ15075950\twikibase-item\n",
      "P10-P1855-Q69063653-c8cdb04c-0\tP10\tP1855\tQ69063653\twikibase-item\n",
      "P10-P1855-Q7378-555592a4-0\tP10\tP1855\tQ7378\twikibase-item\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"$CLAIMS\" --as claim \\\n",
    "--limit 10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-16 20:14:13 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_18 AS graph_18_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "P10-label-en\tP10\tlabel\t'video'@en\n",
      "P1000-label-en\tP1000\tlabel\t'record held'@en\n",
      "P1001-label-en\tP1001\tlabel\t'applies to jurisdiction'@en\n",
      "P1002-label-en\tP1002\tlabel\t'engine configuration'@en\n",
      "P1003-label-en\tP1003\tlabel\t'National Library of Romania ID'@en\n",
      "P1004-label-en\tP1004\tlabel\t'MusicBrainz place ID'@en\n",
      "P1005-label-en\tP1005\tlabel\t'Portuguese National Library ID'@en\n",
      "P1006-label-en\tP1006\tlabel\t'Nationale Thesaurus voor Auteurs ID'@en\n",
      "P1007-label-en\tP1007\tlabel\t'Lattes Platform number'@en\n",
      "P101-label-en\tP101\tlabel\t'field of work'@en\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"$LABELS\" --as label \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-16 20:14:16 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_21 AS graph_21_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trank\tnode2;wikidatatype\n",
      "P10-P1629-Q34508-bcc39400-0\tP10\tP1629\tQ34508\tnormal\twikibase-item\n",
      "P10-P1855-Q15075950-7eff6d65-0\tP10\tP1855\tQ15075950\tnormal\twikibase-item\n",
      "P10-P1855-Q4504-a69d2c73-0\tP10\tP1855\tQ4504\tnormal\twikibase-item\n",
      "P10-P1855-Q69063653-c8cdb04c-0\tP10\tP1855\tQ69063653\tnormal\twikibase-item\n",
      "P10-P1855-Q7378-555592a4-0\tP10\tP1855\tQ7378\tnormal\twikibase-item\n",
      "P10-P2302-Q21502404-d012aef4-0\tP10\tP2302\tQ21502404\tnormal\twikibase-item\n",
      "P10-P2302-Q21510851-5224fe0b-0\tP10\tP2302\tQ21510851\tnormal\twikibase-item\n",
      "P10-P2302-Q21510852-dde2f0ce-0\tP10\tP2302\tQ21510852\tnormal\twikibase-item\n",
      "P10-P2302-Q52004125-d0288d06-0\tP10\tP2302\tQ52004125\tnormal\twikibase-item\n",
      "P10-P2302-Q53869507-974ce3b1-0\tP10\tP2302\tQ53869507\tnormal\twikibase-item\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"$ITEMS\" --as item \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-09-07 01:21:31 sqlstore]: IMPORT graph directly into table graph_52 from /Users/grantxie/Downloads/claims.external-id.tsv.gz ...\n",
      "[2021-09-07 01:38:39 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_52 AS graph_52_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trank\tnode2;wikidatatype\n",
      "P1005-P2264-155572-b048c795-0\tP1005\tP2264\t\"4031\"\tnormal\texternal-id\n",
      "P1005-P2264-3ae570-ebd28455-0\tP1005\tP2264\t\"4032\"\tnormal\texternal-id\n",
      "P1014-P2264-a23718-8fffd32b-0\tP1014\tP2264\t\"48\"\tnormal\texternal-id\n",
      "P1015-P2264-134fad-a7e86696-0\tP1015\tP2264\t\"1617\"\tnormal\texternal-id\n",
      "P1015-P2264-7e9c1a-b2f829b5-0\tP1015\tP2264\t\"564\"\tnormal\texternal-id\n",
      "P1021-P2264-340fcd-59428d77-0\tP1021\tP2264\t\"88\"\tnormal\texternal-id\n",
      "P1043-P2264-797922-58a7ad5f-0\tP1043\tP2264\t\"248\"\tnormal\texternal-id\n",
      "P1045-P2264-d509ce-e51b9649-0\tP1045\tP2264\t\"46\"\tnormal\texternal-id\n",
      "P1047-P2264-63c077-2c42ccf1-0\tP1047\tP2264\t\"962\"\tnormal\texternal-id\n",
      "P1047-P2264-c393fd-099f19d6-0\tP1047\tP2264\t\"43\"\tnormal\texternal-id\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/claims.external-id.tsv.gz\" --as exid\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Output files for Tableau"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:26:54 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_21 AS graph_21_c1\n",
      "     WHERE graph_21_c1.\"label\"=?\n",
      "     AND graph_21_c1.\"node2\"=?\n",
      "  PARAS: ['P31', 'Q16917']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i author \\\n",
    "--match 'item: (c)-[:P31]->(:Q16917)' \\\n",
    "-o hospital.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:27:06 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_22 AS graph_22_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trank\tnode2;wikidatatype\n",
      "Q100-P625-2e35d9-cdfcaf0e-0\tQ100\tP625\t@42.358333333333/-71.0625\tnormal\tglobe-coordinate\n",
      "Q1000-P1332-b48e3a-0ed57897-0\tQ1000\tP1332\t@2.32/11.7\tnormal\tglobe-coordinate\n",
      "Q1000-P1333-3c3826-e820db87-0\tQ1000\tP1333\t@-3.96005/11.15322\tnormal\tglobe-coordinate\n",
      "Q1000-P1334-44067c-eebd06bd-0\tQ1000\tP1334\t@-0.61746/14.5266\tnormal\tglobe-coordinate\n",
      "Q1000-P1335-a8c32c-37c5bb67-0\tQ1000\tP1335\t@-0.62444444/8.70805556\tnormal\tglobe-coordinate\n",
      "Q1000-P625-fa52c4-ce6c2e1f-0\tQ1000\tP625\t@-0.68333055555556/11.5\tnormal\tglobe-coordinate\n",
      "Q100000-P625-58c345-14fbc9fe-0\tQ100000\tP625\t@50.8283/5.7678\tnormal\tglobe-coordinate\n",
      "Q100000001-P625-acd0bb-fb89d825-0\tQ100000001\tP625\t@-35.19629722222222/149.14484166666668\tnormal\tglobe-coordinate\n",
      "Q100000034-P625-e952d9-aea04e31-0\tQ100000034\tP625\t@43.84324/13.01446\tnormal\tglobe-coordinate\n",
      "Q100000050-P625-46b12b-d84c11b4-0\tQ100000050\tP625\t@37.1954/128.8702\tnormal\tglobe-coordinate\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/claims.globe-coordinate.tsv\" --as coor \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:27:22 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT graph_22_c2.\"node1\", graph_22_c2.\"node2\", graph_18_c3.\"node2\"\n",
      "     FROM graph_18 AS graph_18_c3, graph_21 AS graph_21_c1, graph_22 AS graph_22_c2\n",
      "     WHERE graph_21_c1.\"node2\"=?\n",
      "     AND graph_22_c2.\"label\"=?\n",
      "     AND graph_18_c3.\"node1\"=graph_21_c1.\"node1\"\n",
      "     AND graph_21_c1.\"node1\"=graph_22_c2.\"node1\"\n",
      "  PARAS: ['Q16917', 'P625']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i coor -i label\\\n",
    "--match 'item: (c)-[]->(:Q16917), coor:(c)-[:P625]->(d), label:(c)-[]->(e)'\\\n",
    "--return 'c, d, e' \\\n",
    "-o name1.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls = []\n",
    "for i in range(0,50):\n",
    "    ls.append('/Users/grantxie/Downloads/with_nils_author/split_' +str(i) +\".csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>column</th>\n",
       "      <th>row</th>\n",
       "      <th>label</th>\n",
       "      <th>context</th>\n",
       "      <th>filename</th>\n",
       "      <th>column-id</th>\n",
       "      <th>label_clean</th>\n",
       "      <th>kg_id</th>\n",
       "      <th>kg_labels</th>\n",
       "      <th>kg_aliases</th>\n",
       "      <th>...</th>\n",
       "      <th>lof_class_count_tf_idf_score</th>\n",
       "      <th>top5_class_count</th>\n",
       "      <th>lof_property_count_tf_idf_score</th>\n",
       "      <th>top5_property_count</th>\n",
       "      <th>context_property</th>\n",
       "      <th>context_similarity</th>\n",
       "      <th>context_property_similarity_q_node</th>\n",
       "      <th>context_score</th>\n",
       "      <th>siamese_prediction</th>\n",
       "      <th>rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LINDLEY BARBEE</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON|CHRISTINE MITRA KHOSR...</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv-0</td>\n",
       "      <td>LINDLEY BARBEE</td>\n",
       "      <td>NIL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>CHRISTINE MITRA KHOSROPOUR</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON|LINDLEY BARBEE</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv-0</td>\n",
       "      <td>CHRISTINE MITRA KHOSROPOUR</td>\n",
       "      <td>NIL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>MICHAEL J GALE</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON|CAROLYN B COYNE;HUGO ...</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv-0</td>\n",
       "      <td>MICHAEL J GALE</td>\n",
       "      <td>Q6831539</td>\n",
       "      <td>Michael Gale|Michael J Gale</td>\n",
       "      <td>M. J Gale|Michael J. Gale|M. J. Gale|Gale, M. J.</td>\n",
       "      <td>...</td>\n",
       "      <td>0.51241</td>\n",
       "      <td>Q36180:0.031|Q482980:0.030|Q2500638:0.030|Q702...</td>\n",
       "      <td>0.111416</td>\n",
       "      <td>P106:0.034|P735:0.028|P21:0.023|P27:0.009|P646...</td>\n",
       "      <td>P108|Pcoauthor</td>\n",
       "      <td>0.4506$$|0.45$$</td>\n",
       "      <td>P108/0.4506$$|Pcoauthor/0.45$$</td>\n",
       "      <td>0.3632</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>DANIEL M. RATNER</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON|PATRICK S. STAYTON;SH...</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv-0</td>\n",
       "      <td>DANIEL M. RATNER</td>\n",
       "      <td>NIL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>SHAWN J. SKERRETT</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON|COURTNEY CRANE;DANIEL...</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv</td>\n",
       "      <td>tl_person_coinvestigator_with_qnihid.tsv-0</td>\n",
       "      <td>SHAWN J. SKERRETT</td>\n",
       "      <td>NIL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   column  row                       label  \\\n",
       "0       0    0              LINDLEY BARBEE   \n",
       "1       0    1  CHRISTINE MITRA KHOSROPOUR   \n",
       "2       0    2              MICHAEL J GALE   \n",
       "3       0    3            DANIEL M. RATNER   \n",
       "4       0    4           SHAWN J. SKERRETT   \n",
       "\n",
       "                                             context  \\\n",
       "0  UNIVERSITY OF WASHINGTON|CHRISTINE MITRA KHOSR...   \n",
       "1            UNIVERSITY OF WASHINGTON|LINDLEY BARBEE   \n",
       "2  UNIVERSITY OF WASHINGTON|CAROLYN B COYNE;HUGO ...   \n",
       "3  UNIVERSITY OF WASHINGTON|PATRICK S. STAYTON;SH...   \n",
       "4  UNIVERSITY OF WASHINGTON|COURTNEY CRANE;DANIEL...   \n",
       "\n",
       "                                   filename  \\\n",
       "0  tl_person_coinvestigator_with_qnihid.tsv   \n",
       "1  tl_person_coinvestigator_with_qnihid.tsv   \n",
       "2  tl_person_coinvestigator_with_qnihid.tsv   \n",
       "3  tl_person_coinvestigator_with_qnihid.tsv   \n",
       "4  tl_person_coinvestigator_with_qnihid.tsv   \n",
       "\n",
       "                                    column-id                 label_clean  \\\n",
       "0  tl_person_coinvestigator_with_qnihid.tsv-0              LINDLEY BARBEE   \n",
       "1  tl_person_coinvestigator_with_qnihid.tsv-0  CHRISTINE MITRA KHOSROPOUR   \n",
       "2  tl_person_coinvestigator_with_qnihid.tsv-0              MICHAEL J GALE   \n",
       "3  tl_person_coinvestigator_with_qnihid.tsv-0            DANIEL M. RATNER   \n",
       "4  tl_person_coinvestigator_with_qnihid.tsv-0           SHAWN J. SKERRETT   \n",
       "\n",
       "      kg_id                    kg_labels  \\\n",
       "0       NIL                          NaN   \n",
       "1       NIL                          NaN   \n",
       "2  Q6831539  Michael Gale|Michael J Gale   \n",
       "3       NIL                          NaN   \n",
       "4       NIL                          NaN   \n",
       "\n",
       "                                          kg_aliases  ...  \\\n",
       "0                                                NaN  ...   \n",
       "1                                                NaN  ...   \n",
       "2  M. J Gale|Michael J. Gale|M. J. Gale|Gale, M. J.   ...   \n",
       "3                                                NaN  ...   \n",
       "4                                                NaN  ...   \n",
       "\n",
       "  lof_class_count_tf_idf_score  \\\n",
       "0                          NaN   \n",
       "1                          NaN   \n",
       "2                      0.51241   \n",
       "3                          NaN   \n",
       "4                          NaN   \n",
       "\n",
       "                                    top5_class_count  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2  Q36180:0.031|Q482980:0.030|Q2500638:0.030|Q702...   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "   lof_property_count_tf_idf_score  \\\n",
       "0                              NaN   \n",
       "1                              NaN   \n",
       "2                         0.111416   \n",
       "3                              NaN   \n",
       "4                              NaN   \n",
       "\n",
       "                                 top5_property_count  context_property  \\\n",
       "0                                                NaN               NaN   \n",
       "1                                                NaN               NaN   \n",
       "2  P106:0.034|P735:0.028|P21:0.023|P27:0.009|P646...    P108|Pcoauthor   \n",
       "3                                                NaN               NaN   \n",
       "4                                                NaN               NaN   \n",
       "\n",
       "   context_similarity  context_property_similarity_q_node  context_score  \\\n",
       "0                 NaN                                 NaN         0.0000   \n",
       "1                 NaN                                 NaN         0.0000   \n",
       "2     0.4506$$|0.45$$      P108/0.4506$$|Pcoauthor/0.45$$         0.3632   \n",
       "3                 NaN                                 NaN         0.0000   \n",
       "4                 NaN                                 NaN         0.0000   \n",
       "\n",
       "   siamese_prediction  rank  \n",
       "0                 0.0   NaN  \n",
       "1                 0.0   NaN  \n",
       "2                 1.0   1.0  \n",
       "3                 0.0   NaN  \n",
       "4                 0.0   NaN  \n",
       "\n",
       "[5 rows x 39 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/Users/grantxie/Downloads/with_nils_author/split_0.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "original = pd.read_csv('/Users/grantxie/Downloads/tl_person_coinvestigator_with_qnihid.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>person_name</th>\n",
       "      <th>organization_name</th>\n",
       "      <th>coinvestigator_names</th>\n",
       "      <th>person_qnihid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINDLEY BARBEE</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON</td>\n",
       "      <td>CHRISTINE MITRA KHOSROPOUR</td>\n",
       "      <td>QNIHPER11081386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHRISTINE MITRA KHOSROPOUR</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON</td>\n",
       "      <td>LINDLEY BARBEE</td>\n",
       "      <td>QNIHPER12074317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MICHAEL J GALE</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON</td>\n",
       "      <td>CAROLYN B COYNE;HUGO RAMON ROSEN;JOAN M GOVERM...</td>\n",
       "      <td>QNIHPER1940486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DANIEL M. RATNER</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON</td>\n",
       "      <td>PATRICK S. STAYTON;SHAWN J. SKERRETT;TIMOTHY E...</td>\n",
       "      <td>QNIHPER6623689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SHAWN J. SKERRETT</td>\n",
       "      <td>UNIVERSITY OF WASHINGTON</td>\n",
       "      <td>COURTNEY CRANE;DANIEL M. RATNER;KATHIE ANNE WA...</td>\n",
       "      <td>QNIHPER1898012</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  person_name         organization_name  \\\n",
       "0              LINDLEY BARBEE  UNIVERSITY OF WASHINGTON   \n",
       "1  CHRISTINE MITRA KHOSROPOUR  UNIVERSITY OF WASHINGTON   \n",
       "2              MICHAEL J GALE  UNIVERSITY OF WASHINGTON   \n",
       "3            DANIEL M. RATNER  UNIVERSITY OF WASHINGTON   \n",
       "4           SHAWN J. SKERRETT  UNIVERSITY OF WASHINGTON   \n",
       "\n",
       "                                coinvestigator_names    person_qnihid  \n",
       "0                         CHRISTINE MITRA KHOSROPOUR  QNIHPER11081386  \n",
       "1                                     LINDLEY BARBEE  QNIHPER12074317  \n",
       "2  CAROLYN B COYNE;HUGO RAMON ROSEN;JOAN M GOVERM...   QNIHPER1940486  \n",
       "3  PATRICK S. STAYTON;SHAWN J. SKERRETT;TIMOTHY E...   QNIHPER6623689  \n",
       "4  COURTNEY CRANE;DANIEL M. RATNER;KATHIE ANNE WA...   QNIHPER1898012  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "original.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/Downloads/with_nils_author/split_0.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_1.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_2.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_3.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_4.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_5.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_6.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_7.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_8.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_9.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_10.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_11.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_12.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_13.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_14.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_15.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_16.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_17.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_18.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_19.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_20.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_21.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_22.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_23.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_24.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_25.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_26.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_27.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_28.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_29.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_30.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_31.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_32.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_33.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_34.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_35.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_36.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_37.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_38.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_39.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_40.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_41.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_42.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_43.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_44.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_45.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_46.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_47.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_48.csv\n",
      "/Users/grantxie/Downloads/with_nils_author/split_49.csv\n"
     ]
    }
   ],
   "source": [
    "arr = []\n",
    "id_t = []\n",
    "qid = []\n",
    "label = []\n",
    "kg = []\n",
    "mem = []\n",
    "graph = []\n",
    "for file in ls:\n",
    "    print(file)\n",
    "    df = pd.read_csv(file)\n",
    "    for i in range(0, len(df)):\n",
    "        if df['row'][i] in arr:\n",
    "            continue\n",
    "        arr.append(df['row'][i])\n",
    "        label.append(df.label[i])\n",
    "        kg.append(df['kg_id'][i])\n",
    "        id_t.append('')\n",
    "        mem.append('label')\n",
    "        graph.append('author')\n",
    "        qid.append(original['person_qnihid'][int(df['row'][i])])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "new = pd.DataFrame(list(zip(id_t,qid,mem,kg,arr, label,graph)), columns = ['id', 'node1', 'label', 'node2', 'row', 'node2;label','graph'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>row</th>\n",
       "      <th>node2;label</th>\n",
       "      <th>graph</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER11081386</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>0</td>\n",
       "      <td>LINDLEY BARBEE</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER12074317</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>1</td>\n",
       "      <td>CHRISTINE MITRA KHOSROPOUR</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER1940486</td>\n",
       "      <td>label</td>\n",
       "      <td>Q6831539</td>\n",
       "      <td>2</td>\n",
       "      <td>MICHAEL J GALE</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER6623689</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>3</td>\n",
       "      <td>DANIEL M. RATNER</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER1898012</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>4</td>\n",
       "      <td>SHAWN J. SKERRETT</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51563</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER8107460</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>51564</td>\n",
       "      <td>RAINA N. FICHOROVA</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51564</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER2272576</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>51565</td>\n",
       "      <td>DOROTHEA DENISE JENKINS</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51565</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER16431128</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>51566</td>\n",
       "      <td>JACQUELINE GENOVESI</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51566</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER16505596</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>51567</td>\n",
       "      <td>JENNIFER JOVANOVIC</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51567</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER16505541</td>\n",
       "      <td>label</td>\n",
       "      <td>NIL</td>\n",
       "      <td>51568</td>\n",
       "      <td>TONI A. SONDERGELD</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51568 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id            node1  label     node2    row                 node2;label  \\\n",
       "0         QNIHPER11081386  label       NIL      0              LINDLEY BARBEE   \n",
       "1         QNIHPER12074317  label       NIL      1  CHRISTINE MITRA KHOSROPOUR   \n",
       "2          QNIHPER1940486  label  Q6831539      2              MICHAEL J GALE   \n",
       "3          QNIHPER6623689  label       NIL      3            DANIEL M. RATNER   \n",
       "4          QNIHPER1898012  label       NIL      4           SHAWN J. SKERRETT   \n",
       "...   ..              ...    ...       ...    ...                         ...   \n",
       "51563      QNIHPER8107460  label       NIL  51564          RAINA N. FICHOROVA   \n",
       "51564      QNIHPER2272576  label       NIL  51565     DOROTHEA DENISE JENKINS   \n",
       "51565     QNIHPER16431128  label       NIL  51566         JACQUELINE GENOVESI   \n",
       "51566     QNIHPER16505596  label       NIL  51567          JENNIFER JOVANOVIC   \n",
       "51567     QNIHPER16505541  label       NIL  51568          TONI A. SONDERGELD   \n",
       "\n",
       "        graph  \n",
       "0      author  \n",
       "1      author  \n",
       "2      author  \n",
       "3      author  \n",
       "4      author  \n",
       "...       ...  \n",
       "51563  author  \n",
       "51564  author  \n",
       "51565  author  \n",
       "51566  author  \n",
       "51567  author  \n",
       "\n",
       "[51568 rows x 7 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "new = new[new['node2'] != 'NIL']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "new = new.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>row</th>\n",
       "      <th>node2;label</th>\n",
       "      <th>graph</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER1940486</td>\n",
       "      <td>label</td>\n",
       "      <td>Q6831539</td>\n",
       "      <td>2</td>\n",
       "      <td>MICHAEL J GALE</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER2414340</td>\n",
       "      <td>label</td>\n",
       "      <td>Q87796696</td>\n",
       "      <td>5</td>\n",
       "      <td>PATRICK S. STAYTON</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER11537130</td>\n",
       "      <td>label</td>\n",
       "      <td>Q89958961</td>\n",
       "      <td>8</td>\n",
       "      <td>KENNETH NGURE</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER8820221</td>\n",
       "      <td>label</td>\n",
       "      <td>Q61467477</td>\n",
       "      <td>10</td>\n",
       "      <td>ANITHA PASUPATHY</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER8491201</td>\n",
       "      <td>label</td>\n",
       "      <td>Q46001993</td>\n",
       "      <td>11</td>\n",
       "      <td>STEPHANIE M FULLERTON</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21905</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER1865473</td>\n",
       "      <td>label</td>\n",
       "      <td>Q88874640</td>\n",
       "      <td>51553</td>\n",
       "      <td>THERESA L WALUNAS</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21906</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER77800199</td>\n",
       "      <td>label</td>\n",
       "      <td>Q17232943</td>\n",
       "      <td>51555</td>\n",
       "      <td>HEATHER RYAN</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21907</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER16138148</td>\n",
       "      <td>label</td>\n",
       "      <td>Q59751913</td>\n",
       "      <td>51557</td>\n",
       "      <td>TAD SIMONS</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21908</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER6983267</td>\n",
       "      <td>label</td>\n",
       "      <td>Q88663895</td>\n",
       "      <td>51558</td>\n",
       "      <td>VERA P KRYMSKAYA</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21909</th>\n",
       "      <td></td>\n",
       "      <td>QNIHPER1889098</td>\n",
       "      <td>label</td>\n",
       "      <td>Q91381841</td>\n",
       "      <td>51562</td>\n",
       "      <td>SCOTT C STEFFENSEN</td>\n",
       "      <td>author</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21910 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id            node1  label      node2    row            node2;label  \\\n",
       "0          QNIHPER1940486  label   Q6831539      2         MICHAEL J GALE   \n",
       "1          QNIHPER2414340  label  Q87796696      5     PATRICK S. STAYTON   \n",
       "2         QNIHPER11537130  label  Q89958961      8          KENNETH NGURE   \n",
       "3          QNIHPER8820221  label  Q61467477     10       ANITHA PASUPATHY   \n",
       "4          QNIHPER8491201  label  Q46001993     11  STEPHANIE M FULLERTON   \n",
       "...   ..              ...    ...        ...    ...                    ...   \n",
       "21905      QNIHPER1865473  label  Q88874640  51553      THERESA L WALUNAS   \n",
       "21906     QNIHPER77800199  label  Q17232943  51555           HEATHER RYAN   \n",
       "21907     QNIHPER16138148  label  Q59751913  51557             TAD SIMONS   \n",
       "21908      QNIHPER6983267  label  Q88663895  51558       VERA P KRYMSKAYA   \n",
       "21909      QNIHPER1889098  label  Q91381841  51562     SCOTT C STEFFENSEN   \n",
       "\n",
       "        graph  \n",
       "0      author  \n",
       "1      author  \n",
       "2      author  \n",
       "3      author  \n",
       "4      author  \n",
       "...       ...  \n",
       "21905  author  \n",
       "21906  author  \n",
       "21907  author  \n",
       "21908  author  \n",
       "21909  author  \n",
       "\n",
       "[21910 rows x 7 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(new)):\n",
    "    new['id'][i] = 'author' + str(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "new.to_csv('nih_author_kgtk.tsv', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "new.to_csv('/Users/grantxie/Downloads/nig_kgtk.tsv', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Output files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:31:07 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_23 AS graph_23_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trow\tgraph\n",
      "qnode0\tQ219563\tlabel\tUNIVERSITY OF WASHINGTON\t0\tqnode\n",
      "qnode1\tQ5081831\tlabel\tCHARLES R. DREW UNIVERSITY OF MEDICAL & SCIENCE\t1\tqnode\n",
      "qnode2\tQ622664\tlabel\tUNIVERSITY OF CALIFORNIA, SAN DIEGO\t2\tqnode\n",
      "qnode3\tQ812573\tlabel\tBAYLOR UNIVERSITY\t4\tqnode\n",
      "qnode4\tQ49118\tlabel\tBOSTON COLLEGE\t5\tqnode\n",
      "qnode5\tQ6806451\tlabel\tMEDICAL UNIVERSITY OF SOUTH CAROLINA\t6\tqnode\n",
      "qnode6\tQ457281\tlabel\tUNIVERSITY OF ILLINOIS AT URBANA-CHAMPAIGN\t7\tqnode\n",
      "qnode7\tQ168751\tlabel\tDUKE UNIVERSITY\t9\tqnode\n",
      "qnode8\tQ349055\tlabel\tWAYNE STATE UNIVERSITY\t10\tqnode\n",
      "qnode9\tQ104480607\tlabel\tNORTHAMPTON VA MEDICAL CENTER\t11\tqnode\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/kgtk-master/use-cases/nig_kgtk.tsv\" --as nih \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:31:15 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_31 AS graph_31_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [5]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPRO10286324-PNIHtitle-cfa7f1\tQNIHPRO10286324\tPNIHtitle\t\"CD38-TARGETED IMMUNOPET OF MYELOMA: PHASE 2 TRIAL OF CLINICAL APPLICATIONS\"\n",
      "QNIHPRO10286324-PNIHorg-17451a\tQNIHPRO10286324\tPNIHorg\t\"HOAG MEMORIAL HOSPITAL PRESBYTERIAN\"\n",
      "QNIHPRO10286324-PNIHcost-452920\tQNIHPRO10286324\tPNIHcost\t598926\n",
      "QNIHPRO10286324-PNIHsub-730793\tQNIHPRO10286324\tPNIHsub\t\"Antibodies; base; Biopsy; blind; Blood Tests; Bone marrow biopsy; burden of illness; cancer imaging; Cells; Clinical; clinical application; clinical care; Clinical Trials; clinically significant; Combined Modality Therapy; design; Detection; Detection of Minimal Residual Disease; Disease; dosimetry; early phase trial; experience; first-in-human; fluorodeoxyglucose positron emission tomography; Funding Opportunities; Future; Goals; Histologic; Human; human imaging; Image; imaging agent; imaging modality; Immunoglobulins; ImmunoPET; individual patient; innovation; Investigation; Laboratories; Measurement; Measures; Membrane Proteins; Memorial Sloan-Kettering Cancer Center; Methods; Monitor; Monoclonal Antibodies; Multiple Myeloma; multiple myeloma M Protein; novel; outcome forecast; Patient Care; Patient imaging; Patient Monitoring; Patients; PET/CT scan; Phase; Phase I Clinical Trials; phase I trial; Phase II Clinical Trials; phase II trial; Plasma Cell Neoplasm; Plasma Cells; Positron-Emission Tomography; preclinical imaging; predicting response; Prognostic Marker; Proteins; Radiochemistry; Radioimmunotherapy; Radiolabeled; Research Personnel; Residual Tumors; response; Safety; Selection for Treatments; Serum; Site; Surface; targeted imaging; theranostics; Therapeutic Agents; Tracer; Translations; treatment response; tumor; Tumor Burden; United States; uptake; virtual; Visualization; Work;\"\n",
      "QNIHPRO10258167-PNIHtitle-90b590\tQNIHPRO10258167\tPNIHtitle\t\"Mechanism of selective packaging of primer tRNALys3 by HIV-1\"\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/nih_projects/nih_project_new.id.tsv\" --as project \\\n",
    "--limit 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:31:21 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_32 AS graph_32_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [5]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPRO9983961-P8329-QNIHPER1860172\tQNIHPRO9983961\tP8329\tQNIHPER1860172\n",
      "QNIHPRO9950743-P8329-QNIHPER9569267\tQNIHPRO9950743\tP8329\tQNIHPER9569267\n",
      "QNIHPRO9950743-P8329-QNIHPER10426588\tQNIHPRO9950743\tP8329\tQNIHPER10426588\n",
      "QNIHPRO10121452-P8329-QNIHPER6081502\tQNIHPRO10121452\tP8329\tQNIHPER6081502\n",
      "QNIHPRO10121452-P8329-QNIHPER16339995\tQNIHPRO10121452\tP8329\tQNIHPER16339995\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/kgtk-files-nih-V3.0_new/nih_investigator.id.tsv\" --as inv \\\n",
    "--limit 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:31:25 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_33 AS graph_33_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [5]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPER1904041-label-1b2172\tQNIHPER1904041\tlabel\tROBERT STERLING TURNER\n",
      "QNIHPER1904041-PNIHID-475a43\tQNIHPER1904041\tPNIHID\t1904041\n",
      "QNIHPER1904041-P108-QNIHORG2059802\tQNIHPER1904041\tP108\tQNIHORG2059802\n",
      "QNIHORG2059802-PNIHID-d70604\tQNIHORG2059802\tPNIHID\t2059802\n",
      "QNIHPER6240796-label-d0a9cc\tQNIHPER6240796\tlabel\tRUTH OTTMAN\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/kgtk-files-nih-V3.0_new/nih_person.id.tsv\" --as  per\\\n",
    "--limit 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:31:28 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_27 AS graph_27_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [5]\n",
      "---------------------------------------------\n",
      "node1\tlabel\tnode2\n",
      "QNIHPER10001049\tPcoinvestigator_names\tBARBARA ANN CHURCH|EDUARDO MERCADO|JONATHAN D. RODGERS\n",
      "QNIHPER10001128\tPcoinvestigator_names\tMARK A. MC NIVEN\n",
      "QNIHPER10001819\tPcoinvestigator_names\tABRAHAM NA SHAKED|BRIAN DONALD PIENING|JUHI KUMAR|KRZYSZTOF KIRYLUK|MALEK KAMOUN|MARIO C. DENG|MICHAEL P. SNYDER|SANDRA AMARAL\n",
      "QNIHPER10003131\tPcoinvestigator_names\tAARON PAUL BATISTA|BYRON M. YU\n",
      "QNIHPER10003688\tPcoinvestigator_names\tBOB WONG|JANICE MARGARET MORSE\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"coinvestigators.compact.tsv\" --as coinvestigator \\\n",
    "--limit 5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:31:33 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_37 AS graph_37_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [5]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPER10003954-member-Q39050124\tQNIHPER10003954\tmember\tQ39050124\n",
      "QNIHPER10428590-member-Q625321\tQNIHPER10428590\tmember\tQ625321\n",
      "QNIHPER10004092-member-Q5582923\tQNIHPER10004092\tmember\tQ5582923\n",
      "QNIHPER12571233-member-Q28035413\tQNIHPER12571233\tmember\tQ28035413\n",
      "QNIHPER2098890-member-Q50036737\tQNIHPER2098890\tmember\tQ50036737\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"author_hospital.id.tsv\" --as hosp \\\n",
    "--limit 5\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: kgtk [options] command [ / command]*\n",
      "kgtk: error: unrecognized arguments: --optional project: (project)-[]->(cost)\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project\\\n",
    "--optional 'project: (project)-[]->(cost)' \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: kgtk [options] command [ / command]*\n",
      "kgtk: error: unrecognized arguments: --opt project: (project)-[:PNIHcost]->(cost) --opt project: (project)-[:PNIHtitle]->(project_title) --opt project: (project)-[:PNIHsub]->(project_subjects)\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project\\\n",
    "--match 'author: (cluster)-[]->(qnode), nih_author:(qnihauthor)-[]->(qnode), inv:(project)-[]->(qnihauthor), \\\n",
    "        item:(qnode)-[:P108]->(org_node), label:(org_node)-[]->(org_name),label:(qnode)-[]->(author_name), \\\n",
    "        coor:(org_node)-[:P625]->(organization_coor)'\\\n",
    "--opt 'project: (project)-[:PNIHcost]->(cost)' \\\n",
    "--opt 'project: (project)-[:PNIHtitle]->(project_title)' \\\n",
    "--opt 'project: (project)-[:PNIHsub]->(project_subjects)' \\\n",
    "--return 'cluster as cluster_id, author_name as author_name, \\\n",
    "            org_name as org_name, project_title as project_title, project_subjects as project_subjects,\\\n",
    "            kgtk_geo_coords_lat(organization_coor) as lat, kgtk_geo_coords_long(organization_coor) as long, sum(cost)' \\\n",
    "-o nih_summary.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-11-14 00:32:19 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_31 AS graph_31_c1\n",
      "     WHERE graph_31_c1.\"node1\"=?\n",
      "  PARAS: ['QNIHPRO10168561']\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPRO10168561-PNIHtitle-0acc1e\tQNIHPRO10168561\tPNIHtitle\t\"Mechano-molecular regulation of kinetochore function\"\n",
      "QNIHPRO10168561-PNIHorg-9ddda8\tQNIHPRO10168561\tPNIHorg\t\"UNIVERSITY OF MASSACHUSETTS AMHERST\"\n",
      "QNIHPRO10168561-PNIHcost-97c49b\tQNIHPRO10168561\tPNIHcost\t313116\n",
      "QNIHPRO10168561-PNIHsub-757264\tQNIHPRO10168561\tPNIHsub\t\"Adult; Affinity; Amino Acids; Anaphase; Aneuploidy; base; Binding; Biochemical; Biological Assay; Biophysics; cancer diagnosis; Cell division; Cell Division Process; Cell physiology; Cells; Centromere; Chromatin; chromosome missegregation; chromosome movement; chromosome number abnormality; Chromosomes; Complex; Congenital Abnormality; Disease; Drosophila genus; Elements; Embryonic Development; Ensure; experimental study; First Pregnancy Trimester; force sensor; Funding; Generations; Genome; Goals; Homologous Gene; Human; human tissue; improved; In Vitro; innovation; interest; Kinetochores; Knowledge; Lead; Link; Maintenance; Mechanics; mechanotransduction; Microtubules; Mitotic Chromosome; Modeling; Molecular; Molecular Conformation; molecular mass; Names; Nature; Neoplasm Metastasis; novel; Pathologic; pointed protein; Polymers; Population; Positioning Attribute; Prevention; Proliferation Marker; Property; protein complex; protein folding; Proteins; receptor; recruit; Regulation; Regulatory Pathway; Reproducibility; Research; Role; sensor; Signal Transduction; single molecule; Sister Chromatid; Spontaneous abortion; stem; Stretching; Structure; Surface; targeted treatment; Testing; Therapeutic; Time; tissue/cell culture; Tissues; Transducers; Translating; transmission process; tumorigenesis; Work;\"\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project\\\n",
    "--match 'project: (:QNIHPRO10168561)-[]->(qnode)'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: kgtk [options] command [ / command]*\n",
      "kgtk: error: unrecognized arguments: --opt time: (hosp_node)-[:P571]->(hosp_inc) --opt time: (org_node)-[:P571]->(org_inc) --opt quantities: (hosp_node)-[:P6801]->(count_beds) --opt project: (project)-[:PNIHcost]->(cost)\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project -i hosp -i time -i quantities\\\n",
    "--match 'author: (cluster)-[]->(qnode), nih_author:(qnihauthor)-[]->(qnode), inv:(project)-[]->(qnihauthor), \\\n",
    "        item:(qnode)-[:P108]->(org_node), label:(org_node)-[]->(org_name),label:(qnode)-[]->(author_name), \\\n",
    "        coor:(org_node)-[:P625]->(organization_coor), hosp:(qnihauthor)-[]->(hosp_node),\\\n",
    "        label:(hosp_node)-[]->(hosp_name), coor:(hosp_node)-[:P625]->(hosp_coor)'\\\n",
    "--opt 'time: (hosp_node)-[:P571]->(hosp_inc)' \\\n",
    "--opt 'time: (org_node)-[:P571]->(org_inc)' \\\n",
    "--opt 'quantities: (hosp_node)-[:P6801]->(count_beds)' \\\n",
    "--opt 'project: (project)-[:PNIHcost]->(cost)' \\\n",
    "--return 'cluster as cluster_id, org_name as org_name, sum(cost) as organization_award, \\\n",
    "        kgtk_geo_coords_lat(organization_coor) as org_lat, kgtk_geo_coords_long(organization_coor) as org_long,\\\n",
    "        kgtk_date_year(org_inc) as org_inc, \\\n",
    "        hosp_name as hosp_name, kgtk_geo_coords_lat(hosp_coor) as hosp_lat, kgtk_geo_coords_long(hosp_coor) as hosp_long,\\\n",
    "        kgtk_date_year(hosp_inc) as hosp_inc, count_beds as count_beds' \\\n",
    "-o bed_inception_1.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/grantxie/Downloads/NIH'"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('bed_inception_1.tsv',  sep = '\\t')\n",
    "df['count_beds'].fillna(value=df['count_beds'].mean(), inplace=True)\n",
    "df.to_csv('bed_incepition_imputed.tsv', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>org_name</th>\n",
       "      <th>organization_award</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>count_beds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_0_0_0_0_6_93</td>\n",
       "      <td>'Massachusetts Institute of Technology'@en</td>\n",
       "      <td>4658658.0</td>\n",
       "      <td>42.359820</td>\n",
       "      <td>-71.092110</td>\n",
       "      <td>1861.0</td>\n",
       "      <td>'Cambridge Health Alliance'@en</td>\n",
       "      <td>42.374800</td>\n",
       "      <td>-71.104800</td>\n",
       "      <td>1996.0</td>\n",
       "      <td>617.830986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_0_0_0_0_6_93</td>\n",
       "      <td>'Whitehead Institute'@en</td>\n",
       "      <td>4658658.0</td>\n",
       "      <td>42.363132</td>\n",
       "      <td>-71.089476</td>\n",
       "      <td>1982.0</td>\n",
       "      <td>'Cambridge Health Alliance'@en</td>\n",
       "      <td>42.374800</td>\n",
       "      <td>-71.104800</td>\n",
       "      <td>1996.0</td>\n",
       "      <td>617.830986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_0_0_0_0_39_106_193</td>\n",
       "      <td>'Vanderbilt University'@en</td>\n",
       "      <td>2175747.0</td>\n",
       "      <td>36.148649</td>\n",
       "      <td>-86.804972</td>\n",
       "      <td>1873.0</td>\n",
       "      <td>'Childrenâ€™s Hospital at TriStar Centennial'@en</td>\n",
       "      <td>36.153508</td>\n",
       "      <td>-86.807237</td>\n",
       "      <td>NaN</td>\n",
       "      <td>617.830986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_0_0_0_14_19_25_126</td>\n",
       "      <td>'University of California, San Diego'@en</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.881000</td>\n",
       "      <td>-117.238000</td>\n",
       "      <td>1960.0</td>\n",
       "      <td>'Jacobs Medical Center'@en</td>\n",
       "      <td>32.877703</td>\n",
       "      <td>-117.226499</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>364.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_0_0_0_0_14_66_154_718</td>\n",
       "      <td>'Stanford University'@en</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.428229</td>\n",
       "      <td>-122.168858</td>\n",
       "      <td>1891.0</td>\n",
       "      <td>'Stanford University Medical Center'@en</td>\n",
       "      <td>37.434000</td>\n",
       "      <td>-122.175000</td>\n",
       "      <td>1959.0</td>\n",
       "      <td>613.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>cluster_0_1_4_7_9_16_129_368_1416</td>\n",
       "      <td>'University of California, Berkeley'@en</td>\n",
       "      <td>8727072.0</td>\n",
       "      <td>37.870000</td>\n",
       "      <td>-122.259000</td>\n",
       "      <td>1868.0</td>\n",
       "      <td>'Children\\'s Hospital Oakland'@en</td>\n",
       "      <td>37.837500</td>\n",
       "      <td>-122.267000</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>191.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>cluster_0_1_4_7_9_16_129_368_1416</td>\n",
       "      <td>'University of Washington'@en</td>\n",
       "      <td>8727072.0</td>\n",
       "      <td>47.654167</td>\n",
       "      <td>-122.308056</td>\n",
       "      <td>1861.0</td>\n",
       "      <td>'Children\\'s Hospital Oakland'@en</td>\n",
       "      <td>37.837500</td>\n",
       "      <td>-122.267000</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>191.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>cluster_0_1_4_7_9_16_21_144_1142</td>\n",
       "      <td>'California Institute of Technology'@en</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.137500</td>\n",
       "      <td>-118.125000</td>\n",
       "      <td>1891.0</td>\n",
       "      <td>'Huntington Hospital'@en</td>\n",
       "      <td>34.134500</td>\n",
       "      <td>-118.153000</td>\n",
       "      <td>1892.0</td>\n",
       "      <td>625.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>cluster_0_1_4_7_9_16_21_144_1142</td>\n",
       "      <td>'Stanford University'@en</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.428229</td>\n",
       "      <td>-122.168858</td>\n",
       "      <td>1891.0</td>\n",
       "      <td>'Huntington Hospital'@en</td>\n",
       "      <td>34.134500</td>\n",
       "      <td>-118.153000</td>\n",
       "      <td>1892.0</td>\n",
       "      <td>625.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>cluster_0_1_4_7_9_16_21_144_1144</td>\n",
       "      <td>'Oregon Health &amp; Science University'@en</td>\n",
       "      <td>629276.0</td>\n",
       "      <td>45.498917</td>\n",
       "      <td>-122.688750</td>\n",
       "      <td>1974.0</td>\n",
       "      <td>'Doernbecher Children\\'s Hospital'@en</td>\n",
       "      <td>45.499800</td>\n",
       "      <td>-122.688000</td>\n",
       "      <td>1926.0</td>\n",
       "      <td>617.830986</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>299 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            cluster_id  \\\n",
       "0           cluster_0_0_0_0_0_0_0_6_93   \n",
       "1           cluster_0_0_0_0_0_0_0_6_93   \n",
       "2       cluster_0_0_0_0_0_0_39_106_193   \n",
       "3       cluster_0_0_0_0_0_14_19_25_126   \n",
       "4      cluster_0_0_0_0_0_14_66_154_718   \n",
       "..                                 ...   \n",
       "294  cluster_0_1_4_7_9_16_129_368_1416   \n",
       "295  cluster_0_1_4_7_9_16_129_368_1416   \n",
       "296   cluster_0_1_4_7_9_16_21_144_1142   \n",
       "297   cluster_0_1_4_7_9_16_21_144_1142   \n",
       "298   cluster_0_1_4_7_9_16_21_144_1144   \n",
       "\n",
       "                                       org_name  organization_award  \\\n",
       "0    'Massachusetts Institute of Technology'@en           4658658.0   \n",
       "1                      'Whitehead Institute'@en           4658658.0   \n",
       "2                    'Vanderbilt University'@en           2175747.0   \n",
       "3      'University of California, San Diego'@en                 NaN   \n",
       "4                      'Stanford University'@en                 NaN   \n",
       "..                                          ...                 ...   \n",
       "294     'University of California, Berkeley'@en           8727072.0   \n",
       "295               'University of Washington'@en           8727072.0   \n",
       "296     'California Institute of Technology'@en                 NaN   \n",
       "297                    'Stanford University'@en                 NaN   \n",
       "298     'Oregon Health & Science University'@en            629276.0   \n",
       "\n",
       "       org_lat    org_long  org_inc  \\\n",
       "0    42.359820  -71.092110   1861.0   \n",
       "1    42.363132  -71.089476   1982.0   \n",
       "2    36.148649  -86.804972   1873.0   \n",
       "3    32.881000 -117.238000   1960.0   \n",
       "4    37.428229 -122.168858   1891.0   \n",
       "..         ...         ...      ...   \n",
       "294  37.870000 -122.259000   1868.0   \n",
       "295  47.654167 -122.308056   1861.0   \n",
       "296  34.137500 -118.125000   1891.0   \n",
       "297  37.428229 -122.168858   1891.0   \n",
       "298  45.498917 -122.688750   1974.0   \n",
       "\n",
       "                                          hosp_name   hosp_lat   hosp_long  \\\n",
       "0                    'Cambridge Health Alliance'@en  42.374800  -71.104800   \n",
       "1                    'Cambridge Health Alliance'@en  42.374800  -71.104800   \n",
       "2    'Childrenâ€™s Hospital at TriStar Centennial'@en  36.153508  -86.807237   \n",
       "3                        'Jacobs Medical Center'@en  32.877703 -117.226499   \n",
       "4           'Stanford University Medical Center'@en  37.434000 -122.175000   \n",
       "..                                              ...        ...         ...   \n",
       "294               'Children\\'s Hospital Oakland'@en  37.837500 -122.267000   \n",
       "295               'Children\\'s Hospital Oakland'@en  37.837500 -122.267000   \n",
       "296                        'Huntington Hospital'@en  34.134500 -118.153000   \n",
       "297                        'Huntington Hospital'@en  34.134500 -118.153000   \n",
       "298           'Doernbecher Children\\'s Hospital'@en  45.499800 -122.688000   \n",
       "\n",
       "     hosp_inc  count_beds  \n",
       "0      1996.0  617.830986  \n",
       "1      1996.0  617.830986  \n",
       "2         NaN  617.830986  \n",
       "3      2016.0  364.000000  \n",
       "4      1959.0  613.000000  \n",
       "..        ...         ...  \n",
       "294    1912.0  191.000000  \n",
       "295    1912.0  191.000000  \n",
       "296    1892.0  625.000000  \n",
       "297    1892.0  625.000000  \n",
       "298    1926.0  617.830986  \n",
       "\n",
       "[299 rows x 11 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel('bed_incepition_imputed_1.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('nih_sum_cost.tsv', sep = '\\t')\n",
    "df['total_cost'].fillna(value=df['total_cost'].mean(), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel('/Users/grantxie/Downloads/nih_sum_cost_by_org.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-17 05:03:06 sqlstore]: IMPORT graph directly into table graph_38 from /Users/grantxie/Downloads/claims.quantity.tsv.gz ...\n",
      "[2021-08-17 05:12:08 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_38 AS graph_38_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trank\tnode2;wikidatatype\n",
      "P1004-P4876-b5e617-af81a4c5-0\tP1004\tP4876\t+37427\tnormal\tquantity\n",
      "P1014-P4876-605b34-fbb0b790-0\tP1014\tP4876\t+53249\tpreferred\tquantity\n",
      "P1014-P4876-62c45c-db5d6620-0\tP1014\tP4876\t+50825\tnormal\tquantity\n",
      "P1014-P4876-732e1a-9ea8b31a-0\tP1014\tP4876\t+47267\tnormal\tquantity\n",
      "P1014-P4876-f7a212-293bd556-0\tP1014\tP4876\t+46591\tnormal\tquantity\n",
      "P1022-P4876-fa0ca8-40e422f8-0\tP1022\tP4876\t+541\tnormal\tquantity\n",
      "P1024-P4876-8fb399-9696d4ac-0\tP1024\tP4876\t+857\tnormal\tquantity\n",
      "P1025-P4876-f0dc63-acd0fe33-0\tP1025\tP4876\t+13000000\tnormal\tquantity\n",
      "P1042-P1114-27b781-00b6c31d-0\tP1042\tP1114\t+1952404\tnormal\tquantity\n",
      "P1044-P4876-d72e5b-569d2ef4-0\tP1044\tP4876\t+54300000\tnormal\tquantity\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/claims.quantity.tsv.gz\" --as quantities\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kypher -i item -i coor -i label\\\n",
    "--match 'item: (c)-[]->(:Q16917), coor:(c)-[:P625]->(d), label:(c)-[]->(e)'\\\n",
    "--force \\\n",
    "--return 'c, d, e' \\\n",
    "-o name1.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-17 04:22:37 sqlstore]: IMPORT graph directly into table graph_36 from /Users/grantxie/Downloads/nih_match_hospital.tsv ...\n",
      "[2021-08-17 04:22:37 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_36 AS graph_36_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "node1\tnode2\tcoordinates\tnearest_label\tnearest_coordinates\tdistance\n",
      "Q219563\tUNIVERSITY OF WASHINGTON\t@47.65416666666667/-122.30805555555555\t'University of Washington Medical Center'@en\t@47.6489232/-122.3065945\t2.9628626020887572e-05\n",
      "Q5081831\tCHARLES R. DREW UNIVERSITY OF MEDICAL & SCIENCE\t@33.925634/-118.242594\t'Martin Luther King, Jr. Multi-Service Ambulatory Care Center'@en\t@33.9238/-118.242\t3.7163919999995717e-06\n",
      "Q622664\tUNIVERSITY OF CALIFORNIA, SAN DIEGO\t@32.881/-117.238\t'Jacobs Medical Center'@en\t@32.877703/-117.226499\t0.00014314320999992013\n",
      "Q812573\tBAYLOR UNIVERSITY\t@31.5472/-97.1139\t'Scott & White Memorial Hospital'@en\t@31.07753/-97.36383\t0.28305491379999664\n",
      "Q49118\tBOSTON COLLEGE\t@42.335083333333/-71.170361111111\t'Floating Hospital for Children'@en\t@42.353267/-71.173681\t0.00034166739569182304\n",
      "Q6806451\tMEDICAL UNIVERSITY OF SOUTH CAROLINA\t@32.784166666667/-79.950833333333\t'Roper Hospital'@en\t@32.7826/-79.949593\t3.992871222429877e-06\n",
      "Q457281\tUNIVERSITY OF ILLINOIS AT URBANA-CHAMPAIGN\t@40.110538888889/-88.228411111111\t'OSF Heart of Mary Medical Center'@en\t@40.11792/-88.22746\t5.538541357833092e-05\n",
      "Q168751\tDUKE UNIVERSITY\t@36.001111111111/-78.938888888889\t'Duke Cancer Institute'@en\t@36.00496/-78.93689\t1.8809502470368104e-05\n",
      "Q349055\tWAYNE STATE UNIVERSITY\t@42.357344444444/-83.070105555556\t'Dunbar Hospital'@en\t@42.361667/-83.058889\t0.00014449560507558061\n",
      "Q104480607\tNORTHAMPTON VA MEDICAL CENTER\t@35.588877/-82.484897\t'Clarence Barker Memorial Hospital'@en\t@35.565/-82.5411\t0.0037288883379995442\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/nih_match_hospital.tsv\" --as hospital\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-16 20:30:03 sqlstore]: IMPORT graph directly into table graph_30 from /Users/grantxie/Downloads/kgtk-master/use-cases/nih_author_kgtk.tsv ...\n",
      "[2021-08-16 20:30:03 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_30 AS graph_30_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trow\tnode2;label\tgraph\n",
      "author0\tQNIHPER1940486\tlabel\tQ6831539\t2\tMICHAEL J GALE\tauthor\n",
      "author1\tQNIHPER2414340\tlabel\tQ87796696\t5\tPATRICK S. STAYTON\tauthor\n",
      "author2\tQNIHPER11537130\tlabel\tQ89958961\t8\tKENNETH NGURE\tauthor\n",
      "author3\tQNIHPER8820221\tlabel\tQ61467477\t10\tANITHA PASUPATHY\tauthor\n",
      "author4\tQNIHPER8491201\tlabel\tQ46001993\t11\tSTEPHANIE M FULLERTON\tauthor\n",
      "author5\tQNIHPER7882742\tlabel\tQ89866816\t13\tMARION PEPPER\tauthor\n",
      "author6\tQNIHPER8146532\tlabel\tQ89937644\t14\tHORACIO O DE LA IGLESIA\tauthor\n",
      "author7\tQNIHPER10499986\tlabel\tQ56809498\t17\tKEVIN HYBISKE\tauthor\n",
      "author8\tQNIHPER2315018\tlabel\tQ87887672\t18\tKARIN E. BORNFELDT\tauthor\n",
      "author9\tQNIHPER3065493\tlabel\tQ90437737\t21\tRAYMOND SCOTT MCCLELLAND\tauthor\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/kgtk-master/use-cases/nih_author_kgtk.tsv\" --as nih_author \\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/Downloads/NIH\n"
     ]
    }
   ],
   "source": [
    "%cd /Users/grantxie/Downloads/NIH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-09 19:46:06 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT graph_23_c1.\"node1\", graph_23_c1.\"node2\", graph_22_c2.\"node2\"\n",
      "     FROM graph_22 AS graph_22_c2\n",
      "     INNER JOIN graph_23 AS graph_23_c1\n",
      "     ON graph_23_c1.\"node1\" = graph_22_c2.\"node1\"\n",
      "        AND graph_22_c2.\"label\" = ?\n",
      "  PARAS: ['P625']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i coor -i nih\\\n",
    "--match 'nih:(c)-[]->(d), coor:(c)-[:P625]->(e)'\\\n",
    "--return 'c, d, e' \\\n",
    "-o name_nih_new.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating closest hospital for organizations (using kdtree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('name1.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = []\n",
    "y = []\n",
    "\n",
    "for i in range(0, len(df)):\n",
    "    x.append(df.node2[i][1:].split('/')[0])\n",
    "    y.append(df.node2[i][1:].split('/')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "x_c = []\n",
    "y_c = []\n",
    "\n",
    "for i in range(0, len(df)):\n",
    "    x_c.append(float(x[i]))\n",
    "    y_c.append(float(y[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "import kdtree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "emptyTree = kdtree.create(dimensions=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "arr = []\n",
    "for i in range(0, len(df)):\n",
    "    arr.append((x_c[i], y_c[i]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "tree = kdtree.create(arr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "nih = pd.read_csv('/Users/grantxie/Downloads/cluster_coor.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>node2;coor</th>\n",
       "      <th>node2;label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10003954</td>\n",
       "      <td>@41.504/-81.608</td>\n",
       "      <td>'Case Western Reserve University'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10428590</td>\n",
       "      <td>@37.5625/126.945</td>\n",
       "      <td>'Ewha Womans University'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1800</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10004092</td>\n",
       "      <td>@44.564588/-123.275705</td>\n",
       "      <td>'Oregon State University'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER12571233</td>\n",
       "      <td>@42.447222222222/-76.483055555556</td>\n",
       "      <td>'Cornell University'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER2098890</td>\n",
       "      <td>@46.343224/-119.276333</td>\n",
       "      <td>'Pacific Northwest National Laboratory'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_8_259</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9634058</td>\n",
       "      <td>@35.908611111111/-79.049166666667</td>\n",
       "      <td>'University of North Carolina at Chapel Hill'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2378</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_7_2_843</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9728962</td>\n",
       "      <td>@38.907222222222/-77.072777777778</td>\n",
       "      <td>'Georgetown University'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2379</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_7_2_2378</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9883306</td>\n",
       "      <td>@38.912/-77.077</td>\n",
       "      <td>'Georgetown University Medical Center'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2380</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_77_1289</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9807979</td>\n",
       "      <td>@40.73/-73.995</td>\n",
       "      <td>'New York University'@en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2381</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_77_1289</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9859180</td>\n",
       "      <td>@40.73/-73.995</td>\n",
       "      <td>'New York University'@en</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2382 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                       node1   label            node2  \\\n",
       "0    NaN  cluster_0_0_2_2_14_112_322  member  QNIHPER10003954   \n",
       "1    NaN  cluster_0_0_2_2_14_112_322  member  QNIHPER10428590   \n",
       "2    NaN  cluster_0_0_3_6_21_40_1800  member  QNIHPER10004092   \n",
       "3    NaN  cluster_0_0_3_6_21_40_1238  member  QNIHPER12571233   \n",
       "4    NaN  cluster_0_0_3_6_21_40_1238  member   QNIHPER2098890   \n",
       "...   ..                         ...     ...              ...   \n",
       "2377 NaN    cluster_0_0_2_2_14_8_259  member   QNIHPER9634058   \n",
       "2378 NaN     cluster_0_0_3_6_7_2_843  member   QNIHPER9728962   \n",
       "2379 NaN    cluster_0_0_3_6_7_2_2378  member   QNIHPER9883306   \n",
       "2380 NaN  cluster_0_0_3_6_21_77_1289  member   QNIHPER9807979   \n",
       "2381 NaN  cluster_0_0_3_6_21_77_1289  member   QNIHPER9859180   \n",
       "\n",
       "                             node2;coor  \\\n",
       "0                       @41.504/-81.608   \n",
       "1                      @37.5625/126.945   \n",
       "2                @44.564588/-123.275705   \n",
       "3     @42.447222222222/-76.483055555556   \n",
       "4                @46.343224/-119.276333   \n",
       "...                                 ...   \n",
       "2377  @35.908611111111/-79.049166666667   \n",
       "2378  @38.907222222222/-77.072777777778   \n",
       "2379                    @38.912/-77.077   \n",
       "2380                     @40.73/-73.995   \n",
       "2381                     @40.73/-73.995   \n",
       "\n",
       "                                           node2;label  \n",
       "0                 'Case Western Reserve University'@en  \n",
       "1                          'Ewha Womans University'@en  \n",
       "2                         'Oregon State University'@en  \n",
       "3                              'Cornell University'@en  \n",
       "4           'Pacific Northwest National Laboratory'@en  \n",
       "...                                                ...  \n",
       "2377  'University of North Carolina at Chapel Hill'@en  \n",
       "2378                        'Georgetown University'@en  \n",
       "2379         'Georgetown University Medical Center'@en  \n",
       "2380                          'New York University'@en  \n",
       "2381                          'New York University'@en  \n",
       "\n",
       "[2382 rows x 6 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nih"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "nih_vec = []\n",
    "x = []\n",
    "y = []\n",
    "for i in range(0, len(nih)):\n",
    "    x.append(nih['node2;coor'][i][1:].split('/')[0])\n",
    "    y.append(nih['node2;coor'][i][1:].split('/')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = []\n",
    "for i in range(0, len(nih)):\n",
    "    results.append(tree.search_nn((float(x[i]), float(y[i]))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(37.5624, 126.941)"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results[1][0].data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "total = []\n",
    "for i in range(0, len(x_c)):\n",
    "    total.append((x_c[i], y_c[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38.248967"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tree.data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7451"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total.index(results[0][0].data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "index = []\n",
    "for i in range(0, len(results)):\n",
    "    index.append(total.index(results[i][0].data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "nih['nearest_node'] = ''\n",
    "nih['nearest_label'] = ''\n",
    "nih['nearest_coordinates'] = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(nih)):\n",
    "    nih['nearest_node'][i] = df['node1'][index[i]]\n",
    "    nih['nearest_label'][i] = df['node2.1'][index[i]]\n",
    "    nih['nearest_coordinates'][i] = df['node2'][index[i]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "nih['distance'] = -1.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(nih)):\n",
    "    nih['distance'][i] = results[i][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>node2;coor</th>\n",
       "      <th>node2;label</th>\n",
       "      <th>nearest_node</th>\n",
       "      <th>nearest_label</th>\n",
       "      <th>nearest_coordinates</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10003954</td>\n",
       "      <td>@41.504/-81.608</td>\n",
       "      <td>'Case Western Reserve University'@en</td>\n",
       "      <td>Q39050124</td>\n",
       "      <td>'Case Comprehensive Cancer Center'@en</td>\n",
       "      <td>@41.504/-81.608</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10428590</td>\n",
       "      <td>@37.5625/126.945</td>\n",
       "      <td>'Ewha Womans University'@en</td>\n",
       "      <td>Q625321</td>\n",
       "      <td>'Severance Hospital'@en</td>\n",
       "      <td>@37.5624/126.941</td>\n",
       "      <td>0.000016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1800</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10004092</td>\n",
       "      <td>@44.564588/-123.275705</td>\n",
       "      <td>'Oregon State University'@en</td>\n",
       "      <td>Q5582923</td>\n",
       "      <td>'Good Samaritan Regional Medical Center'@en</td>\n",
       "      <td>@44.6022/-123.252</td>\n",
       "      <td>0.001977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER12571233</td>\n",
       "      <td>@42.447222222222/-76.483055555556</td>\n",
       "      <td>'Cornell University'@en</td>\n",
       "      <td>Q28035413</td>\n",
       "      <td>'Cayuga Medical Center'@en</td>\n",
       "      <td>@42.469273/-76.53727</td>\n",
       "      <td>0.003425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER2098890</td>\n",
       "      <td>@46.343224/-119.276333</td>\n",
       "      <td>'Pacific Northwest National Laboratory'@en</td>\n",
       "      <td>Q50036737</td>\n",
       "      <td>'Kadlec Clinic'@en</td>\n",
       "      <td>@46.28146/-119.281346</td>\n",
       "      <td>0.003840</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id                       node1   label            node2  \\\n",
       "0 NaN  cluster_0_0_2_2_14_112_322  member  QNIHPER10003954   \n",
       "1 NaN  cluster_0_0_2_2_14_112_322  member  QNIHPER10428590   \n",
       "2 NaN  cluster_0_0_3_6_21_40_1800  member  QNIHPER10004092   \n",
       "3 NaN  cluster_0_0_3_6_21_40_1238  member  QNIHPER12571233   \n",
       "4 NaN  cluster_0_0_3_6_21_40_1238  member   QNIHPER2098890   \n",
       "\n",
       "                          node2;coor  \\\n",
       "0                    @41.504/-81.608   \n",
       "1                   @37.5625/126.945   \n",
       "2             @44.564588/-123.275705   \n",
       "3  @42.447222222222/-76.483055555556   \n",
       "4             @46.343224/-119.276333   \n",
       "\n",
       "                                  node2;label nearest_node  \\\n",
       "0        'Case Western Reserve University'@en    Q39050124   \n",
       "1                 'Ewha Womans University'@en      Q625321   \n",
       "2                'Oregon State University'@en     Q5582923   \n",
       "3                     'Cornell University'@en    Q28035413   \n",
       "4  'Pacific Northwest National Laboratory'@en    Q50036737   \n",
       "\n",
       "                                 nearest_label    nearest_coordinates  \\\n",
       "0        'Case Comprehensive Cancer Center'@en        @41.504/-81.608   \n",
       "1                      'Severance Hospital'@en       @37.5624/126.941   \n",
       "2  'Good Samaritan Regional Medical Center'@en      @44.6022/-123.252   \n",
       "3                   'Cayuga Medical Center'@en   @42.469273/-76.53727   \n",
       "4                           'Kadlec Clinic'@en  @46.28146/-119.281346   \n",
       "\n",
       "   distance  \n",
       "0  0.000000  \n",
       "1  0.000016  \n",
       "2  0.001977  \n",
       "3  0.003425  \n",
       "4  0.003840  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nih.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "output = pd.DataFrame(list(zip(nih['id'], nih.node2, nih.label, nih.nearest_node)) ,columns = ['id', 'node1', 'label', 'node2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "output.to_csv('author_hospital.tsv', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kgtk add-id --id-style wikidata \\\n",
    "-i \"author_hospital.tsv\" -o author_hospital.id.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER10003954</td>\n",
       "      <td>member</td>\n",
       "      <td>Q39050124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER10428590</td>\n",
       "      <td>member</td>\n",
       "      <td>Q625321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER10004092</td>\n",
       "      <td>member</td>\n",
       "      <td>Q5582923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER12571233</td>\n",
       "      <td>member</td>\n",
       "      <td>Q28035413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER2098890</td>\n",
       "      <td>member</td>\n",
       "      <td>Q50036737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER9634058</td>\n",
       "      <td>member</td>\n",
       "      <td>Q30288421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2378</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER9728962</td>\n",
       "      <td>member</td>\n",
       "      <td>Q5547057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2379</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER9883306</td>\n",
       "      <td>member</td>\n",
       "      <td>Q5547057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2380</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER9807979</td>\n",
       "      <td>member</td>\n",
       "      <td>Q22061077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2381</th>\n",
       "      <td>NaN</td>\n",
       "      <td>QNIHPER9859180</td>\n",
       "      <td>member</td>\n",
       "      <td>Q22061077</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2382 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id            node1   label      node2\n",
       "0    NaN  QNIHPER10003954  member  Q39050124\n",
       "1    NaN  QNIHPER10428590  member    Q625321\n",
       "2    NaN  QNIHPER10004092  member   Q5582923\n",
       "3    NaN  QNIHPER12571233  member  Q28035413\n",
       "4    NaN   QNIHPER2098890  member  Q50036737\n",
       "...   ..              ...     ...        ...\n",
       "2377 NaN   QNIHPER9634058  member  Q30288421\n",
       "2378 NaN   QNIHPER9728962  member   Q5547057\n",
       "2379 NaN   QNIHPER9883306  member   Q5547057\n",
       "2380 NaN   QNIHPER9807979  member  Q22061077\n",
       "2381 NaN   QNIHPER9859180  member  Q22061077\n",
       "\n",
       "[2382 rows x 4 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = nih.groupby('node1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_cluster_dist(cluster):\n",
    "    \n",
    "    group = nih.groupby('node1')\n",
    "    df = group.get_group(cluster)\n",
    "    l = df.nearest_label.unique()\n",
    "    org = df.groupby('nearest_label')\n",
    "\n",
    "    c1 = []\n",
    "    c2 = []\n",
    "    for i in range(0, len(l)):\n",
    "        c1.append(l[i][1:len(l[i])-4])\n",
    "        c2.append(org.get_group(l[i]).count()[1])\n",
    "        \n",
    "    df = pd.DataFrame(list(zip(c1,c2)), columns = ['org', 'count'])\n",
    "    return df.sort_values(by='count', ascending=False).reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>org</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Washington University Medical Center</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Brigham and Women\\'s Hospital</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VA Center for Clinical Management Research</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Center for Emergency Medicine of Western Penns...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NYU Langone Medical Center</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>The Peak Mark Clinic</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>St. John\\'s Episcopal Hospital</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Bloomingdale Insane Asylum</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Morgan Stanley Children\\'s Hospital</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Shands at the University of Florida</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  org  count\n",
       "0                Washington University Medical Center      7\n",
       "1                       Brigham and Women\\'s Hospital      5\n",
       "2          VA Center for Clinical Management Research      5\n",
       "3   Center for Emergency Medicine of Western Penns...      4\n",
       "4                          NYU Langone Medical Center      4\n",
       "..                                                ...    ...\n",
       "57                               The Peak Mark Clinic      1\n",
       "58                     St. John\\'s Episcopal Hospital      1\n",
       "59                         Bloomingdale Insane Asylum      1\n",
       "60                Morgan Stanley Children\\'s Hospital      1\n",
       "61                Shands at the University of Florida      1\n",
       "\n",
       "[62 rows x 2 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_cluster_dist('cluster_0_0_2_2_14_112_322')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "def get_cluster_mean(cluster):\n",
    "    df_temp = pd.read_csv('name1.tsv', sep = '\\t')\n",
    "    group = nih.groupby('node1')\n",
    "    df = group.get_group(cluster).reset_index(drop = True)\n",
    "\n",
    "    df_vec = []\n",
    "    x = []\n",
    "    y = []\n",
    "    for i in range(0, len(df)):\n",
    "        x.append(float(df['node2;coor'][i][1:].split('/')[0]))\n",
    "        y.append(float(df['node2;coor'][i][1:].split('/')[1]))\n",
    "        \n",
    "    x1 = float(sum(x))/float(len(x))\n",
    "    y1 = float(sum(y))/float(len(y))\n",
    "    #print(x1 ,y1)\n",
    "    res = tree.search_nn([x1, y1])\n",
    "    #print(df_temp['node2.1'][total.index(results[i][0].data)][1: len(df_temp['node2.1'][total.index(results[i][0].data)])-4])\n",
    "    #print(res[0])\n",
    "    \n",
    "    tot = 0.0\n",
    "    \n",
    "    for i in range(0, len(x)):\n",
    "        tot = tot + 111 * math.sqrt( (res[0].data[0]-x[i])**2 + (res[0].data[1]-y[i])**2 )\n",
    "    #print(tot/len(x))\n",
    "    \n",
    "    return (df_temp['node2.1'][total.index(results[i][0].data)][1: len(df_temp['node2.1'][total.index(results[i][0].data)])-4], tot/len(x))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('Penn Medicine Princeton Medical Center', 1901.3523863405317)"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_cluster_mean('cluster_0_0_2_2_14_112_322')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "name = []\n",
    "dist = []\n",
    "cl = []\n",
    "count = []\n",
    "cluster = nih.node1.unique()\n",
    "label = []\n",
    "\n",
    "for i in range(0, len(cluster)):\n",
    "    group = nih.groupby('node1')\n",
    "    cl.append(cluster[i])\n",
    "    count.append(len(group.get_group(cluster[i])))\n",
    "    temp = get_cluster_mean(cluster[i])\n",
    "    name.append(temp[0])\n",
    "    dist.append(temp[1])\n",
    "    label.append('Near')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "output = pd.DataFrame(list(zip(cl,name, label, dist,count,)), columns = ['node1', 'label', 'node2', 'node2;dist', 'node1;count'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>node2;dist</th>\n",
       "      <th>node1;count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_3_5_15_45_215</td>\n",
       "      <td>Case Comprehensive Cancer Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>0.357465</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_3_5_30_39_104</td>\n",
       "      <td>Cayuga Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>1.564715</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_2_2_14_146_1984</td>\n",
       "      <td>Cayuga Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>19.162157</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_3_5_15_45_1068</td>\n",
       "      <td>Severance Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>46.704682</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_0_2_3_23_60_1880</td>\n",
       "      <td>Severance Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>170.079363</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>cluster_0_0_3_6_21_40_623</td>\n",
       "      <td>Spaulding Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>4858.154756</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>cluster_0_0_2_2_9_93_2350</td>\n",
       "      <td>James Whitcomb Riley Hospital for Children</td>\n",
       "      <td>Near</td>\n",
       "      <td>5135.594781</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>cluster_0_0_3_6_11_61_1689</td>\n",
       "      <td>Kadlec Clinic</td>\n",
       "      <td>Near</td>\n",
       "      <td>5234.390724</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>cluster_0_0_3_5_30_82_1616</td>\n",
       "      <td>Baylor College of Medicine</td>\n",
       "      <td>Near</td>\n",
       "      <td>6702.576310</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>cluster_0_0_3_6_11_61_1642</td>\n",
       "      <td>Baylor College of Medicine</td>\n",
       "      <td>Near</td>\n",
       "      <td>8357.909891</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>149 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           node1                                       label  \\\n",
       "0      cluster_0_0_3_5_15_45_215            Case Comprehensive Cancer Center   \n",
       "1      cluster_0_0_3_5_30_39_104                       Cayuga Medical Center   \n",
       "2    cluster_0_0_2_2_14_146_1984                       Cayuga Medical Center   \n",
       "3     cluster_0_0_3_5_15_45_1068                          Severance Hospital   \n",
       "4     cluster_0_0_2_3_23_60_1880                          Severance Hospital   \n",
       "..                           ...                                         ...   \n",
       "144    cluster_0_0_3_6_21_40_623                          Spaulding Hospital   \n",
       "145    cluster_0_0_2_2_9_93_2350  James Whitcomb Riley Hospital for Children   \n",
       "146   cluster_0_0_3_6_11_61_1689                               Kadlec Clinic   \n",
       "147   cluster_0_0_3_5_30_82_1616                  Baylor College of Medicine   \n",
       "148   cluster_0_0_3_6_11_61_1642                  Baylor College of Medicine   \n",
       "\n",
       "    node2   node2;dist  node1;count  \n",
       "0    Near     0.357465            1  \n",
       "1    Near     1.564715            4  \n",
       "2    Near    19.162157            4  \n",
       "3    Near    46.704682            2  \n",
       "4    Near   170.079363            2  \n",
       "..    ...          ...          ...  \n",
       "144  Near  4858.154756            9  \n",
       "145  Near  5135.594781           18  \n",
       "146  Near  5234.390724            5  \n",
       "147  Near  6702.576310           10  \n",
       "148  Near  8357.909891           10  \n",
       "\n",
       "[149 rows x 5 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output.sort_values(by='node2;dist', ascending=True).reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "output.sort_values(by='node2;dist', ascending=True).reset_index(drop = True).to_csv('hospital_mean_distance.tsv', index = False, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/grantxie/Downloads/NIH'"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>node2;dist</th>\n",
       "      <th>node1;count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>Penn Medicine Princeton Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>1901.352386</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_3_6_21_40_1800</td>\n",
       "      <td>Norman Regional Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>3734.246675</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>Baylor College of Medicine</td>\n",
       "      <td>Near</td>\n",
       "      <td>1504.803572</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_2_3_22_129_391</td>\n",
       "      <td>University of New Mexico Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>891.357219</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_0_3_5_30_15_1005</td>\n",
       "      <td>NYU Langone Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>1670.367667</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>cluster_0_0_2_3_23_85_782</td>\n",
       "      <td>Cayuga Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>1034.161456</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>cluster_0_0_2_3_24_103_2290</td>\n",
       "      <td>Good Samaritan Regional Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>1764.546490</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>cluster_0_0_2_3_24_103_1216</td>\n",
       "      <td>Cayuga Medical Center</td>\n",
       "      <td>Near</td>\n",
       "      <td>4119.954933</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>cluster_0_0_3_5_30_39_620</td>\n",
       "      <td>Stony Brook University Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>3813.832220</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>cluster_0_0_3_5_28_70_495</td>\n",
       "      <td>Severance Hospital</td>\n",
       "      <td>Near</td>\n",
       "      <td>430.439936</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>149 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           node1                                   label  \\\n",
       "0     cluster_0_0_2_2_14_112_322  Penn Medicine Princeton Medical Center   \n",
       "1     cluster_0_0_3_6_21_40_1800                Norman Regional Hospital   \n",
       "2     cluster_0_0_3_6_21_40_1238              Baylor College of Medicine   \n",
       "3     cluster_0_0_2_3_22_129_391       University of New Mexico Hospital   \n",
       "4     cluster_0_0_3_5_30_15_1005              NYU Langone Medical Center   \n",
       "..                           ...                                     ...   \n",
       "144    cluster_0_0_2_3_23_85_782                   Cayuga Medical Center   \n",
       "145  cluster_0_0_2_3_24_103_2290  Good Samaritan Regional Medical Center   \n",
       "146  cluster_0_0_2_3_24_103_1216                   Cayuga Medical Center   \n",
       "147    cluster_0_0_3_5_30_39_620         Stony Brook University Hospital   \n",
       "148    cluster_0_0_3_5_28_70_495                      Severance Hospital   \n",
       "\n",
       "    node2   node2;dist  node1;count  \n",
       "0    Near  1901.352386          111  \n",
       "1    Near  3734.246675            6  \n",
       "2    Near  1504.803572           10  \n",
       "3    Near   891.357219           16  \n",
       "4    Near  1670.367667           13  \n",
       "..    ...          ...          ...  \n",
       "144  Near  1034.161456            4  \n",
       "145  Near  1764.546490            3  \n",
       "146  Near  4119.954933            4  \n",
       "147  Near  3813.832220            7  \n",
       "148  Near   430.439936            2  \n",
       "\n",
       "[149 rows x 5 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>node1</th>\n",
       "      <th>label</th>\n",
       "      <th>node2</th>\n",
       "      <th>node2;coor</th>\n",
       "      <th>node2;label</th>\n",
       "      <th>nearest_label</th>\n",
       "      <th>nearest_coordinates</th>\n",
       "      <th>distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10003954</td>\n",
       "      <td>@41.504/-81.608</td>\n",
       "      <td>'Case Western Reserve University'@en</td>\n",
       "      <td>'Case Comprehensive Cancer Center'@en</td>\n",
       "      <td>@41.504/-81.608</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_112_322</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10428590</td>\n",
       "      <td>@37.5625/126.945</td>\n",
       "      <td>'Ewha Womans University'@en</td>\n",
       "      <td>'Severance Hospital'@en</td>\n",
       "      <td>@37.5624/126.941</td>\n",
       "      <td>0.000016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1800</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER10004092</td>\n",
       "      <td>@44.564588/-123.275705</td>\n",
       "      <td>'Oregon State University'@en</td>\n",
       "      <td>'Good Samaritan Regional Medical Center'@en</td>\n",
       "      <td>@44.6022/-123.252</td>\n",
       "      <td>0.001977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER12571233</td>\n",
       "      <td>@42.447222222222/-76.483055555556</td>\n",
       "      <td>'Cornell University'@en</td>\n",
       "      <td>'Cayuga Medical Center'@en</td>\n",
       "      <td>@42.469273/-76.53727</td>\n",
       "      <td>0.003425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_40_1238</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER2098890</td>\n",
       "      <td>@46.343224/-119.276333</td>\n",
       "      <td>'Pacific Northwest National Laboratory'@en</td>\n",
       "      <td>'Kadlec Clinic'@en</td>\n",
       "      <td>@46.28146/-119.281346</td>\n",
       "      <td>0.003840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_2_2_14_8_259</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9634058</td>\n",
       "      <td>@35.908611111111/-79.049166666667</td>\n",
       "      <td>'University of North Carolina at Chapel Hill'@en</td>\n",
       "      <td>'University of North Carolina Hospitals'@en</td>\n",
       "      <td>@35.90515/-79.050011</td>\n",
       "      <td>0.000013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2378</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_7_2_843</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9728962</td>\n",
       "      <td>@38.907222222222/-77.072777777778</td>\n",
       "      <td>'Georgetown University'@en</td>\n",
       "      <td>'Georgetown University Medical Center'@en</td>\n",
       "      <td>@38.912/-77.077</td>\n",
       "      <td>0.000041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2379</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_7_2_2378</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9883306</td>\n",
       "      <td>@38.912/-77.077</td>\n",
       "      <td>'Georgetown University Medical Center'@en</td>\n",
       "      <td>'Georgetown University Medical Center'@en</td>\n",
       "      <td>@38.912/-77.077</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2380</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_77_1289</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9807979</td>\n",
       "      <td>@40.73/-73.995</td>\n",
       "      <td>'New York University'@en</td>\n",
       "      <td>'St. John\\'s Episcopal Hospital'@en</td>\n",
       "      <td>@40.725656/-73.991794</td>\n",
       "      <td>0.000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2381</th>\n",
       "      <td>NaN</td>\n",
       "      <td>cluster_0_0_3_6_21_77_1289</td>\n",
       "      <td>member</td>\n",
       "      <td>QNIHPER9859180</td>\n",
       "      <td>@40.73/-73.995</td>\n",
       "      <td>'New York University'@en</td>\n",
       "      <td>'St. John\\'s Episcopal Hospital'@en</td>\n",
       "      <td>@40.725656/-73.991794</td>\n",
       "      <td>0.000029</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2382 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                       node1   label            node2  \\\n",
       "0    NaN  cluster_0_0_2_2_14_112_322  member  QNIHPER10003954   \n",
       "1    NaN  cluster_0_0_2_2_14_112_322  member  QNIHPER10428590   \n",
       "2    NaN  cluster_0_0_3_6_21_40_1800  member  QNIHPER10004092   \n",
       "3    NaN  cluster_0_0_3_6_21_40_1238  member  QNIHPER12571233   \n",
       "4    NaN  cluster_0_0_3_6_21_40_1238  member   QNIHPER2098890   \n",
       "...   ..                         ...     ...              ...   \n",
       "2377 NaN    cluster_0_0_2_2_14_8_259  member   QNIHPER9634058   \n",
       "2378 NaN     cluster_0_0_3_6_7_2_843  member   QNIHPER9728962   \n",
       "2379 NaN    cluster_0_0_3_6_7_2_2378  member   QNIHPER9883306   \n",
       "2380 NaN  cluster_0_0_3_6_21_77_1289  member   QNIHPER9807979   \n",
       "2381 NaN  cluster_0_0_3_6_21_77_1289  member   QNIHPER9859180   \n",
       "\n",
       "                             node2;coor  \\\n",
       "0                       @41.504/-81.608   \n",
       "1                      @37.5625/126.945   \n",
       "2                @44.564588/-123.275705   \n",
       "3     @42.447222222222/-76.483055555556   \n",
       "4                @46.343224/-119.276333   \n",
       "...                                 ...   \n",
       "2377  @35.908611111111/-79.049166666667   \n",
       "2378  @38.907222222222/-77.072777777778   \n",
       "2379                    @38.912/-77.077   \n",
       "2380                     @40.73/-73.995   \n",
       "2381                     @40.73/-73.995   \n",
       "\n",
       "                                           node2;label  \\\n",
       "0                 'Case Western Reserve University'@en   \n",
       "1                          'Ewha Womans University'@en   \n",
       "2                         'Oregon State University'@en   \n",
       "3                              'Cornell University'@en   \n",
       "4           'Pacific Northwest National Laboratory'@en   \n",
       "...                                                ...   \n",
       "2377  'University of North Carolina at Chapel Hill'@en   \n",
       "2378                        'Georgetown University'@en   \n",
       "2379         'Georgetown University Medical Center'@en   \n",
       "2380                          'New York University'@en   \n",
       "2381                          'New York University'@en   \n",
       "\n",
       "                                    nearest_label    nearest_coordinates  \\\n",
       "0           'Case Comprehensive Cancer Center'@en        @41.504/-81.608   \n",
       "1                         'Severance Hospital'@en       @37.5624/126.941   \n",
       "2     'Good Samaritan Regional Medical Center'@en      @44.6022/-123.252   \n",
       "3                      'Cayuga Medical Center'@en   @42.469273/-76.53727   \n",
       "4                              'Kadlec Clinic'@en  @46.28146/-119.281346   \n",
       "...                                           ...                    ...   \n",
       "2377  'University of North Carolina Hospitals'@en   @35.90515/-79.050011   \n",
       "2378    'Georgetown University Medical Center'@en        @38.912/-77.077   \n",
       "2379    'Georgetown University Medical Center'@en        @38.912/-77.077   \n",
       "2380          'St. John\\'s Episcopal Hospital'@en  @40.725656/-73.991794   \n",
       "2381          'St. John\\'s Episcopal Hospital'@en  @40.725656/-73.991794   \n",
       "\n",
       "      distance  \n",
       "0     0.000000  \n",
       "1     0.000016  \n",
       "2     0.001977  \n",
       "3     0.003425  \n",
       "4     0.003840  \n",
       "...        ...  \n",
       "2377  0.000013  \n",
       "2378  0.000041  \n",
       "2379  0.000000  \n",
       "2380  0.000029  \n",
       "2381  0.000029  \n",
       "\n",
       "[2382 rows x 9 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nih"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "nih['lat'] = -1.1\n",
    "nih['long'] = -1.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(nih)):\n",
    "    nih.lat[i] = float(nih['node2;coor'][i][1:].split('/')[0])\n",
    "    nih.long[i] = float(nih['node2;coor'][i][1:].split('/')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "nih.to_excel('nih.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Output Tableau files to show main topics (subjects) of papers for each organization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kgtk add-id --id-style wikidata -i \"/Users/grantxie/Downloads/NIH/nih_sub_sep.tsv\" -o nih_sub_sep.id.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-19 00:19:24 sqlstore]: IMPORT graph directly into table graph_40 from /Users/grantxie/Downloads/NIH/nih_sub_sep.id.tsv ...\n",
      "[2021-08-19 00:19:43 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_40 AS graph_40_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPRO10286324-PNIHsub-fc3d31\tQNIHPRO10286324\tPNIHsub\tantibodies\n",
      "QNIHPRO10286324-PNIHsub-cae662\tQNIHPRO10286324\tPNIHsub\tbase\n",
      "QNIHPRO10286324-PNIHsub-9621fb\tQNIHPRO10286324\tPNIHsub\tbiopsy\n",
      "QNIHPRO10286324-PNIHsub-e5b84c\tQNIHPRO10286324\tPNIHsub\tblind\n",
      "QNIHPRO10286324-PNIHsub-505b64\tQNIHPRO10286324\tPNIHsub\tblood tests\n",
      "QNIHPRO10286324-PNIHsub-d361d3\tQNIHPRO10286324\tPNIHsub\tbone marrow biopsy\n",
      "QNIHPRO10286324-PNIHsub-c933a4\tQNIHPRO10286324\tPNIHsub\tburden of illness\n",
      "QNIHPRO10286324-PNIHsub-81ad91\tQNIHPRO10286324\tPNIHsub\tcancer imaging\n",
      "QNIHPRO10286324-PNIHsub-57104c\tQNIHPRO10286324\tPNIHsub\tcells\n",
      "QNIHPRO10286324-PNIHsub-98569e\tQNIHPRO10286324\tPNIHsub\tclinical\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"nih_sub_sep.id.tsv\" --as subjects_new\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project -i hosp -i time -i quantities\\\n",
    "--match 'author: (cluster)-[]->(qnode), nih_author:(qnihauthor)-[]->(qnode), inv:(project)-[]->(qnihauthor), \\\n",
    "        item:(qnode)-[:P108]->(org_node), label:(org_node)-[]->(org_name),label:(qnode)-[]->(author_name), \\\n",
    "        coor:(org_node)-[:P625]->(organization_coor), hosp:(qnihauthor)-[]->(hosp_node),\\\n",
    "        label:(hosp_node)-[]->(hosp_name), coor:(hosp_node)-[:P625]->(hosp_coor),item:(paper)-[:P50]->(qnode),item:(paper)-[:P921]->(mai)'\\\n",
    "--opt 'time: (hosp_node)-[:P571]->(hosp_inc)' \\\n",
    "--opt 'time: (org_node)-[:P571]->(org_inc)' \\\n",
    "--opt 'quantities: (hosp_node)-[:P6801]->(count_beds)' \\\n",
    "--opt 'project: (project)-[:PNIHcost]->(cost)' \\\n",
    "--return 'cluster as cluster_id, org_name as org_name, sum(cost) as organization_award, \\\n",
    "        kgtk_geo_coords_lat(organization_coor) as org_lat, kgtk_geo_coords_long(organization_coor) as org_long,\\\n",
    "        kgtk_date_year(org_inc) as org_inc, \\\n",
    "        hosp_name as hosp_name, kgtk_geo_coords_lat(hosp_coor) as hosp_lat, kgtk_geo_coords_long(hosp_coor) as hosp_long,\\\n",
    "        kgtk_date_year(hosp_inc) as hosp_inc, count_beds as count_beds' \\\n",
    "-o bed_inception_1.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 04:48:16 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT graph_40_c8.\"node2\" \"_aLias.subject\", count(\"_aLias.subject\") \"_aLias.subject_count\", sum(graph_31_c12.\"node2\") \"_aLias.cost\"\n",
      "     FROM graph_18 AS graph_18_c10\n",
      "     INNER JOIN graph_18 AS graph_18_c5, graph_18 AS graph_18_c6, graph_19 AS graph_19_c1, graph_21 AS graph_21_c4, graph_22 AS graph_22_c11, graph_22 AS graph_22_c7, graph_30 AS graph_30_c2, graph_32 AS graph_32_c3, graph_37 AS graph_37_c9, graph_40 AS graph_40_c8\n",
      "     ON graph_19_c1.\"node2\" = graph_18_c6.\"node1\"\n",
      "        AND graph_19_c1.\"node2\" = graph_21_c4.\"node1\"\n",
      "        AND graph_19_c1.\"node2\" = graph_30_c2.\"node2\"\n",
      "        AND graph_21_c4.\"node2\" = graph_18_c5.\"node1\"\n",
      "        AND graph_21_c4.\"node2\" = graph_22_c7.\"node1\"\n",
      "        AND graph_30_c2.\"node1\" = graph_32_c3.\"node2\"\n",
      "        AND graph_30_c2.\"node1\" = graph_37_c9.\"node1\"\n",
      "        AND graph_32_c3.\"node1\" = graph_40_c8.\"node1\"\n",
      "        AND graph_37_c9.\"node2\" = graph_18_c10.\"node1\"\n",
      "        AND graph_37_c9.\"node2\" = graph_22_c11.\"node1\"\n",
      "        AND graph_21_c4.\"label\" = ?\n",
      "        AND graph_22_c11.\"label\" = ?\n",
      "        AND graph_22_c7.\"label\" = ?\n",
      "     LEFT JOIN graph_31 AS graph_31_c12\n",
      "     ON graph_32_c3.\"node1\" = graph_31_c12.\"node1\"\n",
      "        AND graph_31_c12.\"label\" = ?\n",
      "     LEFT JOIN graph_31 AS graph_31_c13\n",
      "     ON graph_32_c3.\"node1\" = graph_31_c13.\"node1\"\n",
      "        AND graph_31_c13.\"label\" = ?\n",
      "     LEFT JOIN graph_31 AS graph_31_c14\n",
      "     ON graph_32_c3.\"node1\" = graph_31_c14.\"node1\"\n",
      "        AND graph_31_c14.\"label\" = ?\n",
      "     GROUP BY \"_aLias.subject\"\n",
      "     ORDER BY \"_aLias.subject_count\" DESC\n",
      "  PARAS: ['P108', 'P625', 'P625', 'PNIHcost', 'PNIHtitle', 'PNIHsub']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i author_test_new -i nih_author -i inv -i item -i coor -i label -i project -i hosp -i time -i quantities -i subjects_new\\\n",
    "--match 'author: (cluster)-[]->(qnode), nih_author:(qnihauthor)-[]->(qnode), inv:(project)-[]->(qnihauthor), \\\n",
    "        item:(qnode)-[:P108]->(org_node), label:(org_node)-[]->(org_name),label:(qnode)-[]->(author_name), \\\n",
    "        coor:(org_node)-[:P625]->(organization_coor), subjects_new:(project)-[]->(subject), hosp:(qnihauthor)-[]->(hosp_node),\\\n",
    "        label:(hosp_node)-[]->(hosp_name), coor:(hosp_node)-[:P625]->(hosp_coor)'\\\n",
    "--opt 'project: (project)-[:PNIHcost]->(cost)' \\\n",
    "--opt 'project: (project)-[:PNIHtitle]->(project_title)' \\\n",
    "--opt 'project: (project)-[:PNIHsub]->(project_subjects)' \\\n",
    "--return 'subject as subject, \\\n",
    "            count(subject) as subject_count, sum(cost) as cost' \\\n",
    "--order-by \"subject_count desc\"\\\n",
    "-o subjects_total.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('file.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['count_beds'].fillna(value=df['count_beds'].mean(), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel('nih_0824.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster = []\n",
    "org_name = []\n",
    "org_node = []\n",
    "org_lat = []\n",
    "org_long = []\n",
    "hosp_name = []\n",
    "hosp_node = []\n",
    "hosp_lat =[]\n",
    "hosp_long= []\n",
    "\n",
    "subject1 = []\n",
    "subject2 = []\n",
    "subject3 = []\n",
    "subject4 = []\n",
    "subject5 = []\n",
    "\n",
    "subject1_count = []\n",
    "subject2_count = []\n",
    "subject3_count = []\n",
    "subject4_count = []\n",
    "subject5_count = []\n",
    "\n",
    "subject1_cost  = []\n",
    "subject2_cost  = []\n",
    "subject3_cost  = []\n",
    "subject4_cost  = []\n",
    "subject5_cost  = []\n",
    "\n",
    "org_inc = []\n",
    "hosp_inc = []\n",
    "\n",
    "hosp_bed = []\n",
    "\n",
    "last_cluster = ''\n",
    "last_org = ''\n",
    "\n",
    "for i in range(0, len(df)):\n",
    "    \n",
    "    if df.cluster_id[i] == last_cluster and df['org_name'][i] == last_org:\n",
    "        continue\n",
    "    else:\n",
    "        last_cluster = df.cluster_id[i]\n",
    "        last_org = df['org_name'][i]\n",
    "        last_cluster = df.cluster_id[i]\n",
    "        last_org = df['org_name'][i]\n",
    "        cluster.append(df['cluster_id'][i])\n",
    "        org_name.append(df['org_name'][i])\n",
    "        #org_node.append(df['org_node'][i])\n",
    "        org_lat.append(df['org_lat'][i])\n",
    "        org_long.append(df['org_long'][i])\n",
    "        hosp_name.append(df['hosp_name'][i])\n",
    "        #hosp_node.append(df['hosp_node'][i])\n",
    "        hosp_lat.append(df['hosp_lat'][i])\n",
    "        hosp_long.append(df['hosp_long'][i])\n",
    "        \n",
    "        \n",
    "        subject1.append(df.ms_name[i])\n",
    "        subject2.append(df.ms_name[i+1])\n",
    "        subject3.append(df.ms_name[i+2])\n",
    "        subject4.append(df.ms_name[i+3])\n",
    "        subject5.append(df.ms_name[i+4])\n",
    "        \n",
    "        subject1_count.append(df.subject_count[i])\n",
    "        subject2_count.append(df.subject_count[i+1])\n",
    "        subject3_count.append(df.subject_count[i+2])\n",
    "        subject4_count.append(df.subject_count[i+3])\n",
    "        subject5_count.append(df.subject_count[i+4])\n",
    "        \n",
    "        \n",
    "        #subject1_cost.append(df.cost[i])\n",
    "        #subject2_cost.append(df.cost[i+1])\n",
    "        #subject3_cost.append(df.cost[i+2])\n",
    "        #subject4_cost.append(df.cost[i+3])\n",
    "        #subject5_cost.append(df.cost[i+4])\n",
    "        \n",
    "        org_inc.append(df.org_inc[i])\n",
    "        hosp_inc.append(df.hosp_inc[i])\n",
    "        hosp_bed.append(df.count_beds[i])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [],
   "source": [
    "output = pd.DataFrame(list(zip(cluster, org_name,  org_lat, org_long, org_inc, hosp_name,  hosp_lat, hosp_long, hosp_inc, hosp_bed, subject1, \\\n",
    "subject2, subject3, subject4, subject5, subject1_count,subject2_count,subject3_count,subject4_count,subject5_count)), columns = ['cluster', 'org_name', 'org_lat', 'org_long', 'org_inc', 'hosp_name',  'hosp_lat', 'hosp_long', 'hosp_inc', 'hosp_bed', 'subject1', \\\n",
    "'subject2', 'subject3', 'subject4', 'subject5', 'count1', 'count2','count3','count4','count5'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "output['hosp_bed'].fillna(value=output['hosp_bed'].mean(), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>hosp_bed</th>\n",
       "      <th>subject1</th>\n",
       "      <th>subject2</th>\n",
       "      <th>subject3</th>\n",
       "      <th>subject4</th>\n",
       "      <th>subject5</th>\n",
       "      <th>count1</th>\n",
       "      <th>count2</th>\n",
       "      <th>count3</th>\n",
       "      <th>count4</th>\n",
       "      <th>count5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>632.616279</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>3570</td>\n",
       "      <td>3330</td>\n",
       "      <td>2028</td>\n",
       "      <td>2028</td>\n",
       "      <td>1620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_3_12_19_26_39_990</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>796.000000</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>3330</td>\n",
       "      <td>2028</td>\n",
       "      <td>2028</td>\n",
       "      <td>1620</td>\n",
       "      <td>1404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>632.616279</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>'Caenorhabditis elegans'@en</td>\n",
       "      <td>2028</td>\n",
       "      <td>2028</td>\n",
       "      <td>1620</td>\n",
       "      <td>1404</td>\n",
       "      <td>1248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_0_3_12_20_131_236_1292</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>632.616279</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>'Caenorhabditis elegans'@en</td>\n",
       "      <td>'transcriptome'@en</td>\n",
       "      <td>'macromolecule'@en</td>\n",
       "      <td>1620</td>\n",
       "      <td>1404</td>\n",
       "      <td>1248</td>\n",
       "      <td>1248</td>\n",
       "      <td>1092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>632.616279</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>'Caenorhabditis elegans'@en</td>\n",
       "      <td>'transcriptome'@en</td>\n",
       "      <td>'macromolecule'@en</td>\n",
       "      <td>'ribozyme'@en</td>\n",
       "      <td>1404</td>\n",
       "      <td>1248</td>\n",
       "      <td>1248</td>\n",
       "      <td>1092</td>\n",
       "      <td>945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'Human Induced Pluripotent Stem Cells'@en</td>\n",
       "      <td>'autism'@en</td>\n",
       "      <td>'transcriptome'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'cell biology'@en</td>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>cluster_0_1_1_12_29_57_94_232_1261</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>'Chemical modulation of chaperone-mediated aut...</td>\n",
       "      <td>'Fanconi syndrome'@en</td>\n",
       "      <td>'autoinhibition'@en</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'ovarian cancer'@en</td>\n",
       "      <td>'ATRX chromatin remodeler'@en</td>\n",
       "      <td>'ATRX, chromatin remodeler'@en</td>\n",
       "      <td>'Caenorhabditis elegans'@en</td>\n",
       "      <td>'DNA methylation'@en</td>\n",
       "      <td>30</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'ATRX chromatin remodeler'@en</td>\n",
       "      <td>'ATRX, chromatin remodeler'@en</td>\n",
       "      <td>'Caenorhabditis elegans'@en</td>\n",
       "      <td>'DNA methylation'@en</td>\n",
       "      <td>'DNA-binding E3 ubiquitin-protein ligase SNT2 ...</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1019</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'cell line'@en</td>\n",
       "      <td>'adenocarcinoma'@en</td>\n",
       "      <td>'antibody'@en</td>\n",
       "      <td>'biomarker'@en</td>\n",
       "      <td>'chemotherapy'@en</td>\n",
       "      <td>20</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1020 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 cluster  \\\n",
       "0     cluster_0_0_0_5_34_82_165_315_1172   \n",
       "1        cluster_0_0_0_3_12_19_26_39_990   \n",
       "2      cluster_0_1_1_12_27_47_75_129_446   \n",
       "3     cluster_0_0_0_3_12_20_131_236_1292   \n",
       "4      cluster_0_1_1_12_27_47_75_129_446   \n",
       "...                                  ...   \n",
       "1015    cluster_0_0_0_3_12_20_27_233_494   \n",
       "1016  cluster_0_1_1_12_29_57_94_232_1261   \n",
       "1017  cluster_0_1_1_12_27_75_124_219_527   \n",
       "1018    cluster_0_0_0_3_12_20_27_233_494   \n",
       "1019  cluster_0_1_1_12_27_75_124_219_527   \n",
       "\n",
       "                                      org_name    org_lat   org_long  org_inc  \\\n",
       "0                         'Yale University'@en  41.311111 -72.926667   1701.0   \n",
       "1                         'Yale University'@en  41.311111 -72.926667   1701.0   \n",
       "2                         'Yale University'@en  41.311111 -72.926667   1701.0   \n",
       "3                         'Yale University'@en  41.311111 -72.926667   1701.0   \n",
       "4                         'Yale University'@en  41.311111 -72.926667   1701.0   \n",
       "...                                        ...        ...        ...      ...   \n",
       "1015  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   1955.0   \n",
       "1016  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   1955.0   \n",
       "1017  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   1955.0   \n",
       "1018  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   1955.0   \n",
       "1019  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   1955.0   \n",
       "\n",
       "                         hosp_name   hosp_lat   hosp_long  hosp_inc  \\\n",
       "0      'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "1         'UCSF Medical Center'@en  37.763200 -122.458000    1907.0   \n",
       "2      'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "3      'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "4      'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "...                            ...        ...         ...       ...   \n",
       "1015  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "1016  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "1017  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "1018  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "1019  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "\n",
       "        hosp_bed                                   subject1  \\\n",
       "0     632.616279                       'phosphorylation'@en   \n",
       "1     796.000000                                'CRISPR'@en   \n",
       "2     632.616279                        'RNA sequencing'@en   \n",
       "3     632.616279                    'Toll-like receptor'@en   \n",
       "4     632.616279                            'statistics'@en   \n",
       "...          ...                                        ...   \n",
       "1015  225.000000  'Human Induced Pluripotent Stem Cells'@en   \n",
       "1016  225.000000  'BH3 interacting domain death agonist'@en   \n",
       "1017  225.000000                        'ovarian cancer'@en   \n",
       "1018  225.000000              'ATRX chromatin remodeler'@en   \n",
       "1019  225.000000                             'cell line'@en   \n",
       "\n",
       "                            subject2  \\\n",
       "0                        'CRISPR'@en   \n",
       "1                'RNA sequencing'@en   \n",
       "2                      'database'@en   \n",
       "3                    'statistics'@en   \n",
       "4        'Caenorhabditis elegans'@en   \n",
       "...                              ...   \n",
       "1015                     'autism'@en   \n",
       "1016                     'Bcl-xL'@en   \n",
       "1017   'ATRX chromatin remodeler'@en   \n",
       "1018  'ATRX, chromatin remodeler'@en   \n",
       "1019             'adenocarcinoma'@en   \n",
       "\n",
       "                                               subject3  \\\n",
       "0                                   'RNA sequencing'@en   \n",
       "1                                         'database'@en   \n",
       "2                               'Toll-like receptor'@en   \n",
       "3                           'Caenorhabditis elegans'@en   \n",
       "4                                    'transcriptome'@en   \n",
       "...                                                 ...   \n",
       "1015                                 'transcriptome'@en   \n",
       "1016  'Chemical modulation of chaperone-mediated aut...   \n",
       "1017                     'ATRX, chromatin remodeler'@en   \n",
       "1018                        'Caenorhabditis elegans'@en   \n",
       "1019                                      'antibody'@en   \n",
       "\n",
       "                         subject4  \\\n",
       "0                   'database'@en   \n",
       "1         'Toll-like receptor'@en   \n",
       "2                 'statistics'@en   \n",
       "3              'transcriptome'@en   \n",
       "4              'macromolecule'@en   \n",
       "...                           ...   \n",
       "1015                  'CRISPR'@en   \n",
       "1016        'Fanconi syndrome'@en   \n",
       "1017  'Caenorhabditis elegans'@en   \n",
       "1018         'DNA methylation'@en   \n",
       "1019               'biomarker'@en   \n",
       "\n",
       "                                               subject5  count1  count2  \\\n",
       "0                               'Toll-like receptor'@en    3570    3330   \n",
       "1                                       'statistics'@en    3330    2028   \n",
       "2                           'Caenorhabditis elegans'@en    2028    2028   \n",
       "3                                    'macromolecule'@en    1620    1404   \n",
       "4                                         'ribozyme'@en    1404    1248   \n",
       "...                                                 ...     ...     ...   \n",
       "1015                                  'cell biology'@en      66      66   \n",
       "1016                                'autoinhibition'@en      42      42   \n",
       "1017                               'DNA methylation'@en      30      22   \n",
       "1018  'DNA-binding E3 ubiquitin-protein ligase SNT2 ...      22      22   \n",
       "1019                                  'chemotherapy'@en      20      10   \n",
       "\n",
       "      count3  count4  count5  \n",
       "0       2028    2028    1620  \n",
       "1       2028    1620    1404  \n",
       "2       1620    1404    1248  \n",
       "3       1248    1248    1092  \n",
       "4       1248    1092     945  \n",
       "...      ...     ...     ...  \n",
       "1015      66      44      44  \n",
       "1016      42      42      42  \n",
       "1017      22      22      22  \n",
       "1018      22      22      22  \n",
       "1019      10      10      10  \n",
       "\n",
       "[1020 rows x 20 columns]"
      ]
     },
     "execution_count": 328,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "metadata": {},
   "outputs": [],
   "source": [
    "output.to_excel('top5_subject_0820.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "!$kypher -i item -i nih_author -i item -i label\\\n",
    "--match 'item:(paper)-[:P921]->(ms),\\\n",
    "        item:(paper)-[:P50]->(qnode),item:(paper)-[:P921]->(ms_node),\\\n",
    "        label:(ms_node)-[]->(ms_name)'\\\n",
    "--return 'paper,ms, ms_name'\\\n",
    "--limit 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 10:29:24 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT DISTINCT ? \"_aLias.id\", graph_30_c1.\"node2\" \"_aLias.node1\", ? \"_aLias.label\", graph_21_c2.\"node1\" \"_aLias.node2\"\n",
      "     FROM graph_21 AS graph_21_c2\n",
      "     INNER JOIN graph_30 AS graph_30_c1\n",
      "     ON graph_30_c1.\"node2\" = graph_21_c2.\"node2\"\n",
      "        AND graph_21_c2.\"label\" = ?\n",
      "  PARAS: ['', 'P50', 'P50']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i nih_author -i item -i label -i author\\\n",
    "--match 'author: ()-[]->(qnode),item:(paper)-[:P50]->(qnode)'\\\n",
    "--return 'distinct \"\" as id, qnode as node1, \"P50\" as label, paper as node2'\\\n",
    "-o paper.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kgtk add-id --id-style wikidata -i \"/Users/grantxie/Downloads/NIH/paper.tsv\" -o paper.id.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 10:31:34 sqlstore]: IMPORT graph directly into table graph_44 from /Users/grantxie/Downloads/NIH/paper.id.tsv ...\n",
      "[2021-08-20 10:31:35 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_44 AS graph_44_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "Q100287043-P50-Q27316717\tQ100287043\tP50\tQ27316717\n",
      "Q100287043-P50-Q28728777\tQ100287043\tP50\tQ28728777\n",
      "Q100287043-P50-Q33579346\tQ100287043\tP50\tQ33579346\n",
      "Q100287043-P50-Q34769869\tQ100287043\tP50\tQ34769869\n",
      "Q100287043-P50-Q36126966\tQ100287043\tP50\tQ36126966\n",
      "Q100287043-P50-Q36950067\tQ100287043\tP50\tQ36950067\n",
      "Q100287043-P50-Q38741540\tQ100287043\tP50\tQ38741540\n",
      "Q100287043-P50-Q38751767\tQ100287043\tP50\tQ38751767\n",
      "Q100287043-P50-Q44000882\tQ100287043\tP50\tQ44000882\n",
      "Q100287043-P50-Q54264722\tQ100287043\tP50\tQ54264722\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"paper.id.tsv\" --as paper_test_new\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 10:35:22 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_43 AS graph_43_c1\n",
      "     WHERE graph_43_c1.\"node2\" = ?\n",
      "  PARAS: ['Q27316717']\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n"
     ]
    }
   ],
   "source": [
    "\n",
    "!$kypher -i item -i nih_author -i item -i label -i test -i paper_test\\\n",
    "--match 'paper_test:()-[]->(:Q27316717)'\\\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 10:50:28 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT DISTINCT graph_18_c5.\"node2\", graph_18_c4.\"node2\", count(graph_21_c3.\"node2\")\n",
      "     FROM graph_18 AS graph_18_c4\n",
      "     INNER JOIN graph_18 AS graph_18_c5, graph_21 AS graph_21_c3, graph_44 AS graph_44_c2, graph_45 AS graph_45_c1\n",
      "     ON graph_21_c3.\"node2\" = graph_18_c4.\"node1\"\n",
      "        AND graph_44_c2.\"node2\" = graph_21_c3.\"node1\"\n",
      "        AND graph_45_c1.\"node2\" = graph_18_c5.\"node1\"\n",
      "        AND graph_45_c1.\"node2\" = graph_44_c2.\"node1\"\n",
      "        AND graph_21_c3.\"label\" = ?\n",
      "        AND graph_44_c2.\"label\" = ?\n",
      "     GROUP BY graph_18_c5.\"node2\", graph_18_c4.\"node2\"\n",
      "     ORDER BY count(graph_21_c3.\"node2\") DESC\n",
      "  PARAS: ['P921', 'P50']\n",
      "---------------------------------------------\n",
      "[2021-08-20 10:50:28 sqlstore]: CREATE INDEX on table graph_45 column node2 ...\n",
      "[2021-08-20 10:50:28 sqlstore]: ANALYZE INDEX on table graph_45 column node2 ...\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i nih_author -i item -i label -i test -i paper_test_new -i author_test_new\\\n",
    "--match 'author_test_new: ()-[]->(qnode), paper_test_new:(qnode)-[:P50]->(paper),item:(paper)-[:P921]->(ms_node),\\\n",
    "        label:(ms_node)-[]->(ms_name), label:(qnode)-[]->(name)'\\\n",
    "--return 'distinct name, ms_name, count(ms_node)'\\\n",
    "--order 'count(ms_node) desc'\\\n",
    "-o test4.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 10:47:18 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT DISTINCT ? \"_aLias.id\", graph_30_c1.\"node1\" \"_aLias.node1\", ? \"_aLias.label\", graph_30_c1.\"node2\" \"_aLias.node2\"\n",
      "     FROM graph_30 AS graph_30_c1\n",
      "  PARAS: ['', 'same']\n",
      "---------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i nih_author -i item -i label -i paper_test_new\\\n",
    "--match 'author: (a)-[]->(b)'\\\n",
    "--return 'distinct \"\" as id, a as node1, \"same\" as label, b as node2'\\\n",
    "-o author_kgtk_distinct.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$kgtk add-id --id-style wikidata -i \"/Users/grantxie/Downloads/NIH/author_kgtk_distinct.tsv\" -o author_kgtk_distinct.id.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 10:48:48 sqlstore]: DROP graph data table graph_45 from author_test_new\n",
      "[2021-08-20 10:48:48 sqlstore]: IMPORT graph directly into table graph_45 from /Users/grantxie/Downloads/NIH/author_kgtk_distinct.id.tsv ...\n",
      "[2021-08-20 10:48:48 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_45 AS graph_45_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPER1940486-same-Q6831539\tQNIHPER1940486\tsame\tQ6831539\n",
      "QNIHPER2414340-same-Q87796696\tQNIHPER2414340\tsame\tQ87796696\n",
      "QNIHPER11537130-same-Q89958961\tQNIHPER11537130\tsame\tQ89958961\n",
      "QNIHPER8820221-same-Q61467477\tQNIHPER8820221\tsame\tQ61467477\n",
      "QNIHPER8491201-same-Q46001993\tQNIHPER8491201\tsame\tQ46001993\n",
      "QNIHPER7882742-same-Q89866816\tQNIHPER7882742\tsame\tQ89866816\n",
      "QNIHPER8146532-same-Q89937644\tQNIHPER8146532\tsame\tQ89937644\n",
      "QNIHPER10499986-same-Q56809498\tQNIHPER10499986\tsame\tQ56809498\n",
      "QNIHPER2315018-same-Q87887672\tQNIHPER2315018\tsame\tQ87887672\n",
      "QNIHPER3065493-same-Q90437737\tQNIHPER3065493\tsame\tQ90437737\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"author_kgtk_distinct.id.tsv\" --as author_test_new\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 05:55:27 sqlstore]: IMPORT graph via csv.reader into table graph_41 from /Users/grantxie/test_kgtk.tsv ...\n",
      "[2021-08-20 05:55:27 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_41 AS graph_41_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\trow\tnode2;label\tgraph\n",
      "author0\tQNIHPER1940486\tlabel\tQ6831539\t2\tMICHAEL J GALE\tauthor\n",
      "author1\tQNIHPER2414340\tlabel\tQ87796696\t5\tPATRICK S. STAYTON\tauthor\n",
      "author2\tQNIHPER11537130\tlabel\tQ89958961\t8\tKENNETH NGURE\tauthor\n",
      "author3\tQNIHPER8820221\tlabel\tQ61467477\t10\tANITHA PASUPATHY\tauthor\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/test_kgtk.tsv\" --as test\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-20 05:55:52 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT DISTINCT graph_18_c5.\"node2\", graph_18_c4.\"node2\", count(graph_21_c3.\"node2\")\n",
      "     FROM graph_18 AS graph_18_c4\n",
      "     INNER JOIN graph_18 AS graph_18_c5, graph_21 AS graph_21_c2, graph_21 AS graph_21_c3, graph_41 AS graph_41_c1\n",
      "     ON graph_21_c2.\"node1\" = graph_21_c3.\"node1\"\n",
      "        AND graph_21_c3.\"node2\" = graph_18_c4.\"node1\"\n",
      "        AND graph_41_c1.\"node2\" = graph_18_c5.\"node1\"\n",
      "        AND graph_41_c1.\"node2\" = graph_21_c2.\"node2\"\n",
      "        AND graph_21_c2.\"label\" = ?\n",
      "        AND graph_21_c3.\"label\" = ?\n",
      "     GROUP BY graph_18_c5.\"node2\", graph_18_c4.\"node2\"\n",
      "     ORDER BY count(graph_21_c3.\"node2\") DESC\n",
      "  PARAS: ['P50', 'P921']\n",
      "---------------------------------------------\n",
      "[2021-08-20 05:55:52 sqlstore]: CREATE INDEX on table graph_41 column node2 ...\n",
      "[2021-08-20 05:55:52 sqlstore]: ANALYZE INDEX on table graph_41 column node2 ...\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i item -i nih_author -i item -i label -i test\\\n",
    "--match 'test: ()-[]->(qnode),item:(paper)-[:P50]->(qnode),item:(paper)-[:P921]->(ms_node),\\\n",
    "        label:(ms_node)-[]->(ms_name), label:(qnode)-[]->(name)'\\\n",
    "--return 'distinct name, ms_name, count(ms_node)'\\\n",
    "--order 'count(ms_node) desc'\\\n",
    "-o test1.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## file for tableau display"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-26 03:00:17 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT graph_19_c1.\"node1\" \"_aLias.cluster_id\", graph_18_c5.\"node2\" \"_aLias.org_name\", kgtk_geo_coords_lat(graph_22_c7.\"node2\") \"_aLias.org_lat\", kgtk_geo_coords_long(graph_22_c7.\"node2\") \"_aLias.org_long\", kgtk_date_year(graph_35_c18.\"node2\") \"_aLias.org_inc\", graph_18_c9.\"node2\" \"_aLias.hosp_name\", kgtk_geo_coords_lat(graph_22_c10.\"node2\") \"_aLias.hosp_lat\", kgtk_geo_coords_long(graph_22_c10.\"node2\") \"_aLias.hosp_long\", kgtk_date_year(graph_35_c17.\"node2\") \"_aLias.hosp_inc\", graph_38_c19.\"node2\" \"_aLias.count_beds\", graph_18_c14.\"node2\" \"_aLias.ms_name\", count(graph_21_c13.\"node2\") \"_aLias.subject_count\", sum(graph_49_c20.\"node2\") \"_aLias.cost\", graph_49_c16.\"node2\" \"_aLias.start\"\n",
      "     FROM graph_18 AS graph_18_c14\n",
      "     INNER JOIN graph_18 AS graph_18_c15, graph_18 AS graph_18_c5, graph_18 AS graph_18_c6, graph_18 AS graph_18_c9, graph_19 AS graph_19_c1, graph_21 AS graph_21_c13, graph_21 AS graph_21_c4, graph_22 AS graph_22_c10, graph_22 AS graph_22_c7, graph_30 AS graph_30_c2, graph_32 AS graph_32_c3, graph_37 AS graph_37_c8, graph_44 AS graph_44_c12, graph_45 AS graph_45_c11, graph_49 AS graph_49_c16\n",
      "     ON graph_19_c1.\"node2\" = graph_18_c15.\"node1\"\n",
      "        AND graph_19_c1.\"node2\" = graph_18_c6.\"node1\"\n",
      "        AND graph_19_c1.\"node2\" = graph_21_c4.\"node1\"\n",
      "        AND graph_19_c1.\"node2\" = graph_30_c2.\"node2\"\n",
      "        AND graph_19_c1.\"node2\" = graph_44_c12.\"node1\"\n",
      "        AND graph_19_c1.\"node2\" = graph_45_c11.\"node2\"\n",
      "        AND graph_21_c13.\"node2\" = graph_18_c14.\"node1\"\n",
      "        AND graph_21_c4.\"node2\" = graph_18_c5.\"node1\"\n",
      "        AND graph_21_c4.\"node2\" = graph_22_c7.\"node1\"\n",
      "        AND graph_30_c2.\"node1\" = graph_32_c3.\"node2\"\n",
      "        AND graph_30_c2.\"node1\" = graph_37_c8.\"node1\"\n",
      "        AND graph_32_c3.\"node1\" = graph_49_c16.\"node1\"\n",
      "        AND graph_37_c8.\"node2\" = graph_18_c9.\"node1\"\n",
      "        AND graph_37_c8.\"node2\" = graph_22_c10.\"node1\"\n",
      "        AND graph_44_c12.\"node2\" = graph_21_c13.\"node1\"\n",
      "        AND graph_21_c13.\"label\" = ?\n",
      "        AND graph_21_c4.\"label\" = ?\n",
      "        AND graph_22_c10.\"label\" = ?\n",
      "        AND graph_22_c7.\"label\" = ?\n",
      "        AND graph_44_c12.\"label\" = ?\n",
      "        AND graph_49_c16.\"label\" = ?\n",
      "     LEFT JOIN graph_35 AS graph_35_c17\n",
      "     ON graph_37_c8.\"node2\" = graph_35_c17.\"node1\"\n",
      "        AND graph_35_c17.\"label\" = ?\n",
      "     LEFT JOIN graph_35 AS graph_35_c18\n",
      "     ON graph_21_c4.\"node2\" = graph_35_c18.\"node1\"\n",
      "        AND graph_35_c18.\"label\" = ?\n",
      "     LEFT JOIN graph_38 AS graph_38_c19\n",
      "     ON graph_37_c8.\"node2\" = graph_38_c19.\"node1\"\n",
      "        AND graph_38_c19.\"label\" = ?\n",
      "     LEFT JOIN graph_49 AS graph_49_c20\n",
      "     ON graph_32_c3.\"node1\" = graph_49_c20.\"node1\"\n",
      "        AND graph_49_c20.\"label\" = ?\n",
      "     GROUP BY \"_aLias.cluster_id\", \"_aLias.org_name\", \"_aLias.org_lat\", \"_aLias.org_long\", \"_aLias.org_inc\", \"_aLias.hosp_name\", \"_aLias.hosp_lat\", \"_aLias.hosp_long\", \"_aLias.hosp_inc\", \"_aLias.count_beds\", \"_aLias.ms_name\"\n",
      "     ORDER BY \"_aLias.org_name\" DESC, \"_aLias.subject_count\" DESC\n",
      "  PARAS: ['P921', 'P108', 'P625', 'P625', 'P50', 'PNIHstart', 'P571', 'P571', 'P6801', 'PNIHcost']\n",
      "---------------------------------------------\n",
      "[2021-08-26 03:00:17 sqlstore]: CREATE INDEX on table graph_49 column node1 ...\n",
      "[2021-08-26 03:00:19 sqlstore]: ANALYZE INDEX on table graph_49 column node1 ...\n",
      "[2021-08-26 03:00:19 sqlstore]: CREATE INDEX on table graph_49 column label ...\n",
      "[2021-08-26 03:00:20 sqlstore]: ANALYZE INDEX on table graph_49 column label ...\n"
     ]
    }
   ],
   "source": [
    "!$kypher -i author -i nih_author -i inv -i item -i coor -i label -i project -i hosp -i time -i quantities -i author_test_new -i paper_test_new -i nih0825_new\\\n",
    "--match 'author: (cluster)-[]->(qnode), nih_author:(qnihauthor)-[]->(qnode), inv:(project)-[]->(qnihauthor), \\\n",
    "        item:(qnode)-[:P108]->(org_node), label:(org_node)-[]->(org_name),label:(qnode)-[]->(author_name), \\\n",
    "        coor:(org_node)-[:P625]->(organization_coor), hosp:(qnihauthor)-[]->(hosp_node),\\\n",
    "        label:(hosp_node)-[]->(hosp_name), coor:(hosp_node)-[:P625]->(hosp_coor),author_test_new: ()-[]->(qnode), paper_test_new:(qnode)-[:P50]->(paper),item:(paper)-[:P921]->(ms_node),\\\n",
    "        label:(ms_node)-[]->(ms_name), label:(qnode)-[]->(name), nih0825_new:(project)-[:PNIHstart]->(start1)'\\\n",
    "--opt 'time: (hosp_node)-[:P571]->(hosp_inc)' \\\n",
    "--opt 'time: (org_node)-[:P571]->(org_inc)' \\\n",
    "--opt 'quantities: (hosp_node)-[:P6801]->(count_beds)' \\\n",
    "--opt 'nih0825_new: (project)-[:PNIHcost]->(cost)' \\\n",
    "--return 'cluster as cluster_id, org_name as org_name, \\\n",
    "        kgtk_geo_coords_lat(organization_coor) as org_lat, kgtk_geo_coords_long(organization_coor) as org_long,\\\n",
    "        kgtk_date_year(org_inc) as org_inc, \\\n",
    "        hosp_name as hosp_name, kgtk_geo_coords_lat(hosp_coor) as hosp_lat, kgtk_geo_coords_long(hosp_coor) as hosp_long,\\\n",
    "        kgtk_date_year(hosp_inc) as hosp_inc, count_beds as count_beds, ms_name as ms_name, count(ms_node) as subject_count, sum(cost) as cost, start1 as start' \\\n",
    "--order 'org_name desc, subject_count desc'\\\n",
    "-o final.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2021-08-26 02:59:59 sqlstore]: IMPORT graph directly into table graph_49 from /Users/grantxie/Downloads/kgtk-files-nih-V5.0/nih_project_0825.id.tsv ...\n",
      "[2021-08-26 03:00:11 query]: SQL Translation:\n",
      "---------------------------------------------\n",
      "  SELECT *\n",
      "     FROM graph_49 AS graph_49_c1\n",
      "     LIMIT ?\n",
      "  PARAS: [10]\n",
      "---------------------------------------------\n",
      "id\tnode1\tlabel\tnode2\n",
      "QNIHPRO10286324-PNIHtitle-cfa7f1\tQNIHPRO10286324\tPNIHtitle\t\"CD38-TARGETED IMMUNOPET OF MYELOMA: PHASE 2 TRIAL OF CLINICAL APPLICATIONS\"\n",
      "QNIHPRO10286324-PNIHstart-f85450\tQNIHPRO10286324\tPNIHstart\t\"02/01/2021\"\n",
      "QNIHPRO10286324-PNIHorg-QNIHORG205901\tQNIHPRO10286324\tPNIHorg\tQNIHORG205901\n",
      "QNIHPRO10286324-PNIHcost-452920\tQNIHPRO10286324\tPNIHcost\t598926\n",
      "QNIHPRO10286324-PNIHsub-730793\tQNIHPRO10286324\tPNIHsub\t\"Antibodies; base; Biopsy; blind; Blood Tests; Bone marrow biopsy; burden of illness; cancer imaging; Cells; Clinical; clinical application; clinical care; Clinical Trials; clinically significant; Combined Modality Therapy; design; Detection; Detection of Minimal Residual Disease; Disease; dosimetry; early phase trial; experience; first-in-human; fluorodeoxyglucose positron emission tomography; Funding Opportunities; Future; Goals; Histologic; Human; human imaging; Image; imaging agent; imaging modality; Immunoglobulins; ImmunoPET; individual patient; innovation; Investigation; Laboratories; Measurement; Measures; Membrane Proteins; Memorial Sloan-Kettering Cancer Center; Methods; Monitor; Monoclonal Antibodies; Multiple Myeloma; multiple myeloma M Protein; novel; outcome forecast; Patient Care; Patient imaging; Patient Monitoring; Patients; PET/CT scan; Phase; Phase I Clinical Trials; phase I trial; Phase II Clinical Trials; phase II trial; Plasma Cell Neoplasm; Plasma Cells; Positron-Emission Tomography; preclinical imaging; predicting response; Prognostic Marker; Proteins; Radiochemistry; Radioimmunotherapy; Radiolabeled; Research Personnel; Residual Tumors; response; Safety; Selection for Treatments; Serum; Site; Surface; targeted imaging; theranostics; Therapeutic Agents; Tracer; Translations; treatment response; tumor; Tumor Burden; United States; uptake; virtual; Visualization; Work;\"\n",
      "QNIHPRO10258167-PNIHtitle-90b590\tQNIHPRO10258167\tPNIHtitle\t\"Mechanism of selective packaging of primer tRNALys3 by HIV-1\"\n",
      "QNIHPRO10258167-PNIHstart-f6f53e\tQNIHPRO10258167\tPNIHstart\t\"03/22/2021\"\n",
      "QNIHPRO10258167-PNIHorg-QNIHORG9420201\tQNIHPRO10258167\tPNIHorg\tQNIHORG9420201\n",
      "QNIHPRO10258167-PNIHcost-686611\tQNIHPRO10258167\tPNIHcost\t217898\n",
      "QNIHPRO10258167-PNIHsub-dfb3bb\tQNIHPRO10258167\tPNIHsub\t\"Acquired Immunodeficiency Syndrome; Address; Affinity; Amino Acyl-tRNA Synthetases; Aminoacylation; Anisotropy; Anti-Retroviral Agents; base; Binding; Biochemical; Biological Assay; Biology; biophysical techniques; C-terminal; Calorimetry; Capsid Proteins; Catalytic Domain; Cells; Charge; Chemicals; comparative; Complex; Coupled; crosslink; Cryoelectron Microscopy; Crystallization; Data; Dependence; design; Development; Dissection; drug development; Drug resistance; Electron Microscopy; Electrophoretic Mobility Shift Assay; EMSA; Enzymes; Event; experimental study; flexibility; Fluorescence; Fluorescence Anisotropy; Goals; HIV; HIV-1; Individual; Infection; insight; Kinetics; Knowledge; Life Cycle Stages; light scattering; Lysine-Specific tRNA; Lysine-tRNA Ligase; Measures; Modeling; Molecular; Molecular Sieve Chromatography; mutant; Negative Staining; new therapeutic target; Non-Small-Cell Lung Carcinoma; novel; pandemic disease; particle; Pharmaceutical Preparations; Phosphorylation; Play; pol Gene Products; Process; Public Health; Recording of previous events; recruit; Reporting; Research; resistance mutation; Resolution; Reverse Transcription; Role; screening; SP1 gene; Structure; success; Testing; therapeutic development; Titrations; Transcriptional Regulation; Transfer RNA; Vaccines; Validation; Viral Genome; Viral Reverse Transcription; Virion; Work; X-Ray Crystallography;\"\n"
     ]
    }
   ],
   "source": [
    "!$kypher \\\n",
    "-i \"/Users/grantxie/Downloads/kgtk-files-nih-V5.0/nih_project_0825.id.tsv\" --as nih0825_new\\\n",
    "--limit 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = pd.read_csv('final.tsv', sep ='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>count_beds</th>\n",
       "      <th>ms_name</th>\n",
       "      <th>subject_count</th>\n",
       "      <th>cost</th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>5100</td>\n",
       "      <td>3583893760</td>\n",
       "      <td>06/01/2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_3_12_19_26_39_990</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>796.0</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>4662</td>\n",
       "      <td>3295411734</td>\n",
       "      <td>06/01/2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>3198</td>\n",
       "      <td>3663737922</td>\n",
       "      <td>04/01/2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>3198</td>\n",
       "      <td>3663737922</td>\n",
       "      <td>04/01/2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>2214</td>\n",
       "      <td>2536433946</td>\n",
       "      <td>04/01/2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15238</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>'multiple drug resistance'@en</td>\n",
       "      <td>16</td>\n",
       "      <td>5784638</td>\n",
       "      <td>07/01/2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15239</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>16</td>\n",
       "      <td>5784638</td>\n",
       "      <td>07/01/2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15240</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>'substance dependence'@en</td>\n",
       "      <td>16</td>\n",
       "      <td>5784638</td>\n",
       "      <td>07/01/2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15241</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>'team science'@en</td>\n",
       "      <td>16</td>\n",
       "      <td>5784638</td>\n",
       "      <td>07/01/2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15242</th>\n",
       "      <td>cluster_0_1_1_12_27_75_124_219_527</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>'xenograft'@en</td>\n",
       "      <td>16</td>\n",
       "      <td>5784638</td>\n",
       "      <td>07/01/2017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15243 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               cluster_id  \\\n",
       "0      cluster_0_0_0_5_34_82_165_315_1172   \n",
       "1         cluster_0_0_0_3_12_19_26_39_990   \n",
       "2       cluster_0_1_1_12_27_47_75_129_446   \n",
       "3       cluster_0_1_1_12_27_47_75_129_446   \n",
       "4       cluster_0_1_1_12_27_47_75_129_446   \n",
       "...                                   ...   \n",
       "15238  cluster_0_1_1_12_27_75_124_219_527   \n",
       "15239  cluster_0_1_1_12_27_75_124_219_527   \n",
       "15240  cluster_0_1_1_12_27_75_124_219_527   \n",
       "15241  cluster_0_1_1_12_27_75_124_219_527   \n",
       "15242  cluster_0_1_1_12_27_75_124_219_527   \n",
       "\n",
       "                                       org_name    org_lat   org_long  \\\n",
       "0                          'Yale University'@en  41.311111 -72.926667   \n",
       "1                          'Yale University'@en  41.311111 -72.926667   \n",
       "2                          'Yale University'@en  41.311111 -72.926667   \n",
       "3                          'Yale University'@en  41.311111 -72.926667   \n",
       "4                          'Yale University'@en  41.311111 -72.926667   \n",
       "...                                         ...        ...        ...   \n",
       "15238  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   \n",
       "15239  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   \n",
       "15240  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   \n",
       "15241  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   \n",
       "15242  'Albert Einstein College of Medicine'@en  40.850852 -73.844949   \n",
       "\n",
       "       org_inc                     hosp_name   hosp_lat   hosp_long  hosp_inc  \\\n",
       "0       1701.0   'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "1       1701.0      'UCSF Medical Center'@en  37.763200 -122.458000    1907.0   \n",
       "2       1701.0   'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "3       1701.0   'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "4       1701.0   'Smilow Cancer Hospital'@en  41.304278  -72.935676    2009.0   \n",
       "...        ...                           ...        ...         ...       ...   \n",
       "15238   1955.0  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "15239   1955.0  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "15240   1955.0  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "15241   1955.0  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "15242   1955.0  'Calvary Hospital, Bronx'@en  40.847986  -73.843853    1899.0   \n",
       "\n",
       "       count_beds                        ms_name  subject_count        cost  \\\n",
       "0             NaN           'phosphorylation'@en           5100  3583893760   \n",
       "1           796.0                    'CRISPR'@en           4662  3295411734   \n",
       "2             NaN            'RNA sequencing'@en           3198  3663737922   \n",
       "3             NaN                  'database'@en           3198  3663737922   \n",
       "4             NaN                'statistics'@en           2214  2536433946   \n",
       "...           ...                            ...            ...         ...   \n",
       "15238       225.0  'multiple drug resistance'@en             16     5784638   \n",
       "15239       225.0           'phosphorylation'@en             16     5784638   \n",
       "15240       225.0      'substance dependence'@en             16     5784638   \n",
       "15241       225.0              'team science'@en             16     5784638   \n",
       "15242       225.0                 'xenograft'@en             16     5784638   \n",
       "\n",
       "            start  \n",
       "0      06/01/2021  \n",
       "1      06/01/2021  \n",
       "2      04/01/2021  \n",
       "3      04/01/2021  \n",
       "4      04/01/2021  \n",
       "...           ...  \n",
       "15238  07/01/2017  \n",
       "15239  07/01/2017  \n",
       "15240  07/01/2017  \n",
       "15241  07/01/2017  \n",
       "15242  07/01/2017  \n",
       "\n",
       "[15243 rows x 14 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(df)):\n",
    "    df.start[i] = df.start[i][len(df.start[i])-4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['count_beds'].fillna(value=df['count_beds'].mean(), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel('nih_0826_new2.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(df)):\n",
    "    df.start[i] = df.start[i][len(df.start[i])-4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>count_beds</th>\n",
       "      <th>ms_name</th>\n",
       "      <th>subject_count</th>\n",
       "      <th>cost</th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>850</td>\n",
       "      <td>554722240.0</td>\n",
       "      <td>2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_3_12_20_131_236_1292</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>360</td>\n",
       "      <td>121399080.0</td>\n",
       "      <td>2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_0_3_12_19_26_39_990</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>796.0</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>222</td>\n",
       "      <td>460578516.0</td>\n",
       "      <td>2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>180</td>\n",
       "      <td>117470592.0</td>\n",
       "      <td>2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>160</td>\n",
       "      <td>104418304.0</td>\n",
       "      <td>2021</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           cluster_id              org_name    org_lat  \\\n",
       "0  cluster_0_0_0_5_34_82_165_315_1172  'Yale University'@en  41.311111   \n",
       "1  cluster_0_0_0_3_12_20_131_236_1292  'Yale University'@en  41.311111   \n",
       "2     cluster_0_0_0_3_12_19_26_39_990  'Yale University'@en  41.311111   \n",
       "3  cluster_0_0_0_5_34_82_165_315_1172  'Yale University'@en  41.311111   \n",
       "4  cluster_0_0_0_5_34_82_165_315_1172  'Yale University'@en  41.311111   \n",
       "\n",
       "    org_long  org_inc                    hosp_name   hosp_lat   hosp_long  \\\n",
       "0 -72.926667   1701.0  'Smilow Cancer Hospital'@en  41.304278  -72.935676   \n",
       "1 -72.926667   1701.0  'Smilow Cancer Hospital'@en  41.304278  -72.935676   \n",
       "2 -72.926667   1701.0     'UCSF Medical Center'@en  37.763200 -122.458000   \n",
       "3 -72.926667   1701.0  'Smilow Cancer Hospital'@en  41.304278  -72.935676   \n",
       "4 -72.926667   1701.0  'Smilow Cancer Hospital'@en  41.304278  -72.935676   \n",
       "\n",
       "   hosp_inc  count_beds                  ms_name  subject_count         cost  \\\n",
       "0    2009.0         NaN     'phosphorylation'@en            850  554722240.0   \n",
       "1    2009.0         NaN  'Toll-like receptor'@en            360  121399080.0   \n",
       "2    1907.0       796.0              'CRISPR'@en            222  460578516.0   \n",
       "3    2009.0         NaN     'corpus striatum'@en            180  117470592.0   \n",
       "4    2009.0         NaN            'dopamine'@en            160  104418304.0   \n",
       "\n",
       "  start  \n",
       "0  2021  \n",
       "1  2021  \n",
       "2  2021  \n",
       "3  2021  \n",
       "4  2021  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "df['count_beds'].fillna(value=df['count_beds'].mean(), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_excel('nih_0825.xlsx', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Adding internet coverage by county to map background\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('countycode.tsv', sep = '\\t')\n",
    "df.to_excel('FIPS.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['FIPS'] = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov = pd.read_csv('/Users/grantxie/Downloads/broadband_data_2020October.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov = cov[18:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov = cov.reset_index(drop = True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data is to be used only for analysis purposes related to broadband mapping</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AL</td>\n",
       "      <td>1001</td>\n",
       "      <td>Autauga County</td>\n",
       "      <td>0.8057</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL</td>\n",
       "      <td>1003</td>\n",
       "      <td>Baldwin County</td>\n",
       "      <td>0.8362</td>\n",
       "      <td>0.452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AL</td>\n",
       "      <td>1005</td>\n",
       "      <td>Barbour County</td>\n",
       "      <td>0.6891</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AL</td>\n",
       "      <td>1007</td>\n",
       "      <td>Bibb County</td>\n",
       "      <td>0.3368</td>\n",
       "      <td>0.136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AL</td>\n",
       "      <td>1009</td>\n",
       "      <td>Blount County</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Data is to be used only for analysis purposes related to broadband mapping  \\\n",
       "0                                                 AL                           \n",
       "1                                                 AL                           \n",
       "2                                                 AL                           \n",
       "3                                                 AL                           \n",
       "4                                                 AL                           \n",
       "\n",
       "  Unnamed: 1      Unnamed: 2 Unnamed: 3 Unnamed: 4  \n",
       "0       1001  Autauga County     0.8057      0.391  \n",
       "1       1003  Baldwin County     0.8362      0.452  \n",
       "2       1005  Barbour County     0.6891      0.324  \n",
       "3       1007     Bibb County     0.3368      0.136  \n",
       "4       1009   Blount County      0.758      0.199  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data is to be used only for analysis purposes related to broadband mapping</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AL</td>\n",
       "      <td>1001</td>\n",
       "      <td>Autauga County</td>\n",
       "      <td>0.8057</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL</td>\n",
       "      <td>1003</td>\n",
       "      <td>Baldwin County</td>\n",
       "      <td>0.8362</td>\n",
       "      <td>0.452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AL</td>\n",
       "      <td>1005</td>\n",
       "      <td>Barbour County</td>\n",
       "      <td>0.6891</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AL</td>\n",
       "      <td>1007</td>\n",
       "      <td>Bibb County</td>\n",
       "      <td>0.3368</td>\n",
       "      <td>0.136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AL</td>\n",
       "      <td>1009</td>\n",
       "      <td>Blount County</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3137</th>\n",
       "      <td>WY</td>\n",
       "      <td>56037</td>\n",
       "      <td>Sweetwater County</td>\n",
       "      <td>0.9422</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3138</th>\n",
       "      <td>WY</td>\n",
       "      <td>56039</td>\n",
       "      <td>Teton County</td>\n",
       "      <td>0.9508</td>\n",
       "      <td>0.623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3139</th>\n",
       "      <td>WY</td>\n",
       "      <td>56041</td>\n",
       "      <td>Uinta County</td>\n",
       "      <td>0.9963</td>\n",
       "      <td>0.431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3140</th>\n",
       "      <td>WY</td>\n",
       "      <td>56043</td>\n",
       "      <td>Washakie County</td>\n",
       "      <td>0.8903</td>\n",
       "      <td>0.571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3141</th>\n",
       "      <td>WY</td>\n",
       "      <td>56045</td>\n",
       "      <td>Weston County</td>\n",
       "      <td>0.8937</td>\n",
       "      <td>0.152</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3142 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Data is to be used only for analysis purposes related to broadband mapping  \\\n",
       "0                                                    AL                           \n",
       "1                                                    AL                           \n",
       "2                                                    AL                           \n",
       "3                                                    AL                           \n",
       "4                                                    AL                           \n",
       "...                                                 ...                           \n",
       "3137                                                 WY                           \n",
       "3138                                                 WY                           \n",
       "3139                                                 WY                           \n",
       "3140                                                 WY                           \n",
       "3141                                                 WY                           \n",
       "\n",
       "     Unnamed: 1         Unnamed: 2 Unnamed: 3 Unnamed: 4  \n",
       "0          1001     Autauga County     0.8057      0.391  \n",
       "1          1003     Baldwin County     0.8362      0.452  \n",
       "2          1005     Barbour County     0.6891      0.324  \n",
       "3          1007        Bibb County     0.3368      0.136  \n",
       "4          1009      Blount County      0.758      0.199  \n",
       "...         ...                ...        ...        ...  \n",
       "3137      56037  Sweetwater County     0.9422        0.4  \n",
       "3138      56039       Teton County     0.9508      0.623  \n",
       "3139      56041       Uinta County     0.9963      0.431  \n",
       "3140      56043    Washakie County     0.8903      0.571  \n",
       "3141      56045      Weston County     0.8937      0.152  \n",
       "\n",
       "[3142 rows x 5 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(0, len(cov)):\n",
    "    if len(cov['Unnamed: 1'][i]) == 4:\n",
    "        cov['Unnamed: 1'][i] = '0' + str(cov['Unnamed: 1'][i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/opt/anaconda3/envs/kgtk-env/lib/python3.7/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(df)):\n",
    "    df['n1'][i] = cov['Unnamed: 3'][i]\n",
    "    df['n2'][i] = cov['Unnamed: 4'][i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>county</th>\n",
       "      <th>qnode</th>\n",
       "      <th>code</th>\n",
       "      <th>FIPS</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Autauga County</td>\n",
       "      <td>Q156168</td>\n",
       "      <td>1001</td>\n",
       "      <td>01001</td>\n",
       "      <td>0.8057</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Baldwin County</td>\n",
       "      <td>Q156163</td>\n",
       "      <td>1003</td>\n",
       "      <td>01003</td>\n",
       "      <td>0.8362</td>\n",
       "      <td>0.452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barbour County</td>\n",
       "      <td>Q109437</td>\n",
       "      <td>1005</td>\n",
       "      <td>01005</td>\n",
       "      <td>0.6891</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bibb County</td>\n",
       "      <td>Q461204</td>\n",
       "      <td>1007</td>\n",
       "      <td>01007</td>\n",
       "      <td>0.3368</td>\n",
       "      <td>0.136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Blount County</td>\n",
       "      <td>Q111250</td>\n",
       "      <td>1009</td>\n",
       "      <td>01009</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3124</th>\n",
       "      <td>Sweetwater County</td>\n",
       "      <td>Q484194</td>\n",
       "      <td>56037</td>\n",
       "      <td>56037</td>\n",
       "      <td>0.8503</td>\n",
       "      <td>0.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3125</th>\n",
       "      <td>Teton County</td>\n",
       "      <td>Q488912</td>\n",
       "      <td>56039</td>\n",
       "      <td>56039</td>\n",
       "      <td>0.7941</td>\n",
       "      <td>0.373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3126</th>\n",
       "      <td>Uinta County</td>\n",
       "      <td>Q483973</td>\n",
       "      <td>56041</td>\n",
       "      <td>56041</td>\n",
       "      <td>0.9972</td>\n",
       "      <td>0.198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3127</th>\n",
       "      <td>Washakie County</td>\n",
       "      <td>Q112846</td>\n",
       "      <td>56043</td>\n",
       "      <td>56043</td>\n",
       "      <td>0.8722</td>\n",
       "      <td>0.381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3128</th>\n",
       "      <td>Weston County</td>\n",
       "      <td>Q115413</td>\n",
       "      <td>56045</td>\n",
       "      <td>56045</td>\n",
       "      <td>0.9217</td>\n",
       "      <td>0.456</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3129 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 county    qnode   code   FIPS      n1     n2\n",
       "0        Autauga County  Q156168   1001  01001  0.8057  0.391\n",
       "1        Baldwin County  Q156163   1003  01003  0.8362  0.452\n",
       "2        Barbour County  Q109437   1005  01005  0.6891  0.324\n",
       "3           Bibb County  Q461204   1007  01007  0.3368  0.136\n",
       "4         Blount County  Q111250   1009  01009   0.758  0.199\n",
       "...                 ...      ...    ...    ...     ...    ...\n",
       "3124  Sweetwater County  Q484194  56037  56037  0.8503   0.21\n",
       "3125       Teton County  Q488912  56039  56039  0.7941  0.373\n",
       "3126       Uinta County  Q483973  56041  56041  0.9972  0.198\n",
       "3127    Washakie County  Q112846  56043  56043  0.8722  0.381\n",
       "3128      Weston County  Q115413  56045  56045  0.9217  0.456\n",
       "\n",
       "[3129 rows x 6 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_node</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_node</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>hosp_bed</th>\n",
       "      <th>subject1</th>\n",
       "      <th>subject2</th>\n",
       "      <th>subject3</th>\n",
       "      <th>subject4</th>\n",
       "      <th>subject5</th>\n",
       "      <th>FIPS</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>01001</td>\n",
       "      <td>0.8057</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_3_12_20_131_236_1292</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>01003</td>\n",
       "      <td>0.8362</td>\n",
       "      <td>0.452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_0_3_12_19_26_39_990</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>Q7864122</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>796.000000</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>01005</td>\n",
       "      <td>0.6891</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Alzheimer\\'s disease'@en</td>\n",
       "      <td>01007</td>\n",
       "      <td>0.3368</td>\n",
       "      <td>0.136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Alzheimer\\'s disease'@en</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>'inflammation'@en</td>\n",
       "      <td>01009</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'apoptotic process'@en</td>\n",
       "      <td>'molecular chaperones'@en</td>\n",
       "      <td>'schizophrenia'@en</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>17045</td>\n",
       "      <td>0.9221</td>\n",
       "      <td>0.173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>cluster_0_1_1_12_29_57_94_232_1261</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'apoptotic process'@en</td>\n",
       "      <td>'molecular chaperones'@en</td>\n",
       "      <td>'schizophrenia'@en</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>17047</td>\n",
       "      <td>0.7801</td>\n",
       "      <td>0.188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'schizophrenia'@en</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>'Chemical modulation of chaperone-mediated aut...</td>\n",
       "      <td>'Fanconi syndrome'@en</td>\n",
       "      <td>17049</td>\n",
       "      <td>0.9528</td>\n",
       "      <td>0.586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>cluster_0_1_1_12_29_57_94_232_1261</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>'Chemical modulation of chaperone-mediated aut...</td>\n",
       "      <td>'Fanconi syndrome'@en</td>\n",
       "      <td>'autoinhibition'@en</td>\n",
       "      <td>17051</td>\n",
       "      <td>0.5575</td>\n",
       "      <td>0.198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'Human Induced Pluripotent Stem Cells'@en</td>\n",
       "      <td>'autism'@en</td>\n",
       "      <td>'transcriptome'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'cell biology'@en</td>\n",
       "      <td>17053</td>\n",
       "      <td>0.8982</td>\n",
       "      <td>0.349</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>622 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                cluster  \\\n",
       "0    cluster_0_0_0_5_34_82_165_315_1172   \n",
       "1    cluster_0_0_0_3_12_20_131_236_1292   \n",
       "2       cluster_0_0_0_3_12_19_26_39_990   \n",
       "3    cluster_0_0_0_5_34_82_165_315_1172   \n",
       "4     cluster_0_1_1_12_27_47_75_129_446   \n",
       "..                                  ...   \n",
       "617    cluster_0_0_0_3_12_20_27_233_494   \n",
       "618  cluster_0_1_1_12_29_57_94_232_1261   \n",
       "619    cluster_0_0_0_3_12_20_27_233_494   \n",
       "620  cluster_0_1_1_12_29_57_94_232_1261   \n",
       "621    cluster_0_0_0_3_12_20_27_233_494   \n",
       "\n",
       "                                     org_name  org_node    org_lat   org_long  \\\n",
       "0                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "1                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "2                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "3                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "4                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "..                                        ...       ...        ...        ...   \n",
       "617  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "618  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "619  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "620  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "621  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "\n",
       "     org_inc                     hosp_name  hosp_node   hosp_lat   hosp_long  \\\n",
       "0     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "1     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "2     1701.0      'UCSF Medical Center'@en   Q7864122  37.763200 -122.458000   \n",
       "3     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "4     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "..       ...                           ...        ...        ...         ...   \n",
       "617   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "618   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "619   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "620   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "621   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "\n",
       "     hosp_inc    hosp_bed                                   subject1  \\\n",
       "0      2009.0  657.884393                       'phosphorylation'@en   \n",
       "1      2009.0  657.884393                    'Toll-like receptor'@en   \n",
       "2      1907.0  796.000000                                'CRISPR'@en   \n",
       "3      2009.0  657.884393                       'corpus striatum'@en   \n",
       "4      2009.0  657.884393                        'RNA sequencing'@en   \n",
       "..        ...         ...                                        ...   \n",
       "617    1899.0  225.000000                        'RNA sequencing'@en   \n",
       "618    1899.0  225.000000                     'apoptotic process'@en   \n",
       "619    1899.0  225.000000                         'schizophrenia'@en   \n",
       "620    1899.0  225.000000  'BH3 interacting domain death agonist'@en   \n",
       "621    1899.0  225.000000  'Human Induced Pluripotent Stem Cells'@en   \n",
       "\n",
       "                                      subject2  \\\n",
       "0                      'Toll-like receptor'@en   \n",
       "1                                  'CRISPR'@en   \n",
       "2                         'corpus striatum'@en   \n",
       "3                                'dopamine'@en   \n",
       "4                                'database'@en   \n",
       "..                                         ...   \n",
       "617                     'apoptotic process'@en   \n",
       "618                  'molecular chaperones'@en   \n",
       "619  'BH3 interacting domain death agonist'@en   \n",
       "620                                'Bcl-xL'@en   \n",
       "621                                'autism'@en   \n",
       "\n",
       "                                              subject3  \\\n",
       "0                                          'CRISPR'@en   \n",
       "1                                 'corpus striatum'@en   \n",
       "2                                        'dopamine'@en   \n",
       "3                                  'RNA sequencing'@en   \n",
       "4                            'Alzheimer\\'s disease'@en   \n",
       "..                                                 ...   \n",
       "617                          'molecular chaperones'@en   \n",
       "618                                 'schizophrenia'@en   \n",
       "619                                        'Bcl-xL'@en   \n",
       "620  'Chemical modulation of chaperone-mediated aut...   \n",
       "621                                 'transcriptome'@en   \n",
       "\n",
       "                                              subject4  \\\n",
       "0                                 'corpus striatum'@en   \n",
       "1                                        'dopamine'@en   \n",
       "2                                  'RNA sequencing'@en   \n",
       "3                                        'database'@en   \n",
       "4                                      'statistics'@en   \n",
       "..                                                 ...   \n",
       "617                                 'schizophrenia'@en   \n",
       "618          'BH3 interacting domain death agonist'@en   \n",
       "619  'Chemical modulation of chaperone-mediated aut...   \n",
       "620                              'Fanconi syndrome'@en   \n",
       "621                                        'CRISPR'@en   \n",
       "\n",
       "                                      subject5   FIPS      n1     n2  \n",
       "0                                'dopamine'@en  01001  0.8057  0.391  \n",
       "1                          'RNA sequencing'@en  01003  0.8362  0.452  \n",
       "2                                'database'@en  01005  0.6891  0.324  \n",
       "3                    'Alzheimer\\'s disease'@en  01007  0.3368  0.136  \n",
       "4                            'inflammation'@en  01009   0.758  0.199  \n",
       "..                                         ...    ...     ...    ...  \n",
       "617  'BH3 interacting domain death agonist'@en  17045  0.9221  0.173  \n",
       "618                                'Bcl-xL'@en  17047  0.7801  0.188  \n",
       "619                      'Fanconi syndrome'@en  17049  0.9528  0.586  \n",
       "620                        'autoinhibition'@en  17051  0.5575  0.198  \n",
       "621                          'cell biology'@en  17053  0.8982  0.349  \n",
       "\n",
       "[622 rows x 20 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "row = pd.DataFrame([[np.nan] * len(out.columns)], columns=out.columns)\n",
    "out = out.append(row, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(0, 3000):\n",
    "    out = out.append(row, ignore_index=True)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "out = pd.read_excel('/Users/grantxie/Downloads/NIH/top5_subject.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "out['FIPS'] = ''\n",
    "out['n1'] = 0.0\n",
    "out['n2'] = 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "out['county'] = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:1: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_node</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_node</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>...</th>\n",
       "      <th>subject5_count</th>\n",
       "      <th>subject1_cost</th>\n",
       "      <th>subject2_cost</th>\n",
       "      <th>subject3_cost</th>\n",
       "      <th>subject4_cost</th>\n",
       "      <th>subject5_cost</th>\n",
       "      <th>FIPS</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "      <th>county</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_1_3_11_25_80_153_279_1164</td>\n",
       "      <td>'University of California, Los Angeles'@en</td>\n",
       "      <td>Q174710</td>\n",
       "      <td>34.072222</td>\n",
       "      <td>-118.444167</td>\n",
       "      <td>1919.0</td>\n",
       "      <td>'Spaulding Hospital'@en</td>\n",
       "      <td>Q30270420</td>\n",
       "      <td>42.375748</td>\n",
       "      <td>-71.107052</td>\n",
       "      <td>...</td>\n",
       "      <td>24200</td>\n",
       "      <td>1.938386e+11</td>\n",
       "      <td>1.057301e+11</td>\n",
       "      <td>8.268639e+10</td>\n",
       "      <td>1.626617e+10</td>\n",
       "      <td>1.491066e+10</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_1_3_11_25_80_153_279_1164</td>\n",
       "      <td>'Harvard University'@en</td>\n",
       "      <td>Q13371</td>\n",
       "      <td>42.374444</td>\n",
       "      <td>-71.116944</td>\n",
       "      <td>1636.0</td>\n",
       "      <td>'Spaulding Hospital'@en</td>\n",
       "      <td>Q30270420</td>\n",
       "      <td>42.375748</td>\n",
       "      <td>-71.107052</td>\n",
       "      <td>...</td>\n",
       "      <td>66024</td>\n",
       "      <td>1.938386e+11</td>\n",
       "      <td>1.057301e+11</td>\n",
       "      <td>1.803773e+12</td>\n",
       "      <td>8.268639e+10</td>\n",
       "      <td>7.284469e+11</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_2_2_11_1_15_90_574</td>\n",
       "      <td>'University of Texas MD Anderson Cancer Center...</td>\n",
       "      <td>Q1525831</td>\n",
       "      <td>29.707800</td>\n",
       "      <td>-95.397500</td>\n",
       "      <td>1941.0</td>\n",
       "      <td>'University of Texas MD Anderson Cancer Center...</td>\n",
       "      <td>Q1525831</td>\n",
       "      <td>29.707800</td>\n",
       "      <td>-95.397500</td>\n",
       "      <td>...</td>\n",
       "      <td>28998</td>\n",
       "      <td>1.482875e+11</td>\n",
       "      <td>7.544451e+10</td>\n",
       "      <td>7.544451e+10</td>\n",
       "      <td>6.763990e+10</td>\n",
       "      <td>2.341381e+10</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_1_1_12_27_75_144_358_1672</td>\n",
       "      <td>'Harvard University'@en</td>\n",
       "      <td>Q13371</td>\n",
       "      <td>42.374444</td>\n",
       "      <td>-71.116944</td>\n",
       "      <td>1636.0</td>\n",
       "      <td>'Spaulding Hospital'@en</td>\n",
       "      <td>Q30270420</td>\n",
       "      <td>42.375748</td>\n",
       "      <td>-71.107052</td>\n",
       "      <td>...</td>\n",
       "      <td>44016</td>\n",
       "      <td>1.803773e+12</td>\n",
       "      <td>8.268639e+10</td>\n",
       "      <td>7.284469e+11</td>\n",
       "      <td>5.203192e+11</td>\n",
       "      <td>4.856313e+11</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_1_4_7_21_36_48_78_451</td>\n",
       "      <td>'University of Pittsburgh Graduate School of P...</td>\n",
       "      <td>Q7896126</td>\n",
       "      <td>40.442600</td>\n",
       "      <td>-79.958200</td>\n",
       "      <td>1948.0</td>\n",
       "      <td>'Center for Emergency Medicine of Western Penn...</td>\n",
       "      <td>Q5059612</td>\n",
       "      <td>40.439200</td>\n",
       "      <td>-79.958900</td>\n",
       "      <td>...</td>\n",
       "      <td>64240</td>\n",
       "      <td>6.762670e+10</td>\n",
       "      <td>4.830478e+10</td>\n",
       "      <td>3.355911e+10</td>\n",
       "      <td>2.898287e+10</td>\n",
       "      <td>2.796593e+10</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>cluster_0_1_3_9_28_49_65_104_870</td>\n",
       "      <td>'University of Washington'@en</td>\n",
       "      <td>Q219563</td>\n",
       "      <td>47.654167</td>\n",
       "      <td>-122.308056</td>\n",
       "      <td>1861.0</td>\n",
       "      <td>'University of Washington Medical Center'@en</td>\n",
       "      <td>Q7896576</td>\n",
       "      <td>47.648923</td>\n",
       "      <td>-122.306595</td>\n",
       "      <td>...</td>\n",
       "      <td>90</td>\n",
       "      <td>3.228399e+07</td>\n",
       "      <td>3.228399e+07</td>\n",
       "      <td>3.228399e+07</td>\n",
       "      <td>3.228399e+07</td>\n",
       "      <td>3.228399e+07</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>cluster_0_1_3_11_25_81_158_385_1488</td>\n",
       "      <td>'Harvard University'@en</td>\n",
       "      <td>Q13371</td>\n",
       "      <td>42.374444</td>\n",
       "      <td>-71.116944</td>\n",
       "      <td>1636.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>Q7864122</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>...</td>\n",
       "      <td>65</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>cluster_0_1_3_11_25_81_158_385_1488</td>\n",
       "      <td>'University of California, San Francisco'@en</td>\n",
       "      <td>Q1061104</td>\n",
       "      <td>37.763319</td>\n",
       "      <td>-122.458539</td>\n",
       "      <td>1873.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>Q7864122</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>...</td>\n",
       "      <td>65</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td>5.368876e+07</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>cluster_0_0_2_1_19_31_45_398_1684</td>\n",
       "      <td>'University of California, San Diego'@en</td>\n",
       "      <td>Q622664</td>\n",
       "      <td>32.881000</td>\n",
       "      <td>-117.238000</td>\n",
       "      <td>1960.0</td>\n",
       "      <td>'Jacobs Medical Center'@en</td>\n",
       "      <td>Q38250501</td>\n",
       "      <td>32.877703</td>\n",
       "      <td>-117.226499</td>\n",
       "      <td>...</td>\n",
       "      <td>314600</td>\n",
       "      <td>8.909118e+06</td>\n",
       "      <td>8.909118e+06</td>\n",
       "      <td>8.909118e+06</td>\n",
       "      <td>8.909118e+06</td>\n",
       "      <td>1.938386e+11</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>cluster_0_0_0_8_10_15_20_26_511</td>\n",
       "      <td>'Stanford University'@en</td>\n",
       "      <td>Q41506</td>\n",
       "      <td>37.428229</td>\n",
       "      <td>-122.168858</td>\n",
       "      <td>1891.0</td>\n",
       "      <td>'Stanford University Medical Center'@en</td>\n",
       "      <td>Q7598810</td>\n",
       "      <td>37.434000</td>\n",
       "      <td>-122.175000</td>\n",
       "      <td>...</td>\n",
       "      <td>59</td>\n",
       "      <td>5.428472e+06</td>\n",
       "      <td>5.428472e+06</td>\n",
       "      <td>5.428472e+06</td>\n",
       "      <td>5.428472e+06</td>\n",
       "      <td>5.428472e+06</td>\n",
       "      <td></td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>622 rows Ã— 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 cluster  \\\n",
       "0    cluster_0_1_3_11_25_80_153_279_1164   \n",
       "1    cluster_0_1_3_11_25_80_153_279_1164   \n",
       "2         cluster_0_0_2_2_11_1_15_90_574   \n",
       "3    cluster_0_1_1_12_27_75_144_358_1672   \n",
       "4        cluster_0_1_4_7_21_36_48_78_451   \n",
       "..                                   ...   \n",
       "617     cluster_0_1_3_9_28_49_65_104_870   \n",
       "618  cluster_0_1_3_11_25_81_158_385_1488   \n",
       "619  cluster_0_1_3_11_25_81_158_385_1488   \n",
       "620    cluster_0_0_2_1_19_31_45_398_1684   \n",
       "621      cluster_0_0_0_8_10_15_20_26_511   \n",
       "\n",
       "                                              org_name  org_node    org_lat  \\\n",
       "0           'University of California, Los Angeles'@en   Q174710  34.072222   \n",
       "1                              'Harvard University'@en    Q13371  42.374444   \n",
       "2    'University of Texas MD Anderson Cancer Center...  Q1525831  29.707800   \n",
       "3                              'Harvard University'@en    Q13371  42.374444   \n",
       "4    'University of Pittsburgh Graduate School of P...  Q7896126  40.442600   \n",
       "..                                                 ...       ...        ...   \n",
       "617                      'University of Washington'@en   Q219563  47.654167   \n",
       "618                            'Harvard University'@en    Q13371  42.374444   \n",
       "619       'University of California, San Francisco'@en  Q1061104  37.763319   \n",
       "620           'University of California, San Diego'@en   Q622664  32.881000   \n",
       "621                           'Stanford University'@en    Q41506  37.428229   \n",
       "\n",
       "       org_long  org_inc                                          hosp_name  \\\n",
       "0   -118.444167   1919.0                            'Spaulding Hospital'@en   \n",
       "1    -71.116944   1636.0                            'Spaulding Hospital'@en   \n",
       "2    -95.397500   1941.0  'University of Texas MD Anderson Cancer Center...   \n",
       "3    -71.116944   1636.0                            'Spaulding Hospital'@en   \n",
       "4    -79.958200   1948.0  'Center for Emergency Medicine of Western Penn...   \n",
       "..          ...      ...                                                ...   \n",
       "617 -122.308056   1861.0       'University of Washington Medical Center'@en   \n",
       "618  -71.116944   1636.0                           'UCSF Medical Center'@en   \n",
       "619 -122.458539   1873.0                           'UCSF Medical Center'@en   \n",
       "620 -117.238000   1960.0                         'Jacobs Medical Center'@en   \n",
       "621 -122.168858   1891.0            'Stanford University Medical Center'@en   \n",
       "\n",
       "     hosp_node   hosp_lat   hosp_long  ...  subject5_count  subject1_cost  \\\n",
       "0    Q30270420  42.375748  -71.107052  ...           24200   1.938386e+11   \n",
       "1    Q30270420  42.375748  -71.107052  ...           66024   1.938386e+11   \n",
       "2     Q1525831  29.707800  -95.397500  ...           28998   1.482875e+11   \n",
       "3    Q30270420  42.375748  -71.107052  ...           44016   1.803773e+12   \n",
       "4     Q5059612  40.439200  -79.958900  ...           64240   6.762670e+10   \n",
       "..         ...        ...         ...  ...             ...            ...   \n",
       "617   Q7896576  47.648923 -122.306595  ...              90   3.228399e+07   \n",
       "618   Q7864122  37.763200 -122.458000  ...              65   5.368876e+07   \n",
       "619   Q7864122  37.763200 -122.458000  ...              65   5.368876e+07   \n",
       "620  Q38250501  32.877703 -117.226499  ...          314600   8.909118e+06   \n",
       "621   Q7598810  37.434000 -122.175000  ...              59   5.428472e+06   \n",
       "\n",
       "    subject2_cost subject3_cost subject4_cost subject5_cost FIPS   n1   n2  \\\n",
       "0    1.057301e+11  8.268639e+10  1.626617e+10  1.491066e+10       0.0  0.0   \n",
       "1    1.057301e+11  1.803773e+12  8.268639e+10  7.284469e+11       0.0  0.0   \n",
       "2    7.544451e+10  7.544451e+10  6.763990e+10  2.341381e+10       0.0  0.0   \n",
       "3    8.268639e+10  7.284469e+11  5.203192e+11  4.856313e+11       0.0  0.0   \n",
       "4    4.830478e+10  3.355911e+10  2.898287e+10  2.796593e+10       0.0  0.0   \n",
       "..            ...           ...           ...           ...  ...  ...  ...   \n",
       "617  3.228399e+07  3.228399e+07  3.228399e+07  3.228399e+07       0.0  0.0   \n",
       "618  5.368876e+07  5.368876e+07  5.368876e+07  5.368876e+07       0.0  0.0   \n",
       "619  5.368876e+07  5.368876e+07  5.368876e+07  5.368876e+07       0.0  0.0   \n",
       "620  8.909118e+06  8.909118e+06  8.909118e+06  1.938386e+11       0.0  0.0   \n",
       "621  5.428472e+06  5.428472e+06  5.428472e+06  5.428472e+06       0.0  0.0   \n",
       "\n",
       "     county  \n",
       "0            \n",
       "1            \n",
       "2            \n",
       "3            \n",
       "4            \n",
       "..      ...  \n",
       "617          \n",
       "618          \n",
       "619          \n",
       "620          \n",
       "621          \n",
       "\n",
       "[622 rows x 31 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out.append(pd.Series(), ignore_index=True)\n",
    "out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"\n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(df)):\n",
    "    out['FIPS'][i] = df['FIPS'][i]\n",
    "    out['n1'][i] =df['n1'][i]\n",
    "    out['n2'][i] = df['n2'][i]\n",
    "    out['county'][i] = df['county'][i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_node</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_node</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>hosp_bed</th>\n",
       "      <th>subject1</th>\n",
       "      <th>subject2</th>\n",
       "      <th>subject3</th>\n",
       "      <th>subject4</th>\n",
       "      <th>subject5</th>\n",
       "      <th>FIPS</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>01001</td>\n",
       "      <td>0.8057</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_3_12_20_131_236_1292</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>01003</td>\n",
       "      <td>0.8362</td>\n",
       "      <td>0.452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_0_3_12_19_26_39_990</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>Q7864122</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>796.000000</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>01005</td>\n",
       "      <td>0.6891</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Alzheimer\\'s disease'@en</td>\n",
       "      <td>01007</td>\n",
       "      <td>0.3368</td>\n",
       "      <td>0.136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Alzheimer\\'s disease'@en</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>'inflammation'@en</td>\n",
       "      <td>01009</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3619</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3620</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3621</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3622</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3623 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 cluster              org_name org_node  \\\n",
       "0     cluster_0_0_0_5_34_82_165_315_1172  'Yale University'@en   Q49112   \n",
       "1     cluster_0_0_0_3_12_20_131_236_1292  'Yale University'@en   Q49112   \n",
       "2        cluster_0_0_0_3_12_19_26_39_990  'Yale University'@en   Q49112   \n",
       "3     cluster_0_0_0_5_34_82_165_315_1172  'Yale University'@en   Q49112   \n",
       "4      cluster_0_1_1_12_27_47_75_129_446  'Yale University'@en   Q49112   \n",
       "...                                  ...                   ...      ...   \n",
       "3618                                 NaN                   NaN      NaN   \n",
       "3619                                 NaN                   NaN      NaN   \n",
       "3620                                 NaN                   NaN      NaN   \n",
       "3621                                 NaN                   NaN      NaN   \n",
       "3622                                 NaN                   NaN      NaN   \n",
       "\n",
       "        org_lat   org_long  org_inc                    hosp_name  hosp_node  \\\n",
       "0     41.311111 -72.926667   1701.0  'Smilow Cancer Hospital'@en  Q50037095   \n",
       "1     41.311111 -72.926667   1701.0  'Smilow Cancer Hospital'@en  Q50037095   \n",
       "2     41.311111 -72.926667   1701.0     'UCSF Medical Center'@en   Q7864122   \n",
       "3     41.311111 -72.926667   1701.0  'Smilow Cancer Hospital'@en  Q50037095   \n",
       "4     41.311111 -72.926667   1701.0  'Smilow Cancer Hospital'@en  Q50037095   \n",
       "...         ...        ...      ...                          ...        ...   \n",
       "3618        NaN        NaN      NaN                          NaN        NaN   \n",
       "3619        NaN        NaN      NaN                          NaN        NaN   \n",
       "3620        NaN        NaN      NaN                          NaN        NaN   \n",
       "3621        NaN        NaN      NaN                          NaN        NaN   \n",
       "3622        NaN        NaN      NaN                          NaN        NaN   \n",
       "\n",
       "       hosp_lat   hosp_long  hosp_inc    hosp_bed                 subject1  \\\n",
       "0     41.304278  -72.935676    2009.0  657.884393     'phosphorylation'@en   \n",
       "1     41.304278  -72.935676    2009.0  657.884393  'Toll-like receptor'@en   \n",
       "2     37.763200 -122.458000    1907.0  796.000000              'CRISPR'@en   \n",
       "3     41.304278  -72.935676    2009.0  657.884393     'corpus striatum'@en   \n",
       "4     41.304278  -72.935676    2009.0  657.884393      'RNA sequencing'@en   \n",
       "...         ...         ...       ...         ...                      ...   \n",
       "3618        NaN         NaN       NaN         NaN                      NaN   \n",
       "3619        NaN         NaN       NaN         NaN                      NaN   \n",
       "3620        NaN         NaN       NaN         NaN                      NaN   \n",
       "3621        NaN         NaN       NaN         NaN                      NaN   \n",
       "3622        NaN         NaN       NaN         NaN                      NaN   \n",
       "\n",
       "                     subject2                   subject3  \\\n",
       "0     'Toll-like receptor'@en                'CRISPR'@en   \n",
       "1                 'CRISPR'@en       'corpus striatum'@en   \n",
       "2        'corpus striatum'@en              'dopamine'@en   \n",
       "3               'dopamine'@en        'RNA sequencing'@en   \n",
       "4               'database'@en  'Alzheimer\\'s disease'@en   \n",
       "...                       ...                        ...   \n",
       "3618                      NaN                        NaN   \n",
       "3619                      NaN                        NaN   \n",
       "3620                      NaN                        NaN   \n",
       "3621                      NaN                        NaN   \n",
       "3622                      NaN                        NaN   \n",
       "\n",
       "                  subject4                   subject5   FIPS      n1     n2  \n",
       "0     'corpus striatum'@en              'dopamine'@en  01001  0.8057  0.391  \n",
       "1            'dopamine'@en        'RNA sequencing'@en  01003  0.8362  0.452  \n",
       "2      'RNA sequencing'@en              'database'@en  01005  0.6891  0.324  \n",
       "3            'database'@en  'Alzheimer\\'s disease'@en  01007  0.3368  0.136  \n",
       "4          'statistics'@en          'inflammation'@en  01009   0.758  0.199  \n",
       "...                    ...                        ...    ...     ...    ...  \n",
       "3618                   NaN                        NaN    NaN     NaN    NaN  \n",
       "3619                   NaN                        NaN    NaN     NaN    NaN  \n",
       "3620                   NaN                        NaN    NaN     NaN    NaN  \n",
       "3621                   NaN                        NaN    NaN     NaN    NaN  \n",
       "3622                   NaN                        NaN    NaN     NaN    NaN  \n",
       "\n",
       "[3623 rows x 20 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/grantxie/.local/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "for i in range(0, len(cov)):\n",
    "    out.FIPS[i] = cov['Unnamed: 1'][i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster</th>\n",
       "      <th>org_name</th>\n",
       "      <th>org_node</th>\n",
       "      <th>org_lat</th>\n",
       "      <th>org_long</th>\n",
       "      <th>org_inc</th>\n",
       "      <th>hosp_name</th>\n",
       "      <th>hosp_node</th>\n",
       "      <th>hosp_lat</th>\n",
       "      <th>hosp_long</th>\n",
       "      <th>hosp_inc</th>\n",
       "      <th>hosp_bed</th>\n",
       "      <th>subject1</th>\n",
       "      <th>subject2</th>\n",
       "      <th>subject3</th>\n",
       "      <th>subject4</th>\n",
       "      <th>subject5</th>\n",
       "      <th>FIPS</th>\n",
       "      <th>n1</th>\n",
       "      <th>n2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'phosphorylation'@en</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>01001</td>\n",
       "      <td>0.8057</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cluster_0_0_0_3_12_20_131_236_1292</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'Toll-like receptor'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>01003</td>\n",
       "      <td>0.8362</td>\n",
       "      <td>0.452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cluster_0_0_0_3_12_19_26_39_990</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'UCSF Medical Center'@en</td>\n",
       "      <td>Q7864122</td>\n",
       "      <td>37.763200</td>\n",
       "      <td>-122.458000</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>796.000000</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>01005</td>\n",
       "      <td>0.6891</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cluster_0_0_0_5_34_82_165_315_1172</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'corpus striatum'@en</td>\n",
       "      <td>'dopamine'@en</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Alzheimer\\'s disease'@en</td>\n",
       "      <td>01007</td>\n",
       "      <td>0.3368</td>\n",
       "      <td>0.136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cluster_0_1_1_12_27_47_75_129_446</td>\n",
       "      <td>'Yale University'@en</td>\n",
       "      <td>Q49112</td>\n",
       "      <td>41.311111</td>\n",
       "      <td>-72.926667</td>\n",
       "      <td>1701.0</td>\n",
       "      <td>'Smilow Cancer Hospital'@en</td>\n",
       "      <td>Q50037095</td>\n",
       "      <td>41.304278</td>\n",
       "      <td>-72.935676</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>657.884393</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'database'@en</td>\n",
       "      <td>'Alzheimer\\'s disease'@en</td>\n",
       "      <td>'statistics'@en</td>\n",
       "      <td>'inflammation'@en</td>\n",
       "      <td>01009</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'RNA sequencing'@en</td>\n",
       "      <td>'apoptotic process'@en</td>\n",
       "      <td>'molecular chaperones'@en</td>\n",
       "      <td>'schizophrenia'@en</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>17045</td>\n",
       "      <td>0.9221</td>\n",
       "      <td>0.173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>cluster_0_1_1_12_29_57_94_232_1261</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'apoptotic process'@en</td>\n",
       "      <td>'molecular chaperones'@en</td>\n",
       "      <td>'schizophrenia'@en</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>17047</td>\n",
       "      <td>0.7801</td>\n",
       "      <td>0.188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'schizophrenia'@en</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>'Chemical modulation of chaperone-mediated aut...</td>\n",
       "      <td>'Fanconi syndrome'@en</td>\n",
       "      <td>17049</td>\n",
       "      <td>0.9528</td>\n",
       "      <td>0.586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>cluster_0_1_1_12_29_57_94_232_1261</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'BH3 interacting domain death agonist'@en</td>\n",
       "      <td>'Bcl-xL'@en</td>\n",
       "      <td>'Chemical modulation of chaperone-mediated aut...</td>\n",
       "      <td>'Fanconi syndrome'@en</td>\n",
       "      <td>'autoinhibition'@en</td>\n",
       "      <td>17051</td>\n",
       "      <td>0.5575</td>\n",
       "      <td>0.198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>cluster_0_0_0_3_12_20_27_233_494</td>\n",
       "      <td>'Albert Einstein College of Medicine'@en</td>\n",
       "      <td>Q2030894</td>\n",
       "      <td>40.850852</td>\n",
       "      <td>-73.844949</td>\n",
       "      <td>1955.0</td>\n",
       "      <td>'Calvary Hospital, Bronx'@en</td>\n",
       "      <td>Q5024190</td>\n",
       "      <td>40.847986</td>\n",
       "      <td>-73.843853</td>\n",
       "      <td>1899.0</td>\n",
       "      <td>225.000000</td>\n",
       "      <td>'Human Induced Pluripotent Stem Cells'@en</td>\n",
       "      <td>'autism'@en</td>\n",
       "      <td>'transcriptome'@en</td>\n",
       "      <td>'CRISPR'@en</td>\n",
       "      <td>'cell biology'@en</td>\n",
       "      <td>17053</td>\n",
       "      <td>0.8982</td>\n",
       "      <td>0.349</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>622 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                cluster  \\\n",
       "0    cluster_0_0_0_5_34_82_165_315_1172   \n",
       "1    cluster_0_0_0_3_12_20_131_236_1292   \n",
       "2       cluster_0_0_0_3_12_19_26_39_990   \n",
       "3    cluster_0_0_0_5_34_82_165_315_1172   \n",
       "4     cluster_0_1_1_12_27_47_75_129_446   \n",
       "..                                  ...   \n",
       "617    cluster_0_0_0_3_12_20_27_233_494   \n",
       "618  cluster_0_1_1_12_29_57_94_232_1261   \n",
       "619    cluster_0_0_0_3_12_20_27_233_494   \n",
       "620  cluster_0_1_1_12_29_57_94_232_1261   \n",
       "621    cluster_0_0_0_3_12_20_27_233_494   \n",
       "\n",
       "                                     org_name  org_node    org_lat   org_long  \\\n",
       "0                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "1                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "2                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "3                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "4                        'Yale University'@en    Q49112  41.311111 -72.926667   \n",
       "..                                        ...       ...        ...        ...   \n",
       "617  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "618  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "619  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "620  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "621  'Albert Einstein College of Medicine'@en  Q2030894  40.850852 -73.844949   \n",
       "\n",
       "     org_inc                     hosp_name  hosp_node   hosp_lat   hosp_long  \\\n",
       "0     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "1     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "2     1701.0      'UCSF Medical Center'@en   Q7864122  37.763200 -122.458000   \n",
       "3     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "4     1701.0   'Smilow Cancer Hospital'@en  Q50037095  41.304278  -72.935676   \n",
       "..       ...                           ...        ...        ...         ...   \n",
       "617   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "618   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "619   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "620   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "621   1955.0  'Calvary Hospital, Bronx'@en   Q5024190  40.847986  -73.843853   \n",
       "\n",
       "     hosp_inc    hosp_bed                                   subject1  \\\n",
       "0      2009.0  657.884393                       'phosphorylation'@en   \n",
       "1      2009.0  657.884393                    'Toll-like receptor'@en   \n",
       "2      1907.0  796.000000                                'CRISPR'@en   \n",
       "3      2009.0  657.884393                       'corpus striatum'@en   \n",
       "4      2009.0  657.884393                        'RNA sequencing'@en   \n",
       "..        ...         ...                                        ...   \n",
       "617    1899.0  225.000000                        'RNA sequencing'@en   \n",
       "618    1899.0  225.000000                     'apoptotic process'@en   \n",
       "619    1899.0  225.000000                         'schizophrenia'@en   \n",
       "620    1899.0  225.000000  'BH3 interacting domain death agonist'@en   \n",
       "621    1899.0  225.000000  'Human Induced Pluripotent Stem Cells'@en   \n",
       "\n",
       "                                      subject2  \\\n",
       "0                      'Toll-like receptor'@en   \n",
       "1                                  'CRISPR'@en   \n",
       "2                         'corpus striatum'@en   \n",
       "3                                'dopamine'@en   \n",
       "4                                'database'@en   \n",
       "..                                         ...   \n",
       "617                     'apoptotic process'@en   \n",
       "618                  'molecular chaperones'@en   \n",
       "619  'BH3 interacting domain death agonist'@en   \n",
       "620                                'Bcl-xL'@en   \n",
       "621                                'autism'@en   \n",
       "\n",
       "                                              subject3  \\\n",
       "0                                          'CRISPR'@en   \n",
       "1                                 'corpus striatum'@en   \n",
       "2                                        'dopamine'@en   \n",
       "3                                  'RNA sequencing'@en   \n",
       "4                            'Alzheimer\\'s disease'@en   \n",
       "..                                                 ...   \n",
       "617                          'molecular chaperones'@en   \n",
       "618                                 'schizophrenia'@en   \n",
       "619                                        'Bcl-xL'@en   \n",
       "620  'Chemical modulation of chaperone-mediated aut...   \n",
       "621                                 'transcriptome'@en   \n",
       "\n",
       "                                              subject4  \\\n",
       "0                                 'corpus striatum'@en   \n",
       "1                                        'dopamine'@en   \n",
       "2                                  'RNA sequencing'@en   \n",
       "3                                        'database'@en   \n",
       "4                                      'statistics'@en   \n",
       "..                                                 ...   \n",
       "617                                 'schizophrenia'@en   \n",
       "618          'BH3 interacting domain death agonist'@en   \n",
       "619  'Chemical modulation of chaperone-mediated aut...   \n",
       "620                              'Fanconi syndrome'@en   \n",
       "621                                        'CRISPR'@en   \n",
       "\n",
       "                                      subject5   FIPS      n1     n2  \n",
       "0                                'dopamine'@en  01001  0.8057  0.391  \n",
       "1                          'RNA sequencing'@en  01003  0.8362  0.452  \n",
       "2                                'database'@en  01005  0.6891  0.324  \n",
       "3                    'Alzheimer\\'s disease'@en  01007  0.3368  0.136  \n",
       "4                            'inflammation'@en  01009   0.758  0.199  \n",
       "..                                         ...    ...     ...    ...  \n",
       "617  'BH3 interacting domain death agonist'@en  17045  0.9221  0.173  \n",
       "618                                'Bcl-xL'@en  17047  0.7801  0.188  \n",
       "619                      'Fanconi syndrome'@en  17049  0.9528  0.586  \n",
       "620                        'autoinhibition'@en  17051  0.5575  0.198  \n",
       "621                          'cell biology'@en  17053  0.8982  0.349  \n",
       "\n",
       "[622 rows x 20 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out[out['FIPS'] != '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "out.to_excel('/Users/grantxie/Downloads/NIH/nih_0909_new3.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Tags",
  "kernelspec": {
   "display_name": "kgtk_env",
   "language": "python",
   "name": "kgtk_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
